{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/pachterlab/kb_docs/blob/main/Notebooks/Figure_2/Figure_2c/2_align_ebov_mutations.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "# Align mutated ZEBOV sequences using DIAMOND (translated alignment), Kraken2 (translated alignment), kallisto (translated search), and kallisto (standard workflow)"
      ],
      "metadata": {
        "id": "YntPKDXXg4l7"
      },
      "id": "YntPKDXXg4l7"
    },
    {
      "cell_type": "code",
      "execution_count": 1,
      "id": "c482d079-8c0f-4bee-adff-85dd2c359b54",
      "metadata": {
        "id": "c482d079-8c0f-4bee-adff-85dd2c359b54"
      },
      "outputs": [],
      "source": [
        "import glob\n",
        "# Define number of threads to use for alignments\n",
        "threads = 20 # Change to 2 if not using a TPU runtime"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# Download and define folder with mutated ZEBOV sequences\n",
        "!git clone https://github.com/pachterlab/LSCHWCP_2023.git\n",
        "out = \"/content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files\""
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "G6e6Rz3cP1J2",
        "outputId": "6413c455-98df-46bf-e01e-fd99c25ed7e6"
      },
      "id": "G6e6Rz3cP1J2",
      "execution_count": 2,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Cloning into 'LSCHWCP_2023'...\n",
            "remote: Enumerating objects: 3245, done.\u001b[K\n",
            "remote: Counting objects: 100% (1727/1727), done.\u001b[K\n",
            "remote: Compressing objects: 100% (1083/1083), done.\u001b[K\n",
            "remote: Total 3245 (delta 1129), reused 1060 (delta 642), pack-reused 1518 (from 1)\u001b[K\n",
            "Receiving objects: 100% (3245/3245), 288.47 MiB | 41.35 MiB/s, done.\n",
            "Resolving deltas: 100% (1747/1747), done.\n",
            "Updating files: 100% (409/409), done.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "# Align using DIAMOND (translated alignment)\n",
        "NEW: The DIAMOND alignment was added during Nat Biotech reviews."
      ],
      "metadata": {
        "id": "EvLt-Wt81V9M"
      },
      "id": "EvLt-Wt81V9M"
    },
    {
      "cell_type": "code",
      "source": [
        "# Install gget (from dev branch until release)\n",
        "!pip install -q mysql-connector-python\n",
        "!pip install -q git+https://github.com/pachterlab/gget.git@dev\n",
        "import gget"
      ],
      "metadata": {
        "id": "g5LmJvPm4hYB",
        "outputId": "cf8008f3-e9a2-4e52-efe0-c3d962eb0269",
        "colab": {
          "base_uri": "https://localhost:8080/"
        }
      },
      "id": "g5LmJvPm4hYB",
      "execution_count": 3,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m34.4/34.4 MB\u001b[0m \u001b[31m44.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25l\u001b[?25hdone\n",
            "  Building wheel for gget (setup.py) ... \u001b[?25l\u001b[?25hdone\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# ebola_rdrp_aa.fasta contains the Ebola RdRP amino acid sequence from PalmDB (u55137)\n",
        "ebov_aa_fasta = \"/content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/ebola_rdrp_aa.fasta\""
      ],
      "metadata": {
        "id": "hZBf3kYF5mRm"
      },
      "id": "hZBf3kYF5mRm",
      "execution_count": 4,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "# Align unmutated sequences\n",
        "orig_file = \"/content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/SRR12698539_2_extracted_u10.fa\"\n",
        "\n",
        "diamond_res_orig = gget.diamond(orig_file, ebov_aa_fasta, translated=True)\n",
        "\n",
        "# Save results\n",
        "!mkdir -p diamond_alignment\n",
        "diamond_res_orig.to_csv(\"diamond_alignment/unmutated_diamond.csv\", index=False)"
      ],
      "metadata": {
        "id": "I21eCMpX5QEz",
        "outputId": "63efff72-fad6-4e36-c8dd-b338c4e5b557",
        "colab": {
          "base_uri": "https://localhost:8080/"
        }
      },
      "id": "I21eCMpX5QEz",
      "execution_count": 7,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "INFO:gget.utils:Aligning nucleotide query to amino acid reference (blastx mode).\n",
            "INFO:gget.utils:Creating DIAMOND database and initiating alignment...\n",
            "INFO:gget.utils:DIAMOND alignment complete.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "for mut_file in glob.glob(f\"{out}/*_ms.fa\"):\n",
        "    # Translated alignment\n",
        "    temp_df = gget.diamond(\n",
        "        mut_file,\n",
        "        ebov_aa_fasta,\n",
        "        translated=True,\n",
        "        threads=threads,\n",
        "        verbose=False\n",
        "        )\n",
        "\n",
        "    # Save results\n",
        "    temp_df.to_csv(\"diamond_alignment/\" + mut_file.split(\"/\")[-1].split(\".fa\")[0] + \"_diamond.csv\", index=False)"
      ],
      "metadata": {
        "id": "D-g5zN6462Ro"
      },
      "id": "D-g5zN6462Ro",
      "execution_count": 9,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "id": "9aacd273-113b-4d9d-9305-8080a1a25823",
      "metadata": {
        "id": "9aacd273-113b-4d9d-9305-8080a1a25823"
      },
      "source": [
        "# Align using Kraken2 (translated alignment)"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# Install Kraken2 v1.0.2 (defining version for reproducibility)\n",
        "! git clone https://github.com/DerrickWood/kraken2.git --branch v2.1.2\n",
        "!cd kraken2 && ./install_kraken2.sh ./"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "DmQR7w1JTnmt",
        "outputId": "dc0a8af6-17ce-4d40-90e5-3ecfdd55bcfb"
      },
      "id": "DmQR7w1JTnmt",
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Cloning into 'kraken2'...\n",
            "remote: Enumerating objects: 1064, done.\u001b[K\n",
            "remote: Counting objects: 100% (354/354), done.\u001b[K\n",
            "remote: Compressing objects: 100% (75/75), done.\u001b[K\n",
            "remote: Total 1064 (delta 294), reused 280 (delta 279), pack-reused 710\u001b[K\n",
            "Receiving objects: 100% (1064/1064), 444.61 KiB | 3.47 MiB/s, done.\n",
            "Resolving deltas: 100% (778/778), done.\n",
            "Note: switching to '84b2874e0ba5ffc9abaebe630433a430cd0f69f4'.\n",
            "\n",
            "You are in 'detached HEAD' state. You can look around, make experimental\n",
            "changes and commit them, and you can discard any commits you make in this\n",
            "state without impacting any branches by switching back to a branch.\n",
            "\n",
            "If you want to create a new branch to retain commits you create, you may\n",
            "do so (now or later) by using -c with the switch command. Example:\n",
            "\n",
            "  git switch -c <new-branch-name>\n",
            "\n",
            "Or undo this operation with:\n",
            "\n",
            "  git switch -\n",
            "\n",
            "Turn off this advice by setting config variable advice.detachedHead to false\n",
            "\n",
            "make: Entering directory '/content/kraken2/src'\n",
            "g++ -fopenmp -Wall -std=c++11 -O3 -DLINEAR_PROBING   -c -o estimate_capacity.o estimate_capacity.cc\n",
            "g++ -fopenmp -Wall -std=c++11 -O3 -DLINEAR_PROBING   -c -o seqreader.o seqreader.cc\n",
            "g++ -fopenmp -Wall -std=c++11 -O3 -DLINEAR_PROBING   -c -o mmscanner.o mmscanner.cc\n",
            "g++ -fopenmp -Wall -std=c++11 -O3 -DLINEAR_PROBING   -c -o omp_hack.o omp_hack.cc\n",
            "g++ -fopenmp -Wall -std=c++11 -O3 -DLINEAR_PROBING   -c -o utilities.o utilities.cc\n",
            "g++ -fopenmp -Wall -std=c++11 -O3 -DLINEAR_PROBING -o estimate_capacity estimate_capacity.o seqreader.o mmscanner.o omp_hack.o utilities.o\n",
            "g++ -fopenmp -Wall -std=c++11 -O3 -DLINEAR_PROBING   -c -o build_db.o build_db.cc\n",
            "g++ -fopenmp -Wall -std=c++11 -O3 -DLINEAR_PROBING   -c -o mmap_file.o mmap_file.cc\n",
            "g++ -fopenmp -Wall -std=c++11 -O3 -DLINEAR_PROBING   -c -o compact_hash.o compact_hash.cc\n",
            "\u001b[01m\u001b[Kcompact_hash.cc:\u001b[m\u001b[K In constructor ‘\u001b[01m\u001b[Kkraken2::CompactHashTable::CompactHashTable(size_t, size_t, size_t)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Kcompact_hash.cc:30:27:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kcatching polymorphic type ‘\u001b[01m\u001b[Kclass std::bad_alloc\u001b[m\u001b[K’ by value [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wcatch-value=\u0007-Wcatch-value=\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   30 |   } catch (std::bad_alloc \u001b[01;35m\u001b[Kex\u001b[m\u001b[K) {\n",
            "      |                           \u001b[01;35m\u001b[K^~\u001b[m\u001b[K\n",
            "g++ -fopenmp -Wall -std=c++11 -O3 -DLINEAR_PROBING   -c -o taxonomy.o taxonomy.cc\n",
            "g++ -fopenmp -Wall -std=c++11 -O3 -DLINEAR_PROBING -o build_db build_db.o mmap_file.o compact_hash.o taxonomy.o seqreader.o mmscanner.o omp_hack.o utilities.o\n",
            "g++ -fopenmp -Wall -std=c++11 -O3 -DLINEAR_PROBING   -c -o classify.o classify.cc\n",
            "g++ -fopenmp -Wall -std=c++11 -O3 -DLINEAR_PROBING   -c -o reports.o reports.cc\n",
            "g++ -fopenmp -Wall -std=c++11 -O3 -DLINEAR_PROBING   -c -o hyperloglogplus.o hyperloglogplus.cc\n",
            "g++ -fopenmp -Wall -std=c++11 -O3 -DLINEAR_PROBING   -c -o aa_translate.o aa_translate.cc\n",
            "g++ -fopenmp -Wall -std=c++11 -O3 -DLINEAR_PROBING -o classify classify.o reports.o hyperloglogplus.o mmap_file.o compact_hash.o taxonomy.o seqreader.o mmscanner.o omp_hack.o aa_translate.o utilities.o\n",
            "g++ -fopenmp -Wall -std=c++11 -O3 -DLINEAR_PROBING   -c -o dump_table.o dump_table.cc\n",
            "g++ -fopenmp -Wall -std=c++11 -O3 -DLINEAR_PROBING -o dump_table dump_table.o mmap_file.o compact_hash.o omp_hack.o taxonomy.o reports.o hyperloglogplus.o\n",
            "g++ -fopenmp -Wall -std=c++11 -O3 -DLINEAR_PROBING   -c -o lookup_accession_numbers.o lookup_accession_numbers.cc\n",
            "g++ -fopenmp -Wall -std=c++11 -O3 -DLINEAR_PROBING -o lookup_accession_numbers lookup_accession_numbers.o mmap_file.o omp_hack.o utilities.o\n",
            "cp estimate_capacity build_db classify dump_table lookup_accession_numbers \"/content/kraken2/\"\n",
            "make: Leaving directory '/content/kraken2/src'\n",
            "\n",
            "Kraken 2 installation complete.\n",
            "\n",
            "To make things easier for you, you may want to copy/symlink the following\n",
            "files into a directory in your PATH:\n",
            "  /content/kraken2/kraken2\n",
            "  /content/kraken2/kraken2-build\n",
            "  /content/kraken2/kraken2-inspect\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "id": "0e51015d-535c-432f-9fa3-dab5454a92fc",
      "metadata": {
        "id": "0e51015d-535c-432f-9fa3-dab5454a92fc"
      },
      "source": [
        "Create ebov amino acid ref with NCBI taxID and align with Kraken translated search:"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "kraken2 = \"/content/kraken2/kraken2\"\n",
        "kraken2_build = \"/content/kraken2/kraken2-build\""
      ],
      "metadata": {
        "id": "b0eKabOGUpeE"
      },
      "id": "b0eKabOGUpeE",
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "ad4d1a6c-714e-4078-bce4-a8d9c11437b1",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ad4d1a6c-714e-4078-bce4-a8d9c11437b1",
        "outputId": "207a7ad7-a8e7-4880-8ba1-3b7937573644"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Downloading protein accession to taxon map... done.\n",
            "Downloaded accession to taxon map(s)\n",
            "Downloading taxonomy tree data... done.\n",
            "Uncompressing taxonomy data... done.\n",
            "Untarring taxonomy tree data... done.\n"
          ]
        }
      ],
      "source": [
        "%%time\n",
        "!$kraken2_build --db kraken_ebov_protein --protein --download-taxonomy"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "e966d788-d482-4c1e-a419-70c58c64bdfd",
      "metadata": {
        "id": "e966d788-d482-4c1e-a419-70c58c64bdfd"
      },
      "outputs": [],
      "source": [
        "# ebola_rdrp_aa.fasta contains the Ebola RdRP amino acid sequence from PalmDB (u55137) labeled with the NCBI Ebola taxonomy ID (186538)\n",
        "ebov_aa_fasta = \"/content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/ebola_rdrp_aa.fasta\""
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "38bc986e-19dc-42ee-b9d8-12859aeeab42",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "38bc986e-19dc-42ee-b9d8-12859aeeab42",
        "outputId": "12395904-8eee-463c-e44c-dcc47d054db6"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Added \"/content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/ebola_rdrp_aa.fasta\" to library (kraken_ebov_protein)\n"
          ]
        }
      ],
      "source": [
        "# Add amino acid ebola sequence to Kraken db\n",
        "!$kraken2_build \\\n",
        "    --db kraken_ebov_protein \\\n",
        "    --protein \\\n",
        "    --no-masking \\\n",
        "    --add-to-library $ebov_aa_fasta"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "7b7eb540-2ebd-4522-84b3-65719c81fec7",
      "metadata": {
        "id": "7b7eb540-2ebd-4522-84b3-65719c81fec7"
      },
      "source": [
        "Note: Also tried 31mers (instead of default 15mers) which led to no alignments."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "3632c070-3222-4e04-a715-dfdddcb4f345",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "3632c070-3222-4e04-a715-dfdddcb4f345",
        "outputId": "a12bf2a0-983b-4881-87f5-a9ea96fbf760"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Creating sequence ID to taxonomy ID map (step 1)...\n",
            "Sequence ID to taxonomy ID map complete. [0.018s]\n",
            "Estimating required capacity (step 2)...\n",
            "Estimated hash table requirement: 48272 bytes\n",
            "Capacity estimation complete. [0.020s]\n",
            "Building database files (step 3)...\n",
            "Taxonomy parsed and converted.\n",
            "CHT created with 4 bits reserved for taxid.\n",
            "Completed processing of 1 sequences, 109 aa\n",
            "Writing data to disk...  complete.\n",
            "Database files completed. [6.717s]\n",
            "Database construction complete. [Total: 6.778s]\n"
          ]
        }
      ],
      "source": [
        "# Build index\n",
        "!$kraken2_build \\\n",
        "        --protein \\\n",
        "        --db kraken_ebov_protein \\\n",
        "        --build \\\n",
        "        --threads $threads"
      ]
    },
    {
      "cell_type": "markdown",
      "id": "da52c422-fab7-4b7a-b925-a4f547e1f618",
      "metadata": {
        "id": "da52c422-fab7-4b7a-b925-a4f547e1f618"
      },
      "source": [
        "Align with Kraken2 (translated search):"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "3016d0df-0904-431f-a2ba-0ad53ec34f8d",
      "metadata": {
        "id": "3016d0df-0904-431f-a2ba-0ad53ec34f8d"
      },
      "outputs": [],
      "source": [
        "protein_krakendb = \"kraken_ebov_protein\"\n",
        "protein_kraken_out = \"kraken2_alignment/translated\"\n",
        "!mkdir -p $protein_kraken_out"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "d80ee457-6daf-4b0b-837c-71ab19467db9",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "d80ee457-6daf-4b0b-837c-71ab19467db9",
        "outputId": "0e6439cd-b58c-443e-f0f9-19ff93a77b43"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Loading database information... done.\n",
            "\rProcessed 676 sequences (59488 bp) ...\r676 sequences (0.06 Mbp) processed in 0.007s (5534.2 Kseq/m, 487.01 Mbp/m).\n",
            "  636 sequences classified (94.08%)\n",
            "  40 sequences unclassified (5.92%)\n"
          ]
        }
      ],
      "source": [
        "# Align unmutated sequences\n",
        "orig_file = \"/content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/SRR12698539_2_extracted_u10.fa\"\n",
        "out_file1 = \"0_ebov_snp_0_ms.k2report\"\n",
        "out_file2 = \"0_ebov_snp_0_ms.kraken2\"\n",
        "\n",
        "!$kraken2 \\\n",
        "      --db $protein_krakendb \\\n",
        "      --threads $threads \\\n",
        "      --report-minimizer-data \\\n",
        "      --report $protein_kraken_out/$out_file1 \\\n",
        "      $orig_file > $protein_kraken_out/$out_file2"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "for mut_file in glob.glob(f\"{out}/*_ms.fa\"):\n",
        "    out_file1 = mut_file.split(\"/\")[-1].split(\".fa\")[0] + \".k2report\"\n",
        "    out_file2 = mut_file.split(\"/\")[-1].split(\".fa\")[0] + \".kraken2\"\n",
        "\n",
        "    !$kraken2 \\\n",
        "            --db $protein_krakendb \\\n",
        "            --threads $threads \\\n",
        "            --report-minimizer-data \\\n",
        "            --report $protein_kraken_out/$out_file1 \\\n",
        "            $mut_file > $protein_kraken_out/$out_file2"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "xDinG027YMVL",
        "outputId": "967282fd-3d10-4d76-c8a8-5c1a065e6834"
      },
      "id": "xDinG027YMVL",
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Loading database information... done.\n",
            "\rProcessed 676 sequences (59488 bp) ...\r676 sequences (0.06 Mbp) processed in 0.009s (4597.1 Kseq/m, 404.54 Mbp/m).\n",
            "  230 sequences classified (34.02%)\n",
            "  446 sequences unclassified (65.98%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6230.4 Kseq/m, 548.28 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5136.1 Kseq/m, 451.98 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.009s (4638.1 Kseq/m, 408.15 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.009s (4381.1 Kseq/m, 385.53 Mbp/m).\n",
            "  515 sequences classified (76.18%)\n",
            "  161 sequences unclassified (23.82%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5013.6 Kseq/m, 441.20 Mbp/m).\n",
            "  1 sequences classified (0.15%)\n",
            "  675 sequences unclassified (99.85%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5675.1 Kseq/m, 499.41 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (4964.5 Kseq/m, 436.88 Mbp/m).\n",
            "  8 sequences classified (1.18%)\n",
            "  668 sequences unclassified (98.82%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5409.4 Kseq/m, 476.03 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.009s (4509.7 Kseq/m, 396.85 Mbp/m).\n",
            "  3 sequences classified (0.44%)\n",
            "  673 sequences unclassified (99.56%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5261.4 Kseq/m, 463.00 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5482.6 Kseq/m, 482.47 Mbp/m).\n",
            "  1 sequences classified (0.15%)\n",
            "  675 sequences unclassified (99.85%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5122.5 Kseq/m, 450.78 Mbp/m).\n",
            "  101 sequences classified (14.94%)\n",
            "  575 sequences unclassified (85.06%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5945.5 Kseq/m, 523.20 Mbp/m).\n",
            "  205 sequences classified (30.33%)\n",
            "  471 sequences unclassified (69.67%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6077.3 Kseq/m, 534.80 Mbp/m).\n",
            "  96 sequences classified (14.20%)\n",
            "  580 sequences unclassified (85.80%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5315.2 Kseq/m, 467.73 Mbp/m).\n",
            "  3 sequences classified (0.44%)\n",
            "  673 sequences unclassified (99.56%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5639.6 Kseq/m, 496.28 Mbp/m).\n",
            "  52 sequences classified (7.69%)\n",
            "  624 sequences unclassified (92.31%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6105.7 Kseq/m, 537.30 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.010s (4223.7 Kseq/m, 371.68 Mbp/m).\n",
            "  10 sequences classified (1.48%)\n",
            "  666 sequences unclassified (98.52%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5400.1 Kseq/m, 475.21 Mbp/m).\n",
            "  18 sequences classified (2.66%)\n",
            "  658 sequences unclassified (97.34%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.010s (4259.6 Kseq/m, 374.85 Mbp/m).\n",
            "  145 sequences classified (21.45%)\n",
            "  531 sequences unclassified (78.55%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (4986.5 Kseq/m, 438.81 Mbp/m).\n",
            "  5 sequences classified (0.74%)\n",
            "  671 sequences unclassified (99.26%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6034.8 Kseq/m, 531.06 Mbp/m).\n",
            "  277 sequences classified (40.98%)\n",
            "  399 sequences unclassified (59.02%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6039.3 Kseq/m, 531.46 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5281.9 Kseq/m, 464.81 Mbp/m).\n",
            "  1 sequences classified (0.15%)\n",
            "  675 sequences unclassified (99.85%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5105.7 Kseq/m, 449.31 Mbp/m).\n",
            "  286 sequences classified (42.31%)\n",
            "  390 sequences unclassified (57.69%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (4949.4 Kseq/m, 435.54 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5929.0 Kseq/m, 521.75 Mbp/m).\n",
            "  60 sequences classified (8.88%)\n",
            "  616 sequences unclassified (91.12%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (4992.0 Kseq/m, 439.30 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5099.3 Kseq/m, 448.74 Mbp/m).\n",
            "  139 sequences classified (20.56%)\n",
            "  537 sequences unclassified (79.44%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (4966.9 Kseq/m, 437.09 Mbp/m).\n",
            "  9 sequences classified (1.33%)\n",
            "  667 sequences unclassified (98.67%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (4887.3 Kseq/m, 430.09 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5290.2 Kseq/m, 465.54 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6234.2 Kseq/m, 548.61 Mbp/m).\n",
            "  2 sequences classified (0.30%)\n",
            "  674 sequences unclassified (99.70%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5707.0 Kseq/m, 502.22 Mbp/m).\n",
            "  47 sequences classified (6.95%)\n",
            "  629 sequences unclassified (93.05%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6587.6 Kseq/m, 579.71 Mbp/m).\n",
            "  151 sequences classified (22.34%)\n",
            "  525 sequences unclassified (77.66%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5350.2 Kseq/m, 470.82 Mbp/m).\n",
            "  77 sequences classified (11.39%)\n",
            "  599 sequences unclassified (88.61%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5616.2 Kseq/m, 494.22 Mbp/m).\n",
            "  293 sequences classified (43.34%)\n",
            "  383 sequences unclassified (56.66%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5524.4 Kseq/m, 486.15 Mbp/m).\n",
            "  73 sequences classified (10.80%)\n",
            "  603 sequences unclassified (89.20%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (4964.5 Kseq/m, 436.88 Mbp/m).\n",
            "  7 sequences classified (1.04%)\n",
            "  669 sequences unclassified (98.96%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (4849.9 Kseq/m, 426.79 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.010s (3895.5 Kseq/m, 342.80 Mbp/m).\n",
            "  3 sequences classified (0.44%)\n",
            "  673 sequences unclassified (99.56%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5195.3 Kseq/m, 457.19 Mbp/m).\n",
            "  226 sequences classified (33.43%)\n",
            "  450 sequences unclassified (66.57%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6026.7 Kseq/m, 530.35 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6278.6 Kseq/m, 552.52 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5266.2 Kseq/m, 463.42 Mbp/m).\n",
            "  6 sequences classified (0.89%)\n",
            "  670 sequences unclassified (99.11%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6627.5 Kseq/m, 583.22 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6092.8 Kseq/m, 536.17 Mbp/m).\n",
            "  407 sequences classified (60.21%)\n",
            "  269 sequences unclassified (39.79%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5257.3 Kseq/m, 462.64 Mbp/m).\n",
            "  67 sequences classified (9.91%)\n",
            "  609 sequences unclassified (90.09%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.009s (4321.8 Kseq/m, 380.32 Mbp/m).\n",
            "  2 sequences classified (0.30%)\n",
            "  674 sequences unclassified (99.70%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.010s (3986.6 Kseq/m, 350.82 Mbp/m).\n",
            "  4 sequences classified (0.59%)\n",
            "  672 sequences unclassified (99.41%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5144.6 Kseq/m, 452.72 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5725.6 Kseq/m, 503.85 Mbp/m).\n",
            "  1 sequences classified (0.15%)\n",
            "  675 sequences unclassified (99.85%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5128.3 Kseq/m, 451.29 Mbp/m).\n",
            "  1 sequences classified (0.15%)\n",
            "  675 sequences unclassified (99.85%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6021.4 Kseq/m, 529.88 Mbp/m).\n",
            "  151 sequences classified (22.34%)\n",
            "  525 sequences unclassified (77.66%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6528.2 Kseq/m, 574.49 Mbp/m).\n",
            "  6 sequences classified (0.89%)\n",
            "  670 sequences unclassified (99.11%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5490.7 Kseq/m, 483.18 Mbp/m).\n",
            "  2 sequences classified (0.30%)\n",
            "  674 sequences unclassified (99.70%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5461.9 Kseq/m, 480.65 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5624.7 Kseq/m, 494.98 Mbp/m).\n",
            "  2 sequences classified (0.30%)\n",
            "  674 sequences unclassified (99.70%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5178.1 Kseq/m, 455.67 Mbp/m).\n",
            "  3 sequences classified (0.44%)\n",
            "  673 sequences unclassified (99.56%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6000.9 Kseq/m, 528.08 Mbp/m).\n",
            "  96 sequences classified (14.20%)\n",
            "  580 sequences unclassified (85.80%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.009s (4609.6 Kseq/m, 405.65 Mbp/m).\n",
            "  8 sequences classified (1.18%)\n",
            "  668 sequences unclassified (98.82%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5091.0 Kseq/m, 448.01 Mbp/m).\n",
            "  30 sequences classified (4.44%)\n",
            "  646 sequences unclassified (95.56%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5577.6 Kseq/m, 490.83 Mbp/m).\n",
            "  1 sequences classified (0.15%)\n",
            "  675 sequences unclassified (99.85%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5145.2 Kseq/m, 452.78 Mbp/m).\n",
            "  2 sequences classified (0.30%)\n",
            "  674 sequences unclassified (99.70%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (4911.6 Kseq/m, 432.22 Mbp/m).\n",
            "  9 sequences classified (1.33%)\n",
            "  667 sequences unclassified (98.67%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6489.6 Kseq/m, 571.08 Mbp/m).\n",
            "  2 sequences classified (0.30%)\n",
            "  674 sequences unclassified (99.70%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6282.5 Kseq/m, 552.86 Mbp/m).\n",
            "  2 sequences classified (0.30%)\n",
            "  674 sequences unclassified (99.70%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6340.5 Kseq/m, 557.96 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6278.6 Kseq/m, 552.52 Mbp/m).\n",
            "  32 sequences classified (4.73%)\n",
            "  644 sequences unclassified (95.27%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5336.1 Kseq/m, 469.58 Mbp/m).\n",
            "  79 sequences classified (11.69%)\n",
            "  597 sequences unclassified (88.31%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6412.6 Kseq/m, 564.31 Mbp/m).\n",
            "  1 sequences classified (0.15%)\n",
            "  675 sequences unclassified (99.85%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6333.5 Kseq/m, 557.35 Mbp/m).\n",
            "  3 sequences classified (0.44%)\n",
            "  673 sequences unclassified (99.56%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6044.7 Kseq/m, 531.93 Mbp/m).\n",
            "  5 sequences classified (0.74%)\n",
            "  671 sequences unclassified (99.26%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.009s (4669.0 Kseq/m, 410.88 Mbp/m).\n",
            "  82 sequences classified (12.13%)\n",
            "  594 sequences unclassified (87.87%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6280.6 Kseq/m, 552.69 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5783.5 Kseq/m, 508.95 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5118.0 Kseq/m, 450.38 Mbp/m).\n",
            "  400 sequences classified (59.17%)\n",
            "  276 sequences unclassified (40.83%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5334.0 Kseq/m, 469.40 Mbp/m).\n",
            "  2 sequences classified (0.30%)\n",
            "  674 sequences unclassified (99.70%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5435.5 Kseq/m, 478.33 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6191.4 Kseq/m, 544.85 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5950.7 Kseq/m, 523.66 Mbp/m).\n",
            "  21 sequences classified (3.11%)\n",
            "  655 sequences unclassified (96.89%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (4877.3 Kseq/m, 429.21 Mbp/m).\n",
            "  2 sequences classified (0.30%)\n",
            "  674 sequences unclassified (99.70%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.009s (4436.2 Kseq/m, 390.38 Mbp/m).\n",
            "  17 sequences classified (2.51%)\n",
            "  659 sequences unclassified (97.49%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5384.3 Kseq/m, 473.82 Mbp/m).\n",
            "  531 sequences classified (78.55%)\n",
            "  145 sequences unclassified (21.45%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6055.5 Kseq/m, 532.89 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5016.7 Kseq/m, 441.47 Mbp/m).\n",
            "  72 sequences classified (10.65%)\n",
            "  604 sequences unclassified (89.35%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5927.2 Kseq/m, 521.60 Mbp/m).\n",
            "  1 sequences classified (0.15%)\n",
            "  675 sequences unclassified (99.85%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (4931.3 Kseq/m, 433.96 Mbp/m).\n",
            "  14 sequences classified (2.07%)\n",
            "  662 sequences unclassified (97.93%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6583.3 Kseq/m, 579.33 Mbp/m).\n",
            "  28 sequences classified (4.14%)\n",
            "  648 sequences unclassified (95.86%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6022.3 Kseq/m, 529.96 Mbp/m).\n",
            "  1 sequences classified (0.15%)\n",
            "  675 sequences unclassified (99.85%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5409.4 Kseq/m, 476.03 Mbp/m).\n",
            "  17 sequences classified (2.51%)\n",
            "  659 sequences unclassified (97.49%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5434.1 Kseq/m, 478.20 Mbp/m).\n",
            "  60 sequences classified (8.88%)\n",
            "  616 sequences unclassified (91.12%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5085.9 Kseq/m, 447.56 Mbp/m).\n",
            "  53 sequences classified (7.84%)\n",
            "  623 sequences unclassified (92.16%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5244.4 Kseq/m, 461.51 Mbp/m).\n",
            "  6 sequences classified (0.89%)\n",
            "  670 sequences unclassified (99.11%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6645.9 Kseq/m, 584.84 Mbp/m).\n",
            "  3 sequences classified (0.44%)\n",
            "  673 sequences unclassified (99.56%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6264.1 Kseq/m, 551.24 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6111.2 Kseq/m, 537.79 Mbp/m).\n",
            "  7 sequences classified (1.04%)\n",
            "  669 sequences unclassified (98.96%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (4986.5 Kseq/m, 438.81 Mbp/m).\n",
            "  16 sequences classified (2.37%)\n",
            "  660 sequences unclassified (97.63%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6260.2 Kseq/m, 550.90 Mbp/m).\n",
            "  10 sequences classified (1.48%)\n",
            "  666 sequences unclassified (98.52%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6081.9 Kseq/m, 535.20 Mbp/m).\n",
            "  3 sequences classified (0.44%)\n",
            "  673 sequences unclassified (99.56%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5931.6 Kseq/m, 521.98 Mbp/m).\n",
            "  27 sequences classified (3.99%)\n",
            "  649 sequences unclassified (96.01%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6033.9 Kseq/m, 530.98 Mbp/m).\n",
            "  1 sequences classified (0.15%)\n",
            "  675 sequences unclassified (99.85%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5942.9 Kseq/m, 522.97 Mbp/m).\n",
            "  36 sequences classified (5.33%)\n",
            "  640 sequences unclassified (94.67%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (4829.1 Kseq/m, 424.96 Mbp/m).\n",
            "  103 sequences classified (15.24%)\n",
            "  573 sequences unclassified (84.76%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6955.9 Kseq/m, 612.12 Mbp/m).\n",
            "  11 sequences classified (1.63%)\n",
            "  665 sequences unclassified (98.37%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.009s (4763.4 Kseq/m, 419.18 Mbp/m).\n",
            "  529 sequences classified (78.25%)\n",
            "  147 sequences unclassified (21.75%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6305.0 Kseq/m, 554.84 Mbp/m).\n",
            "  527 sequences classified (77.96%)\n",
            "  149 sequences unclassified (22.04%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5721.5 Kseq/m, 503.50 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5791.8 Kseq/m, 509.68 Mbp/m).\n",
            "  84 sequences classified (12.43%)\n",
            "  592 sequences unclassified (87.57%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5831.8 Kseq/m, 513.20 Mbp/m).\n",
            "  392 sequences classified (57.99%)\n",
            "  284 sequences unclassified (42.01%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6627.5 Kseq/m, 583.22 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6515.7 Kseq/m, 573.38 Mbp/m).\n",
            "  4 sequences classified (0.59%)\n",
            "  672 sequences unclassified (99.41%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5836.0 Kseq/m, 513.57 Mbp/m).\n",
            "  2 sequences classified (0.30%)\n",
            "  674 sequences unclassified (99.70%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5853.7 Kseq/m, 515.12 Mbp/m).\n",
            "  10 sequences classified (1.48%)\n",
            "  666 sequences unclassified (98.52%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5955.9 Kseq/m, 524.12 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (4885.6 Kseq/m, 429.93 Mbp/m).\n",
            "  5 sequences classified (0.74%)\n",
            "  671 sequences unclassified (99.26%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6344.4 Kseq/m, 558.31 Mbp/m).\n",
            "  4 sequences classified (0.59%)\n",
            "  672 sequences unclassified (99.41%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6550.4 Kseq/m, 576.43 Mbp/m).\n",
            "  2 sequences classified (0.30%)\n",
            "  674 sequences unclassified (99.70%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5448.7 Kseq/m, 479.48 Mbp/m).\n",
            "  61 sequences classified (9.02%)\n",
            "  615 sequences unclassified (90.98%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5833.5 Kseq/m, 513.34 Mbp/m).\n",
            "  311 sequences classified (46.01%)\n",
            "  365 sequences unclassified (53.99%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5016.1 Kseq/m, 441.41 Mbp/m).\n",
            "  1 sequences classified (0.15%)\n",
            "  675 sequences unclassified (99.85%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6201.8 Kseq/m, 545.76 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6547.2 Kseq/m, 576.15 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5926.4 Kseq/m, 521.52 Mbp/m).\n",
            "  109 sequences classified (16.12%)\n",
            "  567 sequences unclassified (83.88%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.009s (4317.6 Kseq/m, 379.95 Mbp/m).\n",
            "  26 sequences classified (3.85%)\n",
            "  650 sequences unclassified (96.15%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6533.5 Kseq/m, 574.95 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5382.9 Kseq/m, 473.69 Mbp/m).\n",
            "  8 sequences classified (1.18%)\n",
            "  668 sequences unclassified (98.82%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6397.5 Kseq/m, 562.98 Mbp/m).\n",
            "  4 sequences classified (0.59%)\n",
            "  672 sequences unclassified (99.41%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (4787.0 Kseq/m, 421.25 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5620.8 Kseq/m, 494.63 Mbp/m).\n",
            "  4 sequences classified (0.59%)\n",
            "  672 sequences unclassified (99.41%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.010s (4065.4 Kseq/m, 357.75 Mbp/m).\n",
            "  1 sequences classified (0.15%)\n",
            "  675 sequences unclassified (99.85%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5231.5 Kseq/m, 460.37 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6217.0 Kseq/m, 547.10 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5080.2 Kseq/m, 447.05 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5848.6 Kseq/m, 514.68 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6052.8 Kseq/m, 532.65 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5749.9 Kseq/m, 505.99 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5841.9 Kseq/m, 514.08 Mbp/m).\n",
            "  93 sequences classified (13.76%)\n",
            "  583 sequences unclassified (86.24%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5347.4 Kseq/m, 470.57 Mbp/m).\n",
            "  50 sequences classified (7.40%)\n",
            "  626 sequences unclassified (92.60%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5560.0 Kseq/m, 489.28 Mbp/m).\n",
            "  11 sequences classified (1.63%)\n",
            "  665 sequences unclassified (98.37%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (4851.7 Kseq/m, 426.95 Mbp/m).\n",
            "  1 sequences classified (0.15%)\n",
            "  675 sequences unclassified (99.85%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (4951.2 Kseq/m, 435.70 Mbp/m).\n",
            "  8 sequences classified (1.18%)\n",
            "  668 sequences unclassified (98.82%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5450.9 Kseq/m, 479.68 Mbp/m).\n",
            "  149 sequences classified (22.04%)\n",
            "  527 sequences unclassified (77.96%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.009s (4441.5 Kseq/m, 390.85 Mbp/m).\n",
            "  13 sequences classified (1.92%)\n",
            "  663 sequences unclassified (98.08%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6077.3 Kseq/m, 534.80 Mbp/m).\n",
            "  5 sequences classified (0.74%)\n",
            "  671 sequences unclassified (99.26%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5260.7 Kseq/m, 462.94 Mbp/m).\n",
            "  1 sequences classified (0.15%)\n",
            "  675 sequences unclassified (99.85%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5014.2 Kseq/m, 441.25 Mbp/m).\n",
            "  80 sequences classified (11.83%)\n",
            "  596 sequences unclassified (88.17%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6048.3 Kseq/m, 532.25 Mbp/m).\n",
            "  44 sequences classified (6.51%)\n",
            "  632 sequences unclassified (93.49%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (4938.5 Kseq/m, 434.59 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5513.9 Kseq/m, 485.22 Mbp/m).\n",
            "  6 sequences classified (0.89%)\n",
            "  670 sequences unclassified (99.11%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6361.4 Kseq/m, 559.80 Mbp/m).\n",
            "  9 sequences classified (1.33%)\n",
            "  667 sequences unclassified (98.67%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6356.4 Kseq/m, 559.36 Mbp/m).\n",
            "  536 sequences classified (79.29%)\n",
            "  140 sequences unclassified (20.71%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5522.9 Kseq/m, 486.01 Mbp/m).\n",
            "  22 sequences classified (3.25%)\n",
            "  654 sequences unclassified (96.75%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5400.1 Kseq/m, 475.21 Mbp/m).\n",
            "  4 sequences classified (0.59%)\n",
            "  672 sequences unclassified (99.41%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5487.8 Kseq/m, 482.92 Mbp/m).\n",
            "  12 sequences classified (1.78%)\n",
            "  664 sequences unclassified (98.22%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5687.8 Kseq/m, 500.53 Mbp/m).\n",
            "  3 sequences classified (0.44%)\n",
            "  673 sequences unclassified (99.56%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5901.4 Kseq/m, 519.32 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5315.9 Kseq/m, 467.80 Mbp/m).\n",
            "  28 sequences classified (4.14%)\n",
            "  648 sequences unclassified (95.86%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6075.5 Kseq/m, 534.64 Mbp/m).\n",
            "  34 sequences classified (5.03%)\n",
            "  642 sequences unclassified (94.97%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5742.6 Kseq/m, 505.35 Mbp/m).\n",
            "  535 sequences classified (79.14%)\n",
            "  141 sequences unclassified (20.86%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5574.5 Kseq/m, 490.56 Mbp/m).\n",
            "  154 sequences classified (22.78%)\n",
            "  522 sequences unclassified (77.22%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5406.6 Kseq/m, 475.78 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6167.9 Kseq/m, 542.77 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5686.2 Kseq/m, 500.39 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5768.7 Kseq/m, 507.65 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5838.5 Kseq/m, 513.79 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5205.3 Kseq/m, 458.07 Mbp/m).\n",
            "  65 sequences classified (9.62%)\n",
            "  611 sequences unclassified (90.38%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5244.4 Kseq/m, 461.51 Mbp/m).\n",
            "  38 sequences classified (5.62%)\n",
            "  638 sequences unclassified (94.38%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6142.7 Kseq/m, 540.55 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6619.9 Kseq/m, 582.55 Mbp/m).\n",
            "  1 sequences classified (0.15%)\n",
            "  675 sequences unclassified (99.85%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6224.7 Kseq/m, 547.77 Mbp/m).\n",
            "  1 sequences classified (0.15%)\n",
            "  675 sequences unclassified (99.85%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.009s (4437.2 Kseq/m, 390.47 Mbp/m).\n",
            "  4 sequences classified (0.59%)\n",
            "  672 sequences unclassified (99.41%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5646.7 Kseq/m, 496.91 Mbp/m).\n",
            "  84 sequences classified (12.43%)\n",
            "  592 sequences unclassified (87.57%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6088.3 Kseq/m, 535.77 Mbp/m).\n",
            "  401 sequences classified (59.32%)\n",
            "  275 sequences unclassified (40.68%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5336.8 Kseq/m, 469.64 Mbp/m).\n",
            "  404 sequences classified (59.76%)\n",
            "  272 sequences unclassified (40.24%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6048.3 Kseq/m, 532.25 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5404.4 Kseq/m, 475.59 Mbp/m).\n",
            "  406 sequences classified (60.06%)\n",
            "  270 sequences unclassified (39.94%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5388.6 Kseq/m, 474.20 Mbp/m).\n",
            "  110 sequences classified (16.27%)\n",
            "  566 sequences unclassified (83.73%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6164.1 Kseq/m, 542.44 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5817.6 Kseq/m, 511.94 Mbp/m).\n",
            "  6 sequences classified (0.89%)\n",
            "  670 sequences unclassified (99.11%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5554.6 Kseq/m, 488.81 Mbp/m).\n",
            "  63 sequences classified (9.32%)\n",
            "  613 sequences unclassified (90.68%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6402.5 Kseq/m, 563.42 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (4991.4 Kseq/m, 439.24 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5774.5 Kseq/m, 508.15 Mbp/m).\n",
            "  206 sequences classified (30.47%)\n",
            "  470 sequences unclassified (69.53%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6107.5 Kseq/m, 537.46 Mbp/m).\n",
            "  50 sequences classified (7.40%)\n",
            "  626 sequences unclassified (92.60%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5971.7 Kseq/m, 525.51 Mbp/m).\n",
            "  516 sequences classified (76.33%)\n",
            "  160 sequences unclassified (23.67%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6453.5 Kseq/m, 567.90 Mbp/m).\n",
            "  1 sequences classified (0.15%)\n",
            "  675 sequences unclassified (99.85%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5732.9 Kseq/m, 504.49 Mbp/m).\n",
            "  70 sequences classified (10.36%)\n",
            "  606 sequences unclassified (89.64%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5780.2 Kseq/m, 508.66 Mbp/m).\n",
            "  145 sequences classified (21.45%)\n",
            "  531 sequences unclassified (78.55%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5402.2 Kseq/m, 475.40 Mbp/m).\n",
            "  2 sequences classified (0.30%)\n",
            "  674 sequences unclassified (99.70%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6326.6 Kseq/m, 556.74 Mbp/m).\n",
            "  19 sequences classified (2.81%)\n",
            "  657 sequences unclassified (97.19%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5845.2 Kseq/m, 514.38 Mbp/m).\n",
            "  17 sequences classified (2.51%)\n",
            "  659 sequences unclassified (97.49%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6126.9 Kseq/m, 539.17 Mbp/m).\n",
            "  9 sequences classified (1.33%)\n",
            "  667 sequences unclassified (98.67%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6717.5 Kseq/m, 591.14 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6286.4 Kseq/m, 553.21 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5372.2 Kseq/m, 472.75 Mbp/m).\n",
            "  291 sequences classified (43.05%)\n",
            "  385 sequences unclassified (56.95%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6239.0 Kseq/m, 549.04 Mbp/m).\n",
            "  5 sequences classified (0.74%)\n",
            "  671 sequences unclassified (99.26%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.010s (4183.6 Kseq/m, 368.16 Mbp/m).\n",
            "  307 sequences classified (45.41%)\n",
            "  369 sequences unclassified (54.59%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6476.1 Kseq/m, 569.90 Mbp/m).\n",
            "  10 sequences classified (1.48%)\n",
            "  666 sequences unclassified (98.52%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5641.2 Kseq/m, 496.42 Mbp/m).\n",
            "  167 sequences classified (24.70%)\n",
            "  509 sequences unclassified (75.30%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (4824.0 Kseq/m, 424.51 Mbp/m).\n",
            "  161 sequences classified (23.82%)\n",
            "  515 sequences unclassified (76.18%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6142.7 Kseq/m, 540.55 Mbp/m).\n",
            "  409 sequences classified (60.50%)\n",
            "  267 sequences unclassified (39.50%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5610.7 Kseq/m, 493.74 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5171.5 Kseq/m, 455.09 Mbp/m).\n",
            "  47 sequences classified (6.95%)\n",
            "  629 sequences unclassified (93.05%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5389.3 Kseq/m, 474.26 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5507.9 Kseq/m, 484.69 Mbp/m).\n",
            "  199 sequences classified (29.44%)\n",
            "  477 sequences unclassified (70.56%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6494.8 Kseq/m, 571.54 Mbp/m).\n",
            "  407 sequences classified (60.21%)\n",
            "  269 sequences unclassified (39.79%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5989.4 Kseq/m, 527.06 Mbp/m).\n",
            "  2 sequences classified (0.30%)\n",
            "  674 sequences unclassified (99.70%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6770.2 Kseq/m, 595.77 Mbp/m).\n",
            "  2 sequences classified (0.30%)\n",
            "  674 sequences unclassified (99.70%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5339.7 Kseq/m, 469.89 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5357.3 Kseq/m, 471.44 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6391.4 Kseq/m, 562.45 Mbp/m).\n",
            "  4 sequences classified (0.59%)\n",
            "  672 sequences unclassified (99.41%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6160.4 Kseq/m, 542.11 Mbp/m).\n",
            "  44 sequences classified (6.51%)\n",
            "  632 sequences unclassified (93.49%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6198.0 Kseq/m, 545.43 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5064.3 Kseq/m, 445.66 Mbp/m).\n",
            "  63 sequences classified (9.32%)\n",
            "  613 sequences unclassified (90.68%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5858.7 Kseq/m, 515.57 Mbp/m).\n",
            "  3 sequences classified (0.44%)\n",
            "  673 sequences unclassified (99.56%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6583.3 Kseq/m, 579.33 Mbp/m).\n",
            "  288 sequences classified (42.60%)\n",
            "  388 sequences unclassified (57.40%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6320.7 Kseq/m, 556.22 Mbp/m).\n",
            "  10 sequences classified (1.48%)\n",
            "  666 sequences unclassified (98.52%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6134.3 Kseq/m, 539.82 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5939.4 Kseq/m, 522.67 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5728.0 Kseq/m, 504.06 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5447.2 Kseq/m, 479.36 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5661.6 Kseq/m, 498.22 Mbp/m).\n",
            "  22 sequences classified (3.25%)\n",
            "  654 sequences unclassified (96.75%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6081.0 Kseq/m, 535.12 Mbp/m).\n",
            "  208 sequences classified (30.77%)\n",
            "  468 sequences unclassified (69.23%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6358.4 Kseq/m, 559.54 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6027.6 Kseq/m, 530.43 Mbp/m).\n",
            "  7 sequences classified (1.04%)\n",
            "  669 sequences unclassified (98.96%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5482.6 Kseq/m, 482.47 Mbp/m).\n",
            "  79 sequences classified (11.69%)\n",
            "  597 sequences unclassified (88.31%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6450.4 Kseq/m, 567.63 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6316.8 Kseq/m, 555.88 Mbp/m).\n",
            "  57 sequences classified (8.43%)\n",
            "  619 sequences unclassified (91.57%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6037.5 Kseq/m, 531.30 Mbp/m).\n",
            "  162 sequences classified (23.96%)\n",
            "  514 sequences unclassified (76.04%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5681.5 Kseq/m, 499.97 Mbp/m).\n",
            "  213 sequences classified (31.51%)\n",
            "  463 sequences unclassified (68.49%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5656.9 Kseq/m, 497.81 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6217.0 Kseq/m, 547.10 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5884.2 Kseq/m, 517.81 Mbp/m).\n",
            "  405 sequences classified (59.91%)\n",
            "  271 sequences unclassified (40.09%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6160.4 Kseq/m, 542.11 Mbp/m).\n",
            "  64 sequences classified (9.47%)\n",
            "  612 sequences unclassified (90.53%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6013.3 Kseq/m, 529.17 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6106.6 Kseq/m, 537.38 Mbp/m).\n",
            "  3 sequences classified (0.44%)\n",
            "  673 sequences unclassified (99.56%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5362.2 Kseq/m, 471.88 Mbp/m).\n",
            "  3 sequences classified (0.44%)\n",
            "  673 sequences unclassified (99.56%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5210.0 Kseq/m, 458.48 Mbp/m).\n",
            "  30 sequences classified (4.44%)\n",
            "  646 sequences unclassified (95.56%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5857.0 Kseq/m, 515.42 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5627.1 Kseq/m, 495.18 Mbp/m).\n",
            "  53 sequences classified (7.84%)\n",
            "  623 sequences unclassified (92.16%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6020.5 Kseq/m, 529.80 Mbp/m).\n",
            "  405 sequences classified (59.91%)\n",
            "  271 sequences unclassified (40.09%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5985.8 Kseq/m, 526.75 Mbp/m).\n",
            "  3 sequences classified (0.44%)\n",
            "  673 sequences unclassified (99.56%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6062.8 Kseq/m, 533.52 Mbp/m).\n",
            "  49 sequences classified (7.25%)\n",
            "  627 sequences unclassified (92.75%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6235.2 Kseq/m, 548.70 Mbp/m).\n",
            "  1 sequences classified (0.15%)\n",
            "  675 sequences unclassified (99.85%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6456.5 Kseq/m, 568.18 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5677.5 Kseq/m, 499.62 Mbp/m).\n",
            "  96 sequences classified (14.20%)\n",
            "  580 sequences unclassified (85.80%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5510.9 Kseq/m, 484.96 Mbp/m).\n",
            "  1 sequences classified (0.15%)\n",
            "  675 sequences unclassified (99.85%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5078.3 Kseq/m, 446.89 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5453.8 Kseq/m, 479.94 Mbp/m).\n",
            "  5 sequences classified (0.74%)\n",
            "  671 sequences unclassified (99.26%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6205.6 Kseq/m, 546.10 Mbp/m).\n",
            "  4 sequences classified (0.59%)\n",
            "  672 sequences unclassified (99.41%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6288.4 Kseq/m, 553.38 Mbp/m).\n",
            "  15 sequences classified (2.22%)\n",
            "  661 sequences unclassified (97.78%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5531.9 Kseq/m, 486.81 Mbp/m).\n",
            "  546 sequences classified (80.77%)\n",
            "  130 sequences unclassified (19.23%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6016.9 Kseq/m, 529.49 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5841.0 Kseq/m, 514.01 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5851.1 Kseq/m, 514.90 Mbp/m).\n",
            "  12 sequences classified (1.78%)\n",
            "  664 sequences unclassified (98.22%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (4997.5 Kseq/m, 439.78 Mbp/m).\n",
            "  540 sequences classified (79.88%)\n",
            "  136 sequences unclassified (20.12%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5222.1 Kseq/m, 459.54 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6089.2 Kseq/m, 535.85 Mbp/m).\n",
            "  4 sequences classified (0.59%)\n",
            "  672 sequences unclassified (99.41%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5704.6 Kseq/m, 502.01 Mbp/m).\n",
            "  23 sequences classified (3.40%)\n",
            "  653 sequences unclassified (96.60%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5609.2 Kseq/m, 493.61 Mbp/m).\n",
            "  218 sequences classified (32.25%)\n",
            "  458 sequences unclassified (67.75%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5348.8 Kseq/m, 470.69 Mbp/m).\n",
            "  310 sequences classified (45.86%)\n",
            "  366 sequences unclassified (54.14%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6458.6 Kseq/m, 568.36 Mbp/m).\n",
            "  1 sequences classified (0.15%)\n",
            "  675 sequences unclassified (99.85%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.008s (5277.1 Kseq/m, 464.39 Mbp/m).\n",
            "  111 sequences classified (16.42%)\n",
            "  565 sequences unclassified (83.58%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6651.4 Kseq/m, 585.32 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6197.1 Kseq/m, 545.34 Mbp/m).\n",
            "  1 sequences classified (0.15%)\n",
            "  675 sequences unclassified (99.85%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5627.9 Kseq/m, 495.25 Mbp/m).\n",
            "  3 sequences classified (0.44%)\n",
            "  673 sequences unclassified (99.56%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6300.1 Kseq/m, 554.41 Mbp/m).\n",
            "  5 sequences classified (0.74%)\n",
            "  671 sequences unclassified (99.26%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6313.8 Kseq/m, 555.62 Mbp/m).\n",
            "  75 sequences classified (11.09%)\n",
            "  601 sequences unclassified (88.91%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6384.4 Kseq/m, 561.83 Mbp/m).\n",
            "  29 sequences classified (4.29%)\n",
            "  647 sequences unclassified (95.71%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5463.4 Kseq/m, 480.78 Mbp/m).\n",
            "  277 sequences classified (40.98%)\n",
            "  399 sequences unclassified (59.02%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6490.6 Kseq/m, 571.18 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6756.6 Kseq/m, 594.58 Mbp/m).\n",
            "  3 sequences classified (0.44%)\n",
            "  673 sequences unclassified (99.56%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6130.6 Kseq/m, 539.49 Mbp/m).\n",
            "  43 sequences classified (6.36%)\n",
            "  633 sequences unclassified (93.64%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6136.2 Kseq/m, 539.98 Mbp/m).\n",
            "  14 sequences classified (2.07%)\n",
            "  662 sequences unclassified (97.93%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6513.6 Kseq/m, 573.19 Mbp/m).\n",
            "  197 sequences classified (29.14%)\n",
            "  479 sequences unclassified (70.86%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5963.0 Kseq/m, 524.74 Mbp/m).\n",
            "  66 sequences classified (9.76%)\n",
            "  610 sequences unclassified (90.24%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6175.4 Kseq/m, 543.43 Mbp/m).\n",
            "  3 sequences classified (0.44%)\n",
            "  673 sequences unclassified (99.56%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5741.8 Kseq/m, 505.28 Mbp/m).\n",
            "  25 sequences classified (3.70%)\n",
            "  651 sequences unclassified (96.30%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6398.5 Kseq/m, 563.07 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.009s (4596.0 Kseq/m, 404.45 Mbp/m).\n",
            "  287 sequences classified (42.46%)\n",
            "  389 sequences unclassified (57.54%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6050.1 Kseq/m, 532.41 Mbp/m).\n",
            "  3 sequences classified (0.44%)\n",
            "  673 sequences unclassified (99.56%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6482.3 Kseq/m, 570.45 Mbp/m).\n",
            "  29 sequences classified (4.29%)\n",
            "  647 sequences unclassified (95.71%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (5558.4 Kseq/m, 489.14 Mbp/m).\n",
            "  541 sequences classified (80.03%)\n",
            "  135 sequences unclassified (19.97%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6205.6 Kseq/m, 546.10 Mbp/m).\n",
            "  207 sequences classified (30.62%)\n",
            "  469 sequences unclassified (69.38%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6167.9 Kseq/m, 542.77 Mbp/m).\n",
            "  125 sequences classified (18.49%)\n",
            "  551 sequences unclassified (81.51%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.006s (6374.4 Kseq/m, 560.94 Mbp/m).\n",
            "  0 sequences classified (0.00%)\n",
            "  676 sequences unclassified (100.00%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6169.8 Kseq/m, 542.94 Mbp/m).\n",
            "  62 sequences classified (9.17%)\n",
            "  614 sequences unclassified (90.83%)\n",
            "Loading database information... done.\n",
            "676 sequences (0.06 Mbp) processed in 0.007s (6212.3 Kseq/m, 546.68 Mbp/m).\n",
            "  6 sequences classified (0.89%)\n",
            "  670 sequences unclassified (99.11%)\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "id": "3f27390b-57cc-4304-b8c7-ff5e67fc8dcd",
      "metadata": {
        "id": "3f27390b-57cc-4304-b8c7-ff5e67fc8dcd"
      },
      "source": [
        "# Align using kallisto bustools (translated alignment)"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# Install kallisto from source\n",
        "!git clone https://github.com/pachterlab/kallisto.git\n",
        "!cd kallisto && mkdir build && cd build && cmake .. && make"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "157kYvKYY-a3",
        "outputId": "6ee673ed-9c7a-4697-d22a-71c7bdd222ee"
      },
      "id": "157kYvKYY-a3",
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Cloning into 'kallisto'...\n",
            "remote: Enumerating objects: 8080, done.\u001b[K\n",
            "remote: Counting objects: 100% (1162/1162), done.\u001b[K\n",
            "remote: Compressing objects: 100% (313/313), done.\u001b[K\n",
            "remote: Total 8080 (delta 880), reused 985 (delta 847), pack-reused 6918\u001b[K\n",
            "Receiving objects: 100% (8080/8080), 9.11 MiB | 6.93 MiB/s, done.\n",
            "Resolving deltas: 100% (5306/5306), done.\n",
            "\u001b[0mCMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):\n",
            "  Compatibility with CMake < 3.5 will be removed from a future version of\n",
            "  CMake.\n",
            "\n",
            "  Update the VERSION argument <min> value or use a ...<max> suffix to tell\n",
            "  CMake that the project does not need compatibility with older versions.\n",
            "\n",
            "\u001b[0m\n",
            "-- The C compiler identification is GNU 11.4.0\n",
            "-- The CXX compiler identification is GNU 11.4.0\n",
            "-- Detecting C compiler ABI info\n",
            "-- Detecting C compiler ABI info - done\n",
            "-- Check for working C compiler: /usr/bin/cc - skipped\n",
            "-- Detecting C compile features\n",
            "-- Detecting C compile features - done\n",
            "-- Detecting CXX compiler ABI info\n",
            "-- Detecting CXX compiler ABI info - done\n",
            "-- Check for working CXX compiler: /usr/bin/c++ - skipped\n",
            "-- Detecting CXX compile features\n",
            "-- Detecting CXX compile features - done\n",
            "-- Performing Test COMPILER_SUPPORTS_CXX17\n",
            "-- Performing Test COMPILER_SUPPORTS_CXX17 - Success\n",
            "\u001b[0mshared build\u001b[0m\n",
            "-- Performing Test CMAKE_HAVE_LIBC_PTHREAD\n",
            "-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success\n",
            "-- Found Threads: TRUE  \n",
            "-- Found ZLIB: /usr/lib/x86_64-linux-gnu/libz.so (found version \"1.2.11\")  \n",
            "-- Configuring done (1.3s)\n",
            "-- Generating done (0.0s)\n",
            "-- Build files have been written to: /content/kallisto/build\n",
            "[  3%] \u001b[34m\u001b[1mCreating directories for 'bifrost'\u001b[0m\n",
            "[  7%] \u001b[34m\u001b[1mNo download step for 'bifrost'\u001b[0m\n",
            "[ 11%] \u001b[34m\u001b[1mNo update step for 'bifrost'\u001b[0m\n",
            "[ 15%] \u001b[34m\u001b[1mNo patch step for 'bifrost'\u001b[0m\n",
            "[ 19%] \u001b[34m\u001b[1mPerforming configure step for 'bifrost'\u001b[0m\n",
            "\u001b[0mCMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):\n",
            "  Compatibility with CMake < 3.5 will be removed from a future version of\n",
            "  CMake.\n",
            "\n",
            "  Update the VERSION argument <min> value or use a ...<max> suffix to tell\n",
            "  CMake that the project does not need compatibility with older versions.\n",
            "\n",
            "\u001b[0m\n",
            "-- The C compiler identification is GNU 11.4.0\n",
            "-- The CXX compiler identification is GNU 11.4.0\n",
            "-- Detecting C compiler ABI info\n",
            "-- Detecting C compiler ABI info - done\n",
            "-- Check for working C compiler: /usr/bin/cc - skipped\n",
            "-- Detecting C compile features\n",
            "-- Detecting C compile features - done\n",
            "-- Detecting CXX compiler ABI info\n",
            "-- Detecting CXX compiler ABI info - done\n",
            "-- Check for working CXX compiler: /usr/bin/c++ - skipped\n",
            "-- Detecting CXX compile features\n",
            "-- Detecting CXX compile features - done\n",
            "-- Performing Test CMAKE_HAVE_LIBC_PTHREAD\n",
            "-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success\n",
            "-- Found Threads: TRUE  \n",
            "\u001b[0mCompilation architecture: native\u001b[0m\n",
            "\u001b[0mBuild type: Release\u001b[0m\n",
            "\u001b[0mMaximum k-mer size: 31\u001b[0m\n",
            "\u001b[0mMaximum g-mer size: 31\u001b[0m\n",
            "-- Found ZLIB: /usr/lib/x86_64-linux-gnu/libz.so (found version \"1.2.11\")  \n",
            "-- Configuring done (0.5s)\n",
            "-- Generating done (0.0s)\n",
            "-- Build files have been written to: /content/kallisto/ext/bifrost/build\n",
            "[ 23%] \u001b[34m\u001b[1mPerforming build step for 'bifrost'\u001b[0m\n",
            "[  3%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_static.dir/Bifrost.cpp.o\u001b[0m\n",
            "[  6%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_static.dir/BitContainer.cpp.o\u001b[0m\n",
            "[  9%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_static.dir/BlockedBloomFilter.cpp.o\u001b[0m\n",
            "[ 12%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_static.dir/ColorSet.cpp.o\u001b[0m\n",
            "[ 15%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_static.dir/CompressedCoverage.cpp.o\u001b[0m\n",
            "[ 18%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_static.dir/CompressedSequence.cpp.o\u001b[0m\n",
            "[ 21%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_static.dir/FASTX_Parser.cpp.o\u001b[0m\n",
            "[ 25%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_static.dir/GFA_Parser.cpp.o\u001b[0m\n",
            "[ 28%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_static.dir/Kmer.cpp.o\u001b[0m\n",
            "[ 31%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_static.dir/KmerIterator.cpp.o\u001b[0m\n",
            "[ 34%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_static.dir/MinimizerIndex.cpp.o\u001b[0m\n",
            "[ 37%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_static.dir/TinyBitmap.cpp.o\u001b[0m\n",
            "[ 40%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_static.dir/UnitigMap.cpp.o\u001b[0m\n",
            "[ 43%] \u001b[32mBuilding C object src/CMakeFiles/bifrost_static.dir/roaring.c.o\u001b[0m\n",
            "[ 46%] \u001b[32m\u001b[1mLinking CXX static library libbifrost.a\u001b[0m\n",
            "[ 46%] Built target bifrost_static\n",
            "[ 50%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_dynamic.dir/Bifrost.cpp.o\u001b[0m\n",
            "[ 53%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_dynamic.dir/BitContainer.cpp.o\u001b[0m\n",
            "[ 56%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_dynamic.dir/BlockedBloomFilter.cpp.o\u001b[0m\n",
            "[ 59%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_dynamic.dir/ColorSet.cpp.o\u001b[0m\n",
            "[ 62%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_dynamic.dir/CompressedCoverage.cpp.o\u001b[0m\n",
            "[ 65%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_dynamic.dir/CompressedSequence.cpp.o\u001b[0m\n",
            "[ 68%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_dynamic.dir/FASTX_Parser.cpp.o\u001b[0m\n",
            "[ 71%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_dynamic.dir/GFA_Parser.cpp.o\u001b[0m\n",
            "[ 75%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_dynamic.dir/Kmer.cpp.o\u001b[0m\n",
            "[ 78%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_dynamic.dir/KmerIterator.cpp.o\u001b[0m\n",
            "[ 81%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_dynamic.dir/MinimizerIndex.cpp.o\u001b[0m\n",
            "[ 84%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_dynamic.dir/TinyBitmap.cpp.o\u001b[0m\n",
            "[ 87%] \u001b[32mBuilding CXX object src/CMakeFiles/bifrost_dynamic.dir/UnitigMap.cpp.o\u001b[0m\n",
            "[ 90%] \u001b[32mBuilding C object src/CMakeFiles/bifrost_dynamic.dir/roaring.c.o\u001b[0m\n",
            "[ 93%] \u001b[32m\u001b[1mLinking CXX shared library libbifrost.so\u001b[0m\n",
            "[ 93%] Built target bifrost_dynamic\n",
            "[ 96%] \u001b[32mBuilding CXX object src/CMakeFiles/Bifrost.dir/Bifrost.cpp.o\u001b[0m\n",
            "[100%] \u001b[32m\u001b[1mLinking CXX executable Bifrost\u001b[0m\n",
            "[100%] Built target Bifrost\n",
            "[ 26%] \u001b[34m\u001b[1mNo install step for 'bifrost'\u001b[0m\n",
            "[ 30%] \u001b[34m\u001b[1mCompleted 'bifrost'\u001b[0m\n",
            "[ 30%] Built target bifrost\n",
            "[ 34%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/BUSData.cpp.o\u001b[0m\n",
            "[ 38%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/BUSTools.cpp.o\u001b[0m\n",
            "[ 42%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/Bootstrap.cpp.o\u001b[0m\n",
            "[ 46%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/GeneModel.cpp.o\u001b[0m\n",
            "[ 50%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/H5Writer.cpp.o\u001b[0m\n",
            "[ 53%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/KmerIndex.cpp.o\u001b[0m\n",
            "[ 57%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/MinCollector.cpp.o\u001b[0m\n",
            "[ 61%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/PlaintextWriter.cpp.o\u001b[0m\n",
            "[ 65%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/ProcessReads.cpp.o\u001b[0m\n",
            "[ 69%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/PseudoBam.cpp.o\u001b[0m\n",
            "[ 73%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/common.cpp.o\u001b[0m\n",
            "[ 76%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/h5utils.cpp.o\u001b[0m\n",
            "[ 80%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/hash.cpp.o\u001b[0m\n",
            "[ 84%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/main.cpp.o\u001b[0m\n",
            "[ 88%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto_core.dir/weights.cpp.o\u001b[0m\n",
            "[ 92%] \u001b[32m\u001b[1mLinking CXX static library libkallisto_core.a\u001b[0m\n",
            "[ 92%] Built target kallisto_core\n",
            "[ 96%] \u001b[32mBuilding CXX object src/CMakeFiles/kallisto.dir/main.cpp.o\u001b[0m\n",
            "[100%] \u001b[32m\u001b[1mLinking CXX executable kallisto\u001b[0m\n",
            "[100%] Built target kallisto\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# Install bustools from source\n",
        "!git clone https://github.com/BUStools/bustools.git\n",
        "!cd bustools && mkdir build && cd build && cmake .. && make"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "DQzbvJnWZhQC",
        "outputId": "624c6028-d83e-471b-bc7e-aeb50f30ebf9"
      },
      "id": "DQzbvJnWZhQC",
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Cloning into 'bustools'...\n",
            "remote: Enumerating objects: 3582, done.\u001b[K\n",
            "remote: Counting objects: 100% (566/566), done.\u001b[K\n",
            "remote: Compressing objects: 100% (206/206), done.\u001b[K\n",
            "remote: Total 3582 (delta 449), reused 360 (delta 360), pack-reused 3016\u001b[K\n",
            "Receiving objects: 100% (3582/3582), 3.42 MiB | 13.41 MiB/s, done.\n",
            "Resolving deltas: 100% (1467/1467), done.\n",
            "\u001b[0mCMake Deprecation Warning at CMakeLists.txt:1 (cmake_minimum_required):\n",
            "  Compatibility with CMake < 3.5 will be removed from a future version of\n",
            "  CMake.\n",
            "\n",
            "  Update the VERSION argument <min> value or use a ...<max> suffix to tell\n",
            "  CMake that the project does not need compatibility with older versions.\n",
            "\n",
            "\u001b[0m\n",
            "-- The C compiler identification is GNU 11.4.0\n",
            "-- The CXX compiler identification is GNU 11.4.0\n",
            "-- Detecting C compiler ABI info\n",
            "-- Detecting C compiler ABI info - done\n",
            "-- Check for working C compiler: /usr/bin/cc - skipped\n",
            "-- Detecting C compile features\n",
            "-- Detecting C compile features - done\n",
            "-- Detecting CXX compiler ABI info\n",
            "-- Detecting CXX compiler ABI info - done\n",
            "-- Check for working CXX compiler: /usr/bin/c++ - skipped\n",
            "-- Detecting CXX compile features\n",
            "-- Detecting CXX compile features - done\n",
            "\u001b[0mrelease mode\u001b[0m\n",
            "-- Performing Test CMAKE_HAVE_LIBC_PTHREAD\n",
            "-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success\n",
            "-- Found Threads: TRUE  \n",
            "-- Found ZLIB: /usr/lib/x86_64-linux-gnu/libz.so (found version \"1.2.11\")  \n",
            "-- Configuring done (0.5s)\n",
            "-- Generating done (0.0s)\n",
            "-- Build files have been written to: /content/bustools/build\n",
            "[  3%] \u001b[32mBuilding CXX object src/CMakeFiles/bustools_core.dir/BUSData.cpp.o\u001b[0m\n",
            "[  7%] \u001b[32mBuilding CXX object src/CMakeFiles/bustools_core.dir/Common.cpp.o\u001b[0m\n",
            "[ 11%] \u001b[32mBuilding CXX object src/CMakeFiles/bustools_core.dir/bustools_capture.cpp.o\u001b[0m\n",
            "[ 15%] \u001b[32mBuilding CXX object src/CMakeFiles/bustools_core.dir/bustools_clusterhist.cpp.o\u001b[0m\n",
            "[ 19%] \u001b[32mBuilding CXX object src/CMakeFiles/bustools_core.dir/bustools_collapse.cpp.o\u001b[0m\n",
            "[ 23%] \u001b[32mBuilding CXX object src/CMakeFiles/bustools_core.dir/bustools_compress.cpp.o\u001b[0m\n",
            "[ 26%] \u001b[32mBuilding CXX object src/CMakeFiles/bustools_core.dir/bustools_correct.cpp.o\u001b[0m\n",
            "[ 30%] \u001b[32mBuilding CXX object src/CMakeFiles/bustools_core.dir/bustools_count.cpp.o\u001b[0m\n",
            "[ 34%] \u001b[32mBuilding CXX object src/CMakeFiles/bustools_core.dir/bustools_decompress.cpp.o\u001b[0m\n",
            "[ 38%] \u001b[32mBuilding CXX object src/CMakeFiles/bustools_core.dir/bustools_extract.cpp.o\u001b[0m\n",
            "[ 42%] \u001b[32mBuilding CXX object src/CMakeFiles/bustools_core.dir/bustools_inspect.cpp.o\u001b[0m\n",
            "[ 46%] \u001b[32mBuilding CXX object src/CMakeFiles/bustools_core.dir/bustools_linker.cpp.o\u001b[0m\n",
            "[ 50%] \u001b[32mBuilding CXX object src/CMakeFiles/bustools_core.dir/bustools_main.cpp.o\u001b[0m\n",
            "[ 53%] \u001b[32mBuilding CXX object src/CMakeFiles/bustools_core.dir/bustools_mash.cpp.o\u001b[0m\n",
            "[ 57%] \u001b[32mBuilding CXX object src/CMakeFiles/bustools_core.dir/bustools_merge.cpp.o\u001b[0m\n",
            "[ 61%] \u001b[32mBuilding CXX object src/CMakeFiles/bustools_core.dir/bustools_predict.cpp.o\u001b[0m\n",
            "[ 65%] \u001b[32mBuilding CXX object src/CMakeFiles/bustools_core.dir/bustools_project.cpp.o\u001b[0m\n",
            "[ 69%] \u001b[32mBuilding CXX object src/CMakeFiles/bustools_core.dir/bustools_sort.cpp.o\u001b[0m\n",
            "[ 73%] \u001b[32mBuilding CXX object src/CMakeFiles/bustools_core.dir/bustools_text.cpp.o\u001b[0m\n",
            "[ 76%] \u001b[32mBuilding CXX object src/CMakeFiles/bustools_core.dir/bustools_umicorrect.cpp.o\u001b[0m\n",
            "[ 80%] \u001b[32mBuilding CXX object src/CMakeFiles/bustools_core.dir/bustools_whitelist.cpp.o\u001b[0m\n",
            "[ 84%] \u001b[32mBuilding CXX object src/CMakeFiles/bustools_core.dir/hash.cpp.o\u001b[0m\n",
            "[ 88%] \u001b[32mBuilding C object src/CMakeFiles/bustools_core.dir/roaring.c.o\u001b[0m\n",
            "[ 92%] \u001b[32m\u001b[1mLinking CXX static library libbustools_core.a\u001b[0m\n",
            "[ 92%] Built target bustools_core\n",
            "[ 96%] \u001b[32mBuilding CXX object src/CMakeFiles/bustools.dir/bustools_main.cpp.o\u001b[0m\n",
            "[100%] \u001b[32m\u001b[1mLinking CXX executable bustools\u001b[0m\n",
            "[100%] Built target bustools\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# Define paths to kallisto and bustools binaries\n",
        "kallisto = \"/content/kallisto/build/src/kallisto\"\n",
        "bustools = \"/content/bustools/build/src/bustools\""
      ],
      "metadata": {
        "id": "HqIvB8VYZwtJ"
      },
      "id": "HqIvB8VYZwtJ",
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "d83127b9-3656-4b82-b178-0302a8576413",
      "metadata": {
        "id": "d83127b9-3656-4b82-b178-0302a8576413"
      },
      "outputs": [],
      "source": [
        "# Get PalmDB reference files\n",
        "virus_t2g = \"/content/LSCHWCP_2023/PalmDB/palmdb_clustered_t2g.txt\"\n",
        "virus_fasta = \"/content/LSCHWCP_2023/PalmDB/palmdb_rdrp_seqs.fa\""
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Generate reference index:"
      ],
      "metadata": {
        "id": "1wVLRW4KofHW"
      },
      "id": "1wVLRW4KofHW"
    },
    {
      "cell_type": "code",
      "source": [
        "# Download dog and macaque transcriptome and genome (defining Ensembl version (110) here for reproducibility)\n",
        "!pip install -q gget\n",
        "!gget ref -w dna,cdna -r 110 -d canis_lupus_familiaris\n",
        "!gget ref -w dna,cdna -r 110 -d macaca_mulatta"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "tbpps9ysp77U",
        "outputId": "5f791808-3584-474a-8859-aab97bc913d5"
      },
      "id": "tbpps9ysp77U",
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m43.1/43.1 MB\u001b[0m \u001b[31m32.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m25.2/25.2 MB\u001b[0m \u001b[31m55.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
            "\u001b[?25hFri Dec  8 18:45:32 2023 INFO Fetching reference information for canis_lupus_familiaris from Ensembl release: 110.\n",
            "{\n",
            "    \"canis_lupus_familiaris\": {\n",
            "        \"genome_dna\": {\n",
            "            \"ftp\": \"http://ftp.ensembl.org/pub/release-110/fasta/canis_lupus_familiaris/dna/Canis_lupus_familiaris.ROS_Cfam_1.0.dna.toplevel.fa.gz\",\n",
            "            \"ensembl_release\": 110,\n",
            "            \"release_date\": \"2023-04-21\",\n",
            "            \"release_time\": \"17:46\",\n",
            "            \"bytes\": \"693M\"\n",
            "        },\n",
            "        \"transcriptome_cdna\": {\n",
            "            \"ftp\": \"http://ftp.ensembl.org/pub/release-110/fasta/canis_lupus_familiaris/cdna/Canis_lupus_familiaris.ROS_Cfam_1.0.cdna.all.fa.gz\",\n",
            "            \"ensembl_release\": 110,\n",
            "            \"release_date\": \"2023-04-22\",\n",
            "            \"release_time\": \"16:19\",\n",
            "            \"bytes\": \"26M\"\n",
            "        }\n",
            "    }\n",
            "}\n",
            "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
            "                                 Dload  Upload   Total   Spent    Left  Speed\n",
            "100  692M  100  692M    0     0   632k      0  0:18:42  0:18:42 --:--:--  631k\n",
            "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
            "                                 Dload  Upload   Total   Spent    Left  Speed\n",
            "100 26.4M  100 26.4M    0     0   622k      0  0:00:43  0:00:43 --:--:--  638k\n",
            "Fri Dec  8 19:05:04 2023 INFO Fetching reference information for macaca_mulatta from Ensembl release: 110.\n",
            "{\n",
            "    \"macaca_mulatta\": {\n",
            "        \"genome_dna\": {\n",
            "            \"ftp\": \"http://ftp.ensembl.org/pub/release-110/fasta/macaca_mulatta/dna/Macaca_mulatta.Mmul_10.dna.toplevel.fa.gz\",\n",
            "            \"ensembl_release\": 110,\n",
            "            \"release_date\": \"2023-04-21\",\n",
            "            \"release_time\": \"17:52\",\n",
            "            \"bytes\": \"832M\"\n",
            "        },\n",
            "        \"transcriptome_cdna\": {\n",
            "            \"ftp\": \"http://ftp.ensembl.org/pub/release-110/fasta/macaca_mulatta/cdna/Macaca_mulatta.Mmul_10.cdna.all.fa.gz\",\n",
            "            \"ensembl_release\": 110,\n",
            "            \"release_date\": \"2023-04-22\",\n",
            "            \"release_time\": \"18:12\",\n",
            "            \"bytes\": \"30M\"\n",
            "        }\n",
            "    }\n",
            "}\n",
            "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
            "                                 Dload  Upload   Total   Spent    Left  Speed\n",
            "100  832M  100  832M    0     0   635k      0  0:22:20  0:22:20 --:--:--  637k\n",
            "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
            "                                 Dload  Upload   Total   Spent    Left  Speed\n",
            "100 29.7M  100 29.7M    0     0   623k      0  0:00:48  0:00:48 --:--:--  638k\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# Concatenate transcriptomes and genomes into one file\n",
        "canine_macaque_fasta = \"macaque_dog.cdna_dna.fa.gz\"\n",
        "!cat \\\n",
        "  Macaca_mulatta.Mmul_10.cdna.all.fa.gz \\\n",
        "  Macaca_mulatta.Mmul_10.dna.toplevel.fa.gz \\\n",
        "  Canis_lupus_familiaris.ROS_Cfam_1.0.cdna.all.fa.gz \\\n",
        "  Canis_lupus_familiaris.ROS_Cfam_1.0.dna.toplevel.fa.gz \\\n",
        "  > $canine_macaque_fasta"
      ],
      "metadata": {
        "id": "YMTSxQUZp5wg"
      },
      "id": "YMTSxQUZp5wg",
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "%%time\n",
        "# Generate PalmDB reference index palmdb_rdrp_seqs with macaque and dog (MDCK)\n",
        "virus_index = \"palmdb_macaque_dog_dlist_cdna_dna.idx\"\n",
        "\n",
        "!$kallisto index \\\n",
        "    -t $threads \\\n",
        "    --aa \\\n",
        "    --d-list=$canine_macaque_fasta \\\n",
        "    -i $virus_index \\\n",
        "    $virus_fasta"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "_xDX93ApoEVi",
        "outputId": "f7bbf63d-8fe3-4287-e08f-3a87514e4973"
      },
      "id": "_xDX93ApoEVi",
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "[index] --d-list-overhang was set to 3 (with --aa, the d-list overhang must be >= 3)\n",
            "[build] loading fasta file /content/LSCHWCP_2023/PalmDB/palmdb_rdrp_seqs.fa\n",
            "[build] k-mer length: 31\n",
            "KmerStream::KmerStream(): Start computing k-mer cardinality estimations (1/2)\n",
            "KmerStream::KmerStream(): Start computing k-mer cardinality estimations (1/2)\n",
            "KmerStream::KmerStream(): Finished\n",
            "CompactedDBG::build(): Estimated number of k-mers occurring at least once: 37641510\n",
            "CompactedDBG::build(): Estimated number of minimizer occurring at least once: 7877811\n",
            "CompactedDBG::filter(): Processed 87630084 k-mers in 296561 reads\n",
            "CompactedDBG::filter(): Found 37508965 unique k-mers\n",
            "CompactedDBG::filter(): Number of blocks in Bloom filter is 257317\n",
            "CompactedDBG::construct(): Extract approximate unitigs (1/2)\n",
            "CompactedDBG::construct(): Extract approximate unitigs (2/2)\n",
            "CompactedDBG::construct(): Closed all input files\n",
            "\n",
            "CompactedDBG::construct(): Splitting unitigs (1/2)\n",
            "\n",
            "CompactedDBG::construct(): Splitting unitigs (2/2)\n",
            "CompactedDBG::construct(): Before split: 2040576 unitigs\n",
            "CompactedDBG::construct(): After split (1/1): 2040576 unitigs\n",
            "CompactedDBG::construct(): Unitigs split: 1560\n",
            "CompactedDBG::construct(): Unitigs deleted: 0\n",
            "\n",
            "CompactedDBG::construct(): Joining unitigs\n",
            "CompactedDBG::construct(): After join: 2021701 unitigs\n",
            "CompactedDBG::construct(): Joined 19230 unitigs\n",
            "[build] extracting D-list k-mers from \"macaque_dog.cdna_dna.fa.gz\"\n",
            "[build] identified 90815 distinguishing flanking k-mers\n",
            "[build] building MPHF\n",
            "[build] creating equivalence classes ... \n",
            "[build] target de Bruijn graph has k-mer length 31 and minimizer length 23\n",
            "[build] target de Bruijn graph has 2021703 contigs and contains 37541757 k-mers \n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Align with kallisto translated search:"
      ],
      "metadata": {
        "id": "sPH7l6Brog6E"
      },
      "id": "sPH7l6Brog6E"
    },
    {
      "cell_type": "code",
      "source": [
        "# Define folder to save kallisto alignments\n",
        "kallisto_out = \"kallisto_alignment/translated\"\n",
        "!mkdir -p $kallisto_out"
      ],
      "metadata": {
        "id": "iQm1A-u5Z0sY"
      },
      "id": "iQm1A-u5Z0sY",
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "8b55300e-7524-494b-bc96-0f273362009d",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "8b55300e-7524-494b-bc96-0f273362009d",
        "outputId": "9f1c5762-57f0-4fe8-99e7-832a640f29c5"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/SRR12698539_2_extracted_u10.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 676 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 676 BUS records\n",
            "reading time 1.2e-05s\n",
            "sorting time 6.7e-05s\n",
            "writing time 7.1e-05s\n"
          ]
        }
      ],
      "source": [
        "# Align unmutated sequences\n",
        "mut_file = orig_file\n",
        "mut_name = \"0_ebov_snp_0_ms\"\n",
        "\n",
        "!mkdir -p $kallisto_out/$mut_name\n",
        "\n",
        "!$kallisto bus \\\n",
        "    -i $virus_index \\\n",
        "    -o $kallisto_out/$mut_name \\\n",
        "    --aa \\\n",
        "    -x bulk \\\n",
        "    -t $threads \\\n",
        "    $mut_file\n",
        "\n",
        "!$bustools sort -o $kallisto_out/$mut_name/output_sorted.bus $kallisto_out/$mut_name/output.bus\n",
        "\n",
        "!$bustools count \\\n",
        "    --genecounts \\\n",
        "    --cm \\\n",
        "    -o $kallisto_out/$mut_name/bustools_count/ \\\n",
        "    -g $virus_t2g \\\n",
        "    -e $kallisto_out/$mut_name/matrix.ec \\\n",
        "    -t $kallisto_out/$mut_name/transcripts.txt \\\n",
        "    $kallisto_out/$mut_name/output_sorted.bus"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "for mut_file in glob.glob(f\"{out}/*_ms.fa\"):\n",
        "    mut_name = mut_file.split(\"/\")[-1].split(\".fa\")[0]\n",
        "\n",
        "    !mkdir -p $kallisto_out/$mut_name\n",
        "\n",
        "    !$kallisto bus \\\n",
        "        -i $virus_index \\\n",
        "        -o $kallisto_out/$mut_name \\\n",
        "        --aa \\\n",
        "        -x bulk \\\n",
        "        -t $threads \\\n",
        "        $mut_file\n",
        "\n",
        "    !$bustools sort -o $kallisto_out/$mut_name/output_sorted.bus $kallisto_out/$mut_name/output.bus\n",
        "\n",
        "    !$bustools count \\\n",
        "        --genecounts \\\n",
        "        --cm \\\n",
        "        -o $kallisto_out/$mut_name/bustools_count/ \\\n",
        "        -g $virus_t2g \\\n",
        "        -e $kallisto_out/$mut_name/matrix.ec \\\n",
        "        -t $kallisto_out/$mut_name/transcripts.txt \\\n",
        "        $kallisto_out/$mut_name/output_sorted.bus"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "lqACsV1XfjJV",
        "outputId": "1ea5f23a-8cae-4943-c7c8-8fb75049e622"
      },
      "id": "lqACsV1XfjJV",
      "execution_count": null,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_05_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 490 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 490 BUS records\n",
            "reading time 8e-06s\n",
            "sorting time 3.1e-05s\n",
            "writing time 6.4e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_25_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 10 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 10 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 7e-06s\n",
            "writing time 4e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_25_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 18 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 18 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 7e-06s\n",
            "writing time 3.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_24_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 11 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 11 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 6e-06s\n",
            "writing time 3.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_02_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 621 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 621 BUS records\n",
            "reading time 9e-06s\n",
            "sorting time 3.8e-05s\n",
            "writing time 0.000126s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_21_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 32 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 32 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 3e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_28_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 3 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 3 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_19_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 45 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 45 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 9e-06s\n",
            "writing time 0.000333s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_29_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 6 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 6 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_19_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 45 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 45 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 3e-06s\n",
            "writing time 5.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_26_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 8 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 8 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 3.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_24_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 11 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 11 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_07_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 362 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 362 BUS records\n",
            "reading time 7e-06s\n",
            "sorting time 2.3e-05s\n",
            "writing time 6.4e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_05_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 501 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 501 BUS records\n",
            "reading time 7e-06s\n",
            "sorting time 3.2e-05s\n",
            "writing time 7.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_07_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 362 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 362 BUS records\n",
            "reading time 7e-06s\n",
            "sorting time 3e-05s\n",
            "writing time 9.3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_2_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 38 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 38 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 3e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_1_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 271 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 271 BUS records\n",
            "reading time 6e-06s\n",
            "sorting time 2.1e-05s\n",
            "writing time 5.7e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_23_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 17 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 17 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 2.2e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_14_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 110 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 110 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 7e-06s\n",
            "writing time 0.000264s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_12_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 189 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 189 BUS records\n",
            "reading time 6e-06s\n",
            "sorting time 1.7e-05s\n",
            "writing time 6e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_06_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 414 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 414 BUS records\n",
            "reading time 1e-05s\n",
            "sorting time 2.7e-05s\n",
            "writing time 0.000123s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_17_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 81 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 81 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1e-05s\n",
            "writing time 5.8e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_04_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 531 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 531 BUS records\n",
            "reading time 8e-06s\n",
            "sorting time 3.5e-05s\n",
            "writing time 6.8e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_22_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 21 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 21 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 3.6e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_28_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 5 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 5 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 4.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_04_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 535 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 535 BUS records\n",
            "reading time 9e-06s\n",
            "sorting time 3.4e-05s\n",
            "writing time 7.5e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_25_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 10 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 10 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 3.4e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_1_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 271 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 271 BUS records\n",
            "reading time 6e-06s\n",
            "sorting time 1.7e-05s\n",
            "writing time 0.000388s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_27_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 12 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 12 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 8e-06s\n",
            "writing time 3.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_06_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 422 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 422 BUS records\n",
            "reading time 7e-06s\n",
            "sorting time 3.3e-05s\n",
            "writing time 6.7e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_15_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 96 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 96 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1.1e-05s\n",
            "writing time 0.000221s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_27_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 15 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 15 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 8e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_3_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 7 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 7 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_19_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 49 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 49 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 4e-06s\n",
            "writing time 8e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_1_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 263 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 263 BUS records\n",
            "reading time 6e-06s\n",
            "sorting time 1.6e-05s\n",
            "writing time 5.8e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_06_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 447 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 447 BUS records\n",
            "reading time 7e-06s\n",
            "sorting time 2.9e-05s\n",
            "writing time 7e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_08_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 336 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 336 BUS records\n",
            "reading time 6e-06s\n",
            "sorting time 2.1e-05s\n",
            "writing time 7.6e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_04_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 529 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 529 BUS records\n",
            "reading time 8e-06s\n",
            "sorting time 3.3e-05s\n",
            "writing time 0.000282s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_08_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 306 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 306 BUS records\n",
            "reading time 7e-06s\n",
            "sorting time 1.9e-05s\n",
            "writing time 6.8e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_15_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 87 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 87 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1.2e-05s\n",
            "writing time 0.000243s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_26_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 12 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 12 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_22_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 22 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 22 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 4.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_05_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 498 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 498 BUS records\n",
            "reading time 9e-06s\n",
            "sorting time 3.8e-05s\n",
            "writing time 7.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_26_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 9 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 9 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 2.4e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_24_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 16 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 16 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_15_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 101 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 101 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 6e-06s\n",
            "writing time 0.000343s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_28_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 7 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 7 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_03_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 548 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 548 BUS records\n",
            "reading time 9e-06s\n",
            "sorting time 4.2e-05s\n",
            "writing time 6.4e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_08_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 318 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 318 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 1.8e-05s\n",
            "writing time 7.5e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_21_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 26 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 26 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_18_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 67 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 67 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1e-05s\n",
            "writing time 0.000353s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_28_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 5 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 5 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 2.5e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_24_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 13 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 13 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 7e-06s\n",
            "writing time 4.6e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_19_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 43 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 43 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 8e-06s\n",
            "writing time 2.7e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_06_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 442 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 442 BUS records\n",
            "reading time 9e-06s\n",
            "sorting time 2.8e-05s\n",
            "writing time 0.000257s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_16_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 77 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 77 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 6e-06s\n",
            "writing time 5.5e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_18_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 61 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 61 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1e-05s\n",
            "writing time 7.7e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_29_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 8 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 8 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 4.5e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_2_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 30 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 30 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 3e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_17_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 79 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 79 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 4e-06s\n",
            "writing time 4.6e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_07_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 351 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 351 BUS records\n",
            "reading time 6e-06s\n",
            "sorting time 2.2e-05s\n",
            "writing time 6.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_17_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 64 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 64 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 5e-06s\n",
            "writing time 4.7e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_11_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 233 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 233 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1.5e-05s\n",
            "writing time 0.000226s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_19_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 37 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 37 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 3e-06s\n",
            "writing time 3.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_3_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 7 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 7 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 2.3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_14_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 113 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 113 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 7e-06s\n",
            "writing time 5.2e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_16_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 68 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 68 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 5e-06s\n",
            "writing time 5.5e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_2_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 32 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 32 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 8e-06s\n",
            "writing time 4.6e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_3_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 3 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 3 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_11_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 238 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 238 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 1.5e-05s\n",
            "writing time 0.000352s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_08_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 335 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 335 BUS records\n",
            "reading time 7e-06s\n",
            "sorting time 2.1e-05s\n",
            "writing time 0.000256s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_27_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 13 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 13 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 3.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_19_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 54 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 54 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1e-05s\n",
            "writing time 5.6e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_14_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 110 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 110 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 7e-06s\n",
            "writing time 5.3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_08_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 310 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 310 BUS records\n",
            "reading time 7e-06s\n",
            "sorting time 2e-05s\n",
            "writing time 0.000238s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_25_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 11 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 11 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 2.4e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_22_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 24 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 24 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1.8e-05s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_03_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 564 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 564 BUS records\n",
            "reading time 1e-05s\n",
            "sorting time 3.6e-05s\n",
            "writing time 7.5e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_23_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 19 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 19 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_27_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 8 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 8 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_28_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 6 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 6 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 2.8e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_12_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 179 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 179 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1.1e-05s\n",
            "writing time 7e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_21_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 28 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 28 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 8e-06s\n",
            "writing time 3.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_13_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 174 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 174 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1.7e-05s\n",
            "writing time 0.000357s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_02_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 619 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 619 BUS records\n",
            "reading time 1.7e-05s\n",
            "sorting time 3.9e-05s\n",
            "writing time 0.000201s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_28_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 9 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 9 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 3.2e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_09_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 324 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 324 BUS records\n",
            "reading time 6e-06s\n",
            "sorting time 2e-05s\n",
            "writing time 6.3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_19_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 41 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 41 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 3e-06s\n",
            "writing time 0.000226s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_15_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 116 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 116 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 7e-06s\n",
            "writing time 6.6e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_11_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 217 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 217 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 1.3e-05s\n",
            "writing time 5.4e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_24_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 20 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 20 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_13_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 146 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 146 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 9e-06s\n",
            "writing time 5.6e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_1_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 262 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 262 BUS records\n",
            "reading time 7e-06s\n",
            "sorting time 1.7e-05s\n",
            "writing time 0.000192s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_1_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 254 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 254 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 1.6e-05s\n",
            "writing time 5.4e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_17_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 79 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 79 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1e-05s\n",
            "writing time 6e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_2_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 36 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 36 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 3e-06s\n",
            "writing time 2.8e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_26_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 8 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 8 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 3.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_16_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 84 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 84 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 5e-06s\n",
            "writing time 6.2e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_14_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 115 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 115 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 7e-06s\n",
            "writing time 0.000187s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_13_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 150 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 150 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1.4e-05s\n",
            "writing time 0.000315s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_19_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 52 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 52 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 3e-06s\n",
            "writing time 5.6e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_12_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 195 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 195 BUS records\n",
            "reading time 6e-06s\n",
            "sorting time 1.7e-05s\n",
            "writing time 5.7e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_16_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 78 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 78 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 5e-06s\n",
            "writing time 0.000157s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_11_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 207 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 207 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1.9e-05s\n",
            "writing time 0.000189s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_07_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 370 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 370 BUS records\n",
            "reading time 7e-06s\n",
            "sorting time 2.5e-05s\n",
            "writing time 6.5e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_15_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 117 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 117 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 8e-06s\n",
            "writing time 0.000231s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_02_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 608 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 608 BUS records\n",
            "reading time 1e-05s\n",
            "sorting time 3.8e-05s\n",
            "writing time 7.2e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_02_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 623 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 623 BUS records\n",
            "reading time 9e-06s\n",
            "sorting time 3.8e-05s\n",
            "writing time 0.000208s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_3_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 2 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 2 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_09_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 330 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 330 BUS records\n",
            "reading time 7e-06s\n",
            "sorting time 2e-05s\n",
            "writing time 0.000451s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_03_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 569 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 569 BUS records\n",
            "reading time 8e-06s\n",
            "sorting time 3.6e-05s\n",
            "writing time 6.2e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_26_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 10 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 10 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 3.5e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_18_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 53 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 53 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 9e-06s\n",
            "writing time 5.4e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_22_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 23 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 23 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_13_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 140 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 140 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 9e-06s\n",
            "writing time 5.5e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_21_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 35 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 35 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 3e-06s\n",
            "writing time 2.8e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_21_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 32 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 32 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 3e-06s\n",
            "writing time 3.2e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_16_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 72 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 72 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 1e-05s\n",
            "writing time 9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_21_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 32 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 32 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_1_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 258 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 258 BUS records\n",
            "reading time 7e-06s\n",
            "sorting time 2.2e-05s\n",
            "writing time 0.000297s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_04_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 528 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 528 BUS records\n",
            "reading time 9e-06s\n",
            "sorting time 3.4e-05s\n",
            "writing time 5.7e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_2_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 41 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 41 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 4e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_22_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 26 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 26 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 7e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_27_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 10 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 10 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_07_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 373 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 373 BUS records\n",
            "reading time 7e-06s\n",
            "sorting time 2.5e-05s\n",
            "writing time 9.6e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_12_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 177 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 177 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 1.2e-05s\n",
            "writing time 0.000406s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_23_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 30 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 30 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 8e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_15_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 100 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 100 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 7e-06s\n",
            "writing time 0.000116s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_2_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 38 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 38 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 3e-06s\n",
            "writing time 3.2e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_25_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 6 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 6 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 3.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_18_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 82 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 82 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1.1e-05s\n",
            "writing time 5.3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_3_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 6 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 6 BUS records\n",
            "reading time 1.1e-05s\n",
            "sorting time 2e-06s\n",
            "writing time 5.3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_23_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 19 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 19 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 2.4e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_26_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 14 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 14 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 3.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_26_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 15 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 15 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 7e-06s\n",
            "writing time 5.5e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_3_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 8 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 8 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_22_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 19 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 19 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_29_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 4 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 4 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 2.8e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_07_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 348 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 348 BUS records\n",
            "reading time 7e-06s\n",
            "sorting time 2.2e-05s\n",
            "writing time 0.000333s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_11_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 239 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 239 BUS records\n",
            "reading time 6e-06s\n",
            "sorting time 1.5e-05s\n",
            "writing time 6.4e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_13_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 162 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 162 BUS records\n",
            "reading time 6e-06s\n",
            "sorting time 1.6e-05s\n",
            "writing time 0.000369s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_22_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 24 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 24 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_16_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 87 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 87 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 1.1e-05s\n",
            "writing time 6.4e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_06_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 432 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 432 BUS records\n",
            "reading time 8e-06s\n",
            "sorting time 3.4e-05s\n",
            "writing time 6e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_14_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 117 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 117 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1.3e-05s\n",
            "writing time 0.00033s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_18_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 62 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 62 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 4e-06s\n",
            "writing time 9.7e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_2_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 42 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 42 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 3e-06s\n",
            "writing time 6.5e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_09_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 317 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 317 BUS records\n",
            "reading time 6e-06s\n",
            "sorting time 2e-05s\n",
            "writing time 6.2e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_1_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 255 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 255 BUS records\n",
            "reading time 6e-06s\n",
            "sorting time 1.6e-05s\n",
            "writing time 9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_26_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 13 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 13 BUS records\n",
            "reading time 6e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_18_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 72 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 72 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 5e-06s\n",
            "writing time 0.000325s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_14_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 126 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 126 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 8e-06s\n",
            "writing time 5.3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_02_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 597 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 597 BUS records\n",
            "reading time 8e-06s\n",
            "sorting time 4.2e-05s\n",
            "writing time 0.000237s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_12_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 177 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 177 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1.6e-05s\n",
            "writing time 0.000298s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_18_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 69 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 69 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 4e-06s\n",
            "writing time 0.000195s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_13_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 150 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 150 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 1.5e-05s\n",
            "writing time 5.2e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_15_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 80 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 80 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1.1e-05s\n",
            "writing time 5.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_3_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 5 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 5 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 4.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_12_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 195 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 195 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 3.4e-05s\n",
            "writing time 9.4e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_11_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 212 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 212 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 1.3e-05s\n",
            "writing time 9.2e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_02_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 613 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 613 BUS records\n",
            "reading time 9e-06s\n",
            "sorting time 4.5e-05s\n",
            "writing time 7.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_06_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 442 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 442 BUS records\n",
            "reading time 8e-06s\n",
            "sorting time 4.2e-05s\n",
            "writing time 7.5e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_28_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 9 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 9 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 4.8e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_23_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 23 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 23 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 3e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_3_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 7 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 7 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 7e-06s\n",
            "writing time 2.4e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_27_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 6 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 6 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 6e-06s\n",
            "writing time 2.2e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_28_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 9 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 9 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 8e-06s\n",
            "writing time 5e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_09_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 294 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 294 BUS records\n",
            "reading time 7e-06s\n",
            "sorting time 2.4e-05s\n",
            "writing time 0.000102s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_11_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 227 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 227 BUS records\n",
            "reading time 6e-06s\n",
            "sorting time 1.4e-05s\n",
            "writing time 0.000109s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_24_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 12 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 12 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_29_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 7 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 7 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 5.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_21_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 48 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 48 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 4e-06s\n",
            "writing time 0.000331s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_17_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 84 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 84 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 6e-06s\n",
            "writing time 0.000222s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_09_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 316 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 316 BUS records\n",
            "reading time 7e-06s\n",
            "sorting time 2e-05s\n",
            "writing time 0.000378s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_03_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 571 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 571 BUS records\n",
            "reading time 9e-06s\n",
            "sorting time 3.6e-05s\n",
            "writing time 0.000142s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_03_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 568 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 568 BUS records\n",
            "reading time 9e-06s\n",
            "sorting time 3.6e-05s\n",
            "writing time 0.000123s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_26_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 18 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 18 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 3.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_03_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 570 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 570 BUS records\n",
            "reading time 8e-06s\n",
            "sorting time 3.6e-05s\n",
            "writing time 0.000226s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_07_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 385 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 385 BUS records\n",
            "reading time 6e-06s\n",
            "sorting time 2.3e-05s\n",
            "writing time 0.000206s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_24_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 9 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 9 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 3.4e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_16_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 78 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 78 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 5e-06s\n",
            "writing time 0.000242s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_08_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 315 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 315 BUS records\n",
            "reading time 6e-06s\n",
            "sorting time 2.6e-05s\n",
            "writing time 0.000289s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_3_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 6 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 6 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 3.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_3_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 7 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 7 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 4.5e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_05_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 464 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 464 BUS records\n",
            "reading time 9e-06s\n",
            "sorting time 3.1e-05s\n",
            "writing time 0.000533s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_09_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 318 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 318 BUS records\n",
            "reading time 7e-06s\n",
            "sorting time 2e-05s\n",
            "writing time 0.000223s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_02_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 611 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 611 BUS records\n",
            "reading time 9e-06s\n",
            "sorting time 3.7e-05s\n",
            "writing time 7.4e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_22_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 20 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 20 BUS records\n",
            "reading time 6e-06s\n",
            "sorting time 3e-06s\n",
            "writing time 3.5e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_08_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 338 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 338 BUS records\n",
            "reading time 6e-06s\n",
            "sorting time 2.7e-05s\n",
            "writing time 5.6e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_06_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 412 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 412 BUS records\n",
            "reading time 8e-06s\n",
            "sorting time 2.7e-05s\n",
            "writing time 0.000184s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_17_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 75 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 75 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 5e-06s\n",
            "writing time 0.000198s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_13_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 156 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 156 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 9e-06s\n",
            "writing time 6.4e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_13_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 142 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 142 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 8e-06s\n",
            "writing time 5.4e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_15_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 101 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 101 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 6e-06s\n",
            "writing time 6.3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_22_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 29 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 29 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 3e-06s\n",
            "writing time 3.5e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_27_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 12 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 12 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_04_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 531 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 531 BUS records\n",
            "reading time 9e-06s\n",
            "sorting time 3.4e-05s\n",
            "writing time 0.000322s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_15_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 104 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 104 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1.2e-05s\n",
            "writing time 6.8e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_04_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 543 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 543 BUS records\n",
            "reading time 9e-06s\n",
            "sorting time 4e-05s\n",
            "writing time 6.3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_14_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 127 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 127 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 8e-06s\n",
            "writing time 5.4e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_06_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 408 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 408 BUS records\n",
            "reading time 8e-06s\n",
            "sorting time 2.7e-05s\n",
            "writing time 6.2e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_06_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 427 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 427 BUS records\n",
            "reading time 8e-06s\n",
            "sorting time 2.9e-05s\n",
            "writing time 0.000201s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_03_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 568 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 568 BUS records\n",
            "reading time 8e-06s\n",
            "sorting time 3.6e-05s\n",
            "writing time 7.6e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_25_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 12 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 12 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_11_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 217 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 217 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 1.4e-05s\n",
            "writing time 9.4e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_24_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 18 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 18 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 7e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_05_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 483 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 483 BUS records\n",
            "reading time 7e-06s\n",
            "sorting time 3.1e-05s\n",
            "writing time 0.000191s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_03_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 584 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 584 BUS records\n",
            "reading time 9e-06s\n",
            "sorting time 3.7e-05s\n",
            "writing time 6.6e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_16_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 84 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 84 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1.1e-05s\n",
            "writing time 5.5e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_24_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 11 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 11 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_24_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 14 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 14 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_29_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 7 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 7 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 2.8e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_18_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 75 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 75 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 5e-06s\n",
            "writing time 5.8e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_1_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 265 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 265 BUS records\n",
            "reading time 6e-06s\n",
            "sorting time 1.7e-05s\n",
            "writing time 5.5e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_27_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 13 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 13 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 3.6e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_08_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 308 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 308 BUS records\n",
            "reading time 6e-06s\n",
            "sorting time 3.9e-05s\n",
            "writing time 6.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_19_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 44 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 44 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 9e-06s\n",
            "writing time 0.00023s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_04_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 520 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 520 BUS records\n",
            "reading time 8e-06s\n",
            "sorting time 4.3e-05s\n",
            "writing time 6.5e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_14_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 125 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 125 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 8e-06s\n",
            "writing time 5.5e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_25_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 14 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 14 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 8e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_26_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 13 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 13 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 2.8e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_23_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 21 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 21 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 4.5e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_25_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 12 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 12 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_13_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 154 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 154 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1e-05s\n",
            "writing time 4.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_05_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 480 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 480 BUS records\n",
            "reading time 8e-06s\n",
            "sorting time 3.1e-05s\n",
            "writing time 0.000197s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_29_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 4 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 4 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_17_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 85 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 85 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1e-05s\n",
            "writing time 0.000302s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_09_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 319 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 319 BUS records\n",
            "reading time 7e-06s\n",
            "sorting time 2.1e-05s\n",
            "writing time 5.8e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_21_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 26 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 26 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_09_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 306 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 306 BUS records\n",
            "reading time 8e-06s\n",
            "sorting time 2.4e-05s\n",
            "writing time 0.000147s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_06_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 449 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 449 BUS records\n",
            "reading time 8e-06s\n",
            "sorting time 3.5e-05s\n",
            "writing time 0.000288s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_05_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 475 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 475 BUS records\n",
            "reading time 8e-06s\n",
            "sorting time 3.1e-05s\n",
            "writing time 9.7e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_27_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 10 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 10 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 3.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_23_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 18 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 18 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 4.5e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_03_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 564 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 564 BUS records\n",
            "reading time 8e-06s\n",
            "sorting time 4.1e-05s\n",
            "writing time 6.5e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_09_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 335 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 335 BUS records\n",
            "reading time 6e-06s\n",
            "sorting time 2.1e-05s\n",
            "writing time 0.000104s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_29_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 3 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 3 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 7e-06s\n",
            "writing time 3.3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_19_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 43 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 43 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 3e-06s\n",
            "writing time 6.3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_18_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 71 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 71 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1e-05s\n",
            "writing time 5.3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_11_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 208 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 208 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 1.3e-05s\n",
            "writing time 5.6e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_29_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 4 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 4 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 2.8e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_1_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 273 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 273 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 1.8e-05s\n",
            "writing time 5.5e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_03_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 570 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 570 BUS records\n",
            "reading time 9e-06s\n",
            "sorting time 4.4e-05s\n",
            "writing time 0.000425s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_21_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 29 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 29 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 8e-06s\n",
            "writing time 3.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_1_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 283 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 283 BUS records\n",
            "reading time 6e-06s\n",
            "sorting time 2.4e-05s\n",
            "writing time 5.7e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_23_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 23 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 23 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 9e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_23_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 17 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 17 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_07_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 342 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 342 BUS records\n",
            "reading time 6e-06s\n",
            "sorting time 2.8e-05s\n",
            "writing time 0.000123s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_2_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 23 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 23 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 6.2e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_28_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 7 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 7 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 5.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_17_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 73 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 73 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1e-05s\n",
            "writing time 0.000213s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_15_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 96 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 96 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 7e-06s\n",
            "writing time 5.5e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_13_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 161 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 161 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1e-05s\n",
            "writing time 5.3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_02_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 620 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 620 BUS records\n",
            "reading time 9e-06s\n",
            "sorting time 3.8e-05s\n",
            "writing time 6.5e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_29_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 9 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 9 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_23_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 9 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 9 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 2.3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_14_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 123 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 123 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 8e-06s\n",
            "writing time 5.6e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_02_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 622 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 622 BUS records\n",
            "reading time 9e-06s\n",
            "sorting time 3.9e-05s\n",
            "writing time 0.000307s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_25_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 15 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 15 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2.3e-05s\n",
            "writing time 6.2e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_17_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 76 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 76 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1.2e-05s\n",
            "writing time 0.000312s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_12_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 192 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 192 BUS records\n",
            "reading time 6e-06s\n",
            "sorting time 1.2e-05s\n",
            "writing time 6.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_05_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 466 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 466 BUS records\n",
            "reading time 8e-06s\n",
            "sorting time 3e-05s\n",
            "writing time 0.000399s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_04_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 522 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 522 BUS records\n",
            "reading time 8e-06s\n",
            "sorting time 3.3e-05s\n",
            "writing time 0.000201s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_28_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 11 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 11 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 3.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_07_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 353 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 353 BUS records\n",
            "reading time 8e-06s\n",
            "sorting time 2.2e-05s\n",
            "writing time 6.8e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_27_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 11 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 11 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 7e-06s\n",
            "writing time 3.4e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_21_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 26 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 26 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 3e-06s\n",
            "writing time 5.2e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_18_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 58 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 58 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 9e-06s\n",
            "writing time 0.000302s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_17_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 91 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 91 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 6e-06s\n",
            "writing time 6.8e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_08_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 341 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 341 BUS records\n",
            "reading time 7e-06s\n",
            "sorting time 2.7e-05s\n",
            "writing time 7.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_12_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 204 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 204 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1.2e-05s\n",
            "writing time 0.000415s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_04_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 534 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 534 BUS records\n",
            "reading time 1.6e-05s\n",
            "sorting time 3.4e-05s\n",
            "writing time 0.000137s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_25_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 8 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 8 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 6e-06s\n",
            "writing time 4.7e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_2_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 33 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 33 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 3e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_11_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 235 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 235 BUS records\n",
            "reading time 6e-06s\n",
            "sorting time 1.5e-05s\n",
            "writing time 0.000206s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_14_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 123 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 123 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 1.3e-05s\n",
            "writing time 5.3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_05_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 486 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 486 BUS records\n",
            "reading time 1e-05s\n",
            "sorting time 3.9e-05s\n",
            "writing time 0.000143s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_09_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 317 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 317 BUS records\n",
            "reading time 8e-06s\n",
            "sorting time 2e-05s\n",
            "writing time 0.000135s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_16_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 73 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 73 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 1.1e-05s\n",
            "writing time 0.000353s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_12_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 185 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 185 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1.1e-05s\n",
            "writing time 7e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_22_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 25 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 25 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 3.3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_04_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 541 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 541 BUS records\n",
            "reading time 1e-05s\n",
            "sorting time 3.4e-05s\n",
            "writing time 7.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_2_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 54 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 54 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 4e-06s\n",
            "writing time 0.000342s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_12_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 186 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 186 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 1.2e-05s\n",
            "writing time 0.000201s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_02_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 621 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 621 BUS records\n",
            "reading time 1e-05s\n",
            "sorting time 3.8e-05s\n",
            "writing time 0.000138s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_05_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 471 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 471 BUS records\n",
            "reading time 8e-06s\n",
            "sorting time 3.4e-05s\n",
            "writing time 0.000124s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_07_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 357 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 357 BUS records\n",
            "reading time 6e-06s\n",
            "sorting time 3e-05s\n",
            "writing time 6.7e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_29_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 9 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 9 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 7e-06s\n",
            "writing time 3.7e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_08_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 344 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 344 BUS records\n",
            "reading time 6e-06s\n",
            "sorting time 2.2e-05s\n",
            "writing time 0.000319s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 296,561\n",
            "[index] number of k-mers: 37,541,757\n",
            "[index] number of D-list k-mers: 90,815\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_16_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 66 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 66 BUS records\n",
            "reading time 1e-05s\n",
            "sorting time 5e-06s\n",
            "writing time 0.000395s\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "id": "5a9efe25-a9a8-402b-872c-3fb896fa4527",
      "metadata": {
        "id": "5a9efe25-a9a8-402b-872c-3fb896fa4527"
      },
      "source": [
        "# Align using kallisto bustools (standard workflow)"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "The notebook showing how the ZEBOV reference files 'ebov_GCA_000848505.idx' and 'ebov_GCA_000848505_t2g.txt' were generated can be found [here](https://github.com/pachterlab/LSCHWCP_2023/blob/main/Notebooks/ebola_ref/generate_ebov_ref.ipynb)."
      ],
      "metadata": {
        "id": "xdEpr6RATQ5x"
      },
      "id": "xdEpr6RATQ5x"
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "24a7b97e-1772-4b77-b9db-9690f8eefc1b",
      "metadata": {
        "id": "24a7b97e-1772-4b77-b9db-9690f8eefc1b"
      },
      "outputs": [],
      "source": [
        "kallisto_out = f\"kallisto_alignment/standard\"\n",
        "!mkdir -p $kallisto_out\n",
        "\n",
        "# Ebola reference index generated using the standard kb ref workflow\n",
        "# and the GCA_000848505 Zaire ebolavirus genome assembly\n",
        "ebola_t2g = \"/content/LSCHWCP_2023/Notebooks/ebola_ref/ebov_GCA_000848505_t2g.txt\"\n",
        "ebola_index = \"/content/LSCHWCP_2023/Notebooks/ebola_ref/ebov_GCA_000848505.idx\""
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "74836649-46ca-49d9-b17e-a20e3ffcd8d4",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "74836649-46ca-49d9-b17e-a20e3ffcd8d4",
        "outputId": "2ef20f71-1863-413b-a1b7-b2706fce58ea"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/SRR12698539_2_extracted_u10.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 662 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 662 BUS records\n",
            "reading time 1e-05s\n",
            "sorting time 1.8e-05s\n",
            "writing time 2.9e-05s\n"
          ]
        }
      ],
      "source": [
        "mut_file = orig_file\n",
        "mut_name = \"0_ebov_snp_0_ms\"\n",
        "\n",
        "!mkdir -p $kallisto_out/$mut_name\n",
        "\n",
        "!$kallisto bus \\\n",
        "    -i $ebola_index \\\n",
        "    -o $kallisto_out/$mut_name \\\n",
        "    -x bulk \\\n",
        "    -t $threads \\\n",
        "    $mut_file\n",
        "\n",
        "!$bustools sort -o $kallisto_out/$mut_name/output_sorted.bus $kallisto_out/$mut_name/output.bus\n",
        "\n",
        "!$bustools count \\\n",
        "    --genecounts \\\n",
        "    --cm \\\n",
        "    -o $kallisto_out/$mut_name/bustools_count/ \\\n",
        "    -g $ebola_t2g \\\n",
        "    -e $kallisto_out/$mut_name/matrix.ec \\\n",
        "    -t $kallisto_out/$mut_name/transcripts.txt \\\n",
        "    $kallisto_out/$mut_name/output_sorted.bus"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "5e71bde7-2519-43bf-9477-e8868af0dddb",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "5e71bde7-2519-43bf-9477-e8868af0dddb",
        "outputId": "e1f31fec-6cbb-433f-8cec-d700f3df6ae8"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_05_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 410 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 410 BUS records\n",
            "reading time 7e-06s\n",
            "sorting time 1e-05s\n",
            "writing time 3.5e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_25_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_25_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_24_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_02_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 638 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 638 BUS records\n",
            "reading time 9e-06s\n",
            "sorting time 2.3e-05s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_21_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 1 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 1 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 6e-06s\n",
            "writing time 3.4e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_28_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_19_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 5 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 5 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_29_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 2e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_19_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 2 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 2 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 6e-06s\n",
            "writing time 3.3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_26_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 2e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_24_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_07_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 217 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 217 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 1.1e-05s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_05_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 398 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 398 BUS records\n",
            "reading time 9e-06s\n",
            "sorting time 1e-05s\n",
            "writing time 3.8e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_07_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 207 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 207 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 5e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_2_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 1 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 1 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 7e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_1_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 106 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 106 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 3e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_23_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_14_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 19 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 19 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_12_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 46 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 46 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 8e-06s\n",
            "writing time 2.8e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_06_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 270 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 270 BUS records\n",
            "reading time 7e-06s\n",
            "sorting time 1.2e-05s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_17_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 14 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 14 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_04_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 476 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 476 BUS records\n",
            "reading time 8e-06s\n",
            "sorting time 1.2e-05s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_22_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_28_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_04_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 485 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 485 BUS records\n",
            "reading time 8e-06s\n",
            "sorting time 1.2e-05s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_25_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_1_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 96 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 96 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 9e-06s\n",
            "writing time 2.8e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_27_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_06_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 303 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 303 BUS records\n",
            "reading time 7e-06s\n",
            "sorting time 8e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_15_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 15 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 15 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 2.2e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_27_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_3_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_19_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 4 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 4 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 7e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_1_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 122 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 122 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 9e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_06_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 295 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 295 BUS records\n",
            "reading time 6e-06s\n",
            "sorting time 1.3e-05s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_08_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 139 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 139 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 4e-06s\n",
            "writing time 2.6e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_04_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 498 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 498 BUS records\n",
            "reading time 8e-06s\n",
            "sorting time 1.3e-05s\n",
            "writing time 3.7e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_08_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 141 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 141 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 5e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_15_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 10 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 10 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 3.7e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_26_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_22_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 2 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 2 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 7e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_05_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 366 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 366 BUS records\n",
            "reading time 7e-06s\n",
            "sorting time 1e-05s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_26_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_24_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_15_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 16 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 16 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_28_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_03_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 573 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 573 BUS records\n",
            "reading time 8e-06s\n",
            "sorting time 2.3e-05s\n",
            "writing time 2.8e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_08_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 131 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 131 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 3e-06s\n",
            "writing time 2.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_21_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 1 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 1 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 7e-06s\n",
            "writing time 8.5e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_18_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 6 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 6 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_28_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 2e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_24_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 1 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 1 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 2.8e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_19_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 2 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 2 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 2.7e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_06_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 275 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 275 BUS records\n",
            "reading time 6e-06s\n",
            "sorting time 7e-06s\n",
            "writing time 3.8e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_16_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 6 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 6 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 3.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_18_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 4 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 4 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 3.6e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_29_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_2_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 2 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 2 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 2.8e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_17_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 11 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 11 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 3.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_07_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 203 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 203 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 1.1e-05s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_17_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 7 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 7 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 7e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_11_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 65 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 65 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 3e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_19_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 2 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 2 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_3_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_14_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 14 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 14 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_16_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 5 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 5 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_2_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_3_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_11_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 62 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 62 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 8e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_08_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 174 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 174 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 5e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_27_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_19_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 9 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 9 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 3.3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_14_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 15 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 15 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 3.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_08_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 149 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 149 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 5e-06s\n",
            "writing time 2.8e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_25_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_22_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 3 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 3 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_03_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 586 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 586 BUS records\n",
            "reading time 9e-06s\n",
            "sorting time 1.6e-05s\n",
            "writing time 2.5e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_23_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_27_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_28_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_12_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 54 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 54 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 2.8e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_21_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_13_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 43 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 43 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 2.8e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_02_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 631 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 631 BUS records\n",
            "reading time 9e-06s\n",
            "sorting time 1.7e-05s\n",
            "writing time 2.8e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_28_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 1 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 1 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 3.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_09_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 140 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 140 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 4e-06s\n",
            "writing time 2.4e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_19_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 6 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 6 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_15_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 12 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 12 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_11_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 74 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 74 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_24_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_13_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 38 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 38 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_1_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 93 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 93 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 3e-06s\n",
            "writing time 4.4e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_1_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 103 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 103 BUS records\n",
            "reading time 6e-06s\n",
            "sorting time 3e-06s\n",
            "writing time 3.2e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_17_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 7 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 7 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 3.8e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_2_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 1 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 1 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 2.7e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_26_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_16_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 9 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 9 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_14_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 17 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 17 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 3.5e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_13_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 34 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 34 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 3.2e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_19_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 2 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 2 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_12_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 50 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 50 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 2.2e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_16_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 9 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 9 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 7e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_11_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 65 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 65 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 2.8e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_07_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 217 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 217 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 5e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_15_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 15 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 15 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_02_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 637 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 637 BUS records\n",
            "reading time 9e-06s\n",
            "sorting time 1.7e-05s\n",
            "writing time 2.3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_02_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 632 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 632 BUS records\n",
            "reading time 9e-06s\n",
            "sorting time 2.6e-05s\n",
            "writing time 3.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_3_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_09_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 159 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 159 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 4e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_03_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 578 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 578 BUS records\n",
            "reading time 1e-05s\n",
            "sorting time 1.6e-05s\n",
            "writing time 3.4e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_26_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_18_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 7 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 7 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_22_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 2 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 2 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 7e-06s\n",
            "writing time 3.2e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_13_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 23 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 23 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 2.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_21_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 1 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 1 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 3.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_21_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_16_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 9 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 9 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 3.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_21_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 1 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 1 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 3.4e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_1_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 100 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 100 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 3e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_04_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 480 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 480 BUS records\n",
            "reading time 8e-06s\n",
            "sorting time 1.2e-05s\n",
            "writing time 7.5e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_2_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 2 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 2 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 8.5e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_22_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_27_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_07_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 207 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 207 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 1.1e-05s\n",
            "writing time 3.2e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_12_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 52 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 52 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 7e-06s\n",
            "writing time 3.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_23_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_15_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 12 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 12 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 3.7e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_2_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_25_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 2 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 2 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 2.6e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_18_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 6 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 6 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 7e-06s\n",
            "writing time 2.4e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_3_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_23_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_26_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_26_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_3_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_22_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 1 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 1 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 4.3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_29_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_07_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 207 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 207 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1e-05s\n",
            "writing time 2.8e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_11_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 76 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 76 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 2.4e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_13_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 31 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 31 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 3.2e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_22_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_16_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 6 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 6 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 3.4e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_06_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 288 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 288 BUS records\n",
            "reading time 6e-06s\n",
            "sorting time 8e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_14_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 17 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 17 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 3.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_18_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 3 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 3 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 9e-06s\n",
            "writing time 3.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_2_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 1 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 1 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_09_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 157 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 157 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 4e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_1_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 101 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 101 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 3e-06s\n",
            "writing time 2.5e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_26_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_18_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 7 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 7 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_14_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 30 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 30 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 8e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_02_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 636 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 636 BUS records\n",
            "reading time 9e-06s\n",
            "sorting time 2.5e-05s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_12_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 48 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 48 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 8e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_18_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 2 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 2 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_13_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 35 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 35 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 8e-06s\n",
            "writing time 3.3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_15_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 12 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 12 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 7e-06s\n",
            "writing time 6e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_3_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_12_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 63 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 63 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 0.000108s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_11_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 78 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 78 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 8e-06s\n",
            "writing time 3.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_02_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 637 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 637 BUS records\n",
            "reading time 1e-05s\n",
            "sorting time 1.7e-05s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_06_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 325 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 325 BUS records\n",
            "reading time 7e-06s\n",
            "sorting time 9e-06s\n",
            "writing time 3.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_28_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_23_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 1 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 1 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_3_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_27_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_28_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_09_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 134 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 134 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 1e-05s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_11_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 78 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 78 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_24_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_29_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_21_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 3 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 3 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 7e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_17_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 13 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 13 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 7e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_09_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 137 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 137 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 4e-06s\n",
            "writing time 2.5e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_03_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 577 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 577 BUS records\n",
            "reading time 9e-06s\n",
            "sorting time 2.2e-05s\n",
            "writing time 3.8e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_03_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 571 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 571 BUS records\n",
            "reading time 9e-06s\n",
            "sorting time 1.6e-05s\n",
            "writing time 3.2e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_26_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_03_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 577 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 577 BUS records\n",
            "reading time 9e-06s\n",
            "sorting time 1.6e-05s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_07_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 210 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 210 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 5e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_24_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_16_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 9 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 9 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_08_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 124 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 124 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 4e-06s\n",
            "writing time 3.3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_3_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_3_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_05_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 361 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 361 BUS records\n",
            "reading time 8e-06s\n",
            "sorting time 1.5e-05s\n",
            "writing time 3.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_09_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 140 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 140 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 1e-05s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_02_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 635 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 635 BUS records\n",
            "reading time 8e-06s\n",
            "sorting time 1.7e-05s\n",
            "writing time 3.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_22_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_08_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 148 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 148 BUS records\n",
            "reading time 7e-06s\n",
            "sorting time 5e-06s\n",
            "writing time 4.2e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_06_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 280 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 280 BUS records\n",
            "reading time 6e-06s\n",
            "sorting time 8e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_17_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 7 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 7 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 3.3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_13_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 30 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 30 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 2.7e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_13_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 31 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 31 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 2.8e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_15_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 18 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 18 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_22_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_27_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_04_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 467 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 467 BUS records\n",
            "reading time 8e-06s\n",
            "sorting time 1.1e-05s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_15_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 16 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 16 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 2.3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_04_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 500 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 500 BUS records\n",
            "reading time 8e-06s\n",
            "sorting time 1.2e-05s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_14_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 22 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 22 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_06_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 274 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 274 BUS records\n",
            "reading time 6e-06s\n",
            "sorting time 8e-06s\n",
            "writing time 2.8e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_06_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 305 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 305 BUS records\n",
            "reading time 6e-06s\n",
            "sorting time 1.4e-05s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_03_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 581 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 581 BUS records\n",
            "reading time 9e-06s\n",
            "sorting time 1.6e-05s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_25_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_11_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 68 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 68 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 3e-06s\n",
            "writing time 3.3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_24_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_05_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 401 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 401 BUS records\n",
            "reading time 1.4e-05s\n",
            "sorting time 1e-05s\n",
            "writing time 3.2e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_03_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 583 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 583 BUS records\n",
            "reading time 9e-06s\n",
            "sorting time 1.6e-05s\n",
            "writing time 3.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_16_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 8 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 8 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_24_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_24_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_29_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_18_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 3 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 3 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 2.4e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_1_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 114 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 114 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 3e-06s\n",
            "writing time 4.8e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_27_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_08_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 144 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 144 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 4e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_19_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 5 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 5 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 8e-06s\n",
            "writing time 2.2e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_04_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 482 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 482 BUS records\n",
            "reading time 7e-06s\n",
            "sorting time 1.8e-05s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_14_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 24 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 24 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 6.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_25_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 2e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_26_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_23_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_25_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_13_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 36 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 36 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_05_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 395 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 395 BUS records\n",
            "reading time 7e-06s\n",
            "sorting time 1e-05s\n",
            "writing time 3.4e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_29_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_17_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 9 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 9 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 7e-06s\n",
            "writing time 4e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_09_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 174 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 174 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 5e-06s\n",
            "writing time 2.8e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_21_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_09_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 139 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 139 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 4e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_06_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 305 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 305 BUS records\n",
            "reading time 6e-06s\n",
            "sorting time 8e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_05_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 379 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 379 BUS records\n",
            "reading time 7e-06s\n",
            "sorting time 1.5e-05s\n",
            "writing time 3.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_27_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_23_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 1 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 1 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_03_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 574 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 574 BUS records\n",
            "reading time 9e-06s\n",
            "sorting time 1.5e-05s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_09_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 144 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 144 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 4e-06s\n",
            "writing time 3.2e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_29_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_19_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 2 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 2 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 2.8e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_18_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 6 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 6 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 3.2e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_11_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 64 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 64 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 3e-06s\n",
            "writing time 6e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_29_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_1_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 94 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 94 BUS records\n",
            "reading time 1.9e-05s\n",
            "sorting time 3e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_03_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 588 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 588 BUS records\n",
            "reading time 9e-06s\n",
            "sorting time 1.6e-05s\n",
            "writing time 3.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_21_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 2e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_1_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 117 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 117 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 3e-06s\n",
            "writing time 3.7e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_23_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_23_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_07_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 214 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 214 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 5e-06s\n",
            "writing time 2.8e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_2_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 1 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 1 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 7e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_28_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_17_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 8 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 8 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 7e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_15_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 13 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 13 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 7e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_13_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 29 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 29 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 3.5e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_02_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 638 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 638 BUS records\n",
            "reading time 9e-06s\n",
            "sorting time 1.7e-05s\n",
            "writing time 3.4e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_29_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_23_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_14_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 26 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 26 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 7e-06s\n",
            "writing time 2.4e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_02_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 639 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 639 BUS records\n",
            "reading time 1e-05s\n",
            "sorting time 1.7e-05s\n",
            "writing time 3.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_25_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_17_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 8 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 8 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_12_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 45 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 45 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 8e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_05_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 383 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 383 BUS records\n",
            "reading time 6e-06s\n",
            "sorting time 1e-05s\n",
            "writing time 3.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_04_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 479 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 479 BUS records\n",
            "reading time 8e-06s\n",
            "sorting time 1.7e-05s\n",
            "writing time 3.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_28_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_07_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 197 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 197 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 5e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_27_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_21_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 1 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 1 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 1e-06s\n",
            "writing time 2.8e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_18_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 5 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 5 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 7e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_17_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 8 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 8 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 7e-06s\n",
            "writing time 3.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_08_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 142 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 142 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 1e-05s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_12_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 48 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 48 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 3e-06s\n",
            "writing time 3.5e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_04_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 475 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 475 BUS records\n",
            "reading time 1.5e-05s\n",
            "sorting time 1.1e-05s\n",
            "writing time 8.7e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_25_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 1 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 1 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/4_ebov_snp_2_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 3 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 3 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 3.4e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_11_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 62 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 62 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 3e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_14_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 20 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 20 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 3.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_05_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 399 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 399 BUS records\n",
            "reading time 7e-06s\n",
            "sorting time 1e-05s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/3_ebov_snp_09_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 146 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 146 BUS records\n",
            "reading time 6e-06s\n",
            "sorting time 5e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_16_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 5 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 5 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 7e-06s\n",
            "writing time 5.1e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_12_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 40 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 40 BUS records\n",
            "reading time 5e-06s\n",
            "sorting time 2e-06s\n",
            "writing time 2.4e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/5_ebov_snp_22_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_04_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 477 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 477 BUS records\n",
            "reading time 8e-06s\n",
            "sorting time 1.2e-05s\n",
            "writing time 3.6e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/9_ebov_snp_2_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 2 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 2 BUS records\n",
            "reading time 1e-05s\n",
            "sorting time 2e-06s\n",
            "writing time 6e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/0_ebov_snp_12_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 52 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 52 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 3e-06s\n",
            "writing time 2.3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/1_ebov_snp_02_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 636 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 636 BUS records\n",
            "reading time 9e-06s\n",
            "sorting time 1.8e-05s\n",
            "writing time 3.5e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_05_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 374 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 374 BUS records\n",
            "reading time 7e-06s\n",
            "sorting time 9e-06s\n",
            "writing time 2.4e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/6_ebov_snp_07_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 221 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 221 BUS records\n",
            "reading time 6e-06s\n",
            "sorting time 6e-06s\n",
            "writing time 3e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/7_ebov_snp_29_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 0 reads pseudoaligned[~warn] no reads pseudoaligned.\n",
            "\n",
            "\n",
            "Read in 0 BUS records\n",
            "reading time 3e-06s\n",
            "sorting time 0s\n",
            "writing time 0s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/8_ebov_snp_08_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 139 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 139 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 5e-06s\n",
            "writing time 2.9e-05s\n",
            "\n",
            "[index] k-mer length: 31\n",
            "[index] number of targets: 38\n",
            "[index] number of k-mers: 17,819\n",
            "[quant] running in single-end mode\n",
            "[quant] will process file 1: /content/LSCHWCP_2023/Notebooks/Figure_2/Figure_2c/mutated_files/2_ebov_snp_16_ms.fa\n",
            "[quant] finding pseudoalignments for all files ... done\n",
            "[quant] processed 676 reads, 11 reads pseudoaligned\n",
            "\n",
            " all fits in buffer\n",
            "Read in 11 BUS records\n",
            "reading time 4e-06s\n",
            "sorting time 6e-06s\n",
            "writing time 2.9e-05s\n"
          ]
        }
      ],
      "source": [
        "for mut_file in glob.glob(f\"{out}/*_ms.fa\"):\n",
        "    mut_name = mut_file.split(\"/\")[-1].split(\".fa\")[0]\n",
        "\n",
        "    !mkdir -p $kallisto_out/$mut_name\n",
        "\n",
        "    !$kallisto bus \\\n",
        "        -i $ebola_index \\\n",
        "        -o $kallisto_out/$mut_name \\\n",
        "        -x bulk \\\n",
        "        -t $threads \\\n",
        "        $mut_file\n",
        "\n",
        "    !$bustools sort -o $kallisto_out/$mut_name/output_sorted.bus $kallisto_out/$mut_name/output.bus\n",
        "\n",
        "    !$bustools count \\\n",
        "        --genecounts \\\n",
        "        --cm \\\n",
        "        -o $kallisto_out/$mut_name/bustools_count/ \\\n",
        "        -g $ebola_t2g \\\n",
        "        -e $kallisto_out/$mut_name/matrix.ec \\\n",
        "        -t $kallisto_out/$mut_name/transcripts.txt \\\n",
        "        $kallisto_out/$mut_name/output_sorted.bus"
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Download generated alignments:"
      ],
      "metadata": {
        "id": "tMzaZyxhuRSL"
      },
      "id": "tMzaZyxhuRSL"
    },
    {
      "cell_type": "code",
      "source": [
        "# Remove all kallisto files we do not need anymore to reduce size of alignment folders\n",
        "for alignment_type in [\"standard\", \"translated\"]:\n",
        "  for folder in glob.glob(f\"/content/kallisto_alignment/{alignment_type}/*\"):\n",
        "    for file in glob.glob(f\"{folder}/*.*\"):\n",
        "      !rm $file"
      ],
      "metadata": {
        "id": "-yJqpnUMed7u"
      },
      "id": "-yJqpnUMed7u",
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "id": "6f497de2-2cf5-4413-8842-3a6933c18632",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 1000
        },
        "id": "6f497de2-2cf5-4413-8842-3a6933c18632",
        "outputId": "0bfb2999-8459-497d-8aac-ff1bbe974599"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "kallisto_alignment/\n",
            "kallisto_alignment/translated/\n",
            "kallisto_alignment/translated/2_ebov_snp_03_ms/\n",
            "kallisto_alignment/translated/2_ebov_snp_03_ms/bustools_count/\n",
            "kallisto_alignment/translated/2_ebov_snp_03_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/2_ebov_snp_03_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_03_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_22_ms/\n",
            "kallisto_alignment/translated/1_ebov_snp_22_ms/bustools_count/\n",
            "kallisto_alignment/translated/1_ebov_snp_22_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/1_ebov_snp_22_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_22_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_09_ms/\n",
            "kallisto_alignment/translated/3_ebov_snp_09_ms/bustools_count/\n",
            "kallisto_alignment/translated/3_ebov_snp_09_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/3_ebov_snp_09_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_09_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_1_ms/\n",
            "kallisto_alignment/translated/8_ebov_snp_1_ms/bustools_count/\n",
            "kallisto_alignment/translated/8_ebov_snp_1_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/8_ebov_snp_1_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_1_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_11_ms/\n",
            "kallisto_alignment/translated/2_ebov_snp_11_ms/bustools_count/\n",
            "kallisto_alignment/translated/2_ebov_snp_11_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/2_ebov_snp_11_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_11_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_06_ms/\n",
            "kallisto_alignment/translated/0_ebov_snp_06_ms/bustools_count/\n",
            "kallisto_alignment/translated/0_ebov_snp_06_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/0_ebov_snp_06_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_06_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_16_ms/\n",
            "kallisto_alignment/translated/2_ebov_snp_16_ms/bustools_count/\n",
            "kallisto_alignment/translated/2_ebov_snp_16_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/2_ebov_snp_16_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_16_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_02_ms/\n",
            "kallisto_alignment/translated/5_ebov_snp_02_ms/bustools_count/\n",
            "kallisto_alignment/translated/5_ebov_snp_02_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/5_ebov_snp_02_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_02_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_25_ms/\n",
            "kallisto_alignment/translated/4_ebov_snp_25_ms/bustools_count/\n",
            "kallisto_alignment/translated/4_ebov_snp_25_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/4_ebov_snp_25_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_25_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_2_ms/\n",
            "kallisto_alignment/translated/7_ebov_snp_2_ms/bustools_count/\n",
            "kallisto_alignment/translated/7_ebov_snp_2_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/7_ebov_snp_2_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_2_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_28_ms/\n",
            "kallisto_alignment/translated/9_ebov_snp_28_ms/bustools_count/\n",
            "kallisto_alignment/translated/9_ebov_snp_28_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/9_ebov_snp_28_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_28_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_06_ms/\n",
            "kallisto_alignment/translated/5_ebov_snp_06_ms/bustools_count/\n",
            "kallisto_alignment/translated/5_ebov_snp_06_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/5_ebov_snp_06_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_06_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_18_ms/\n",
            "kallisto_alignment/translated/1_ebov_snp_18_ms/bustools_count/\n",
            "kallisto_alignment/translated/1_ebov_snp_18_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/1_ebov_snp_18_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_18_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_05_ms/\n",
            "kallisto_alignment/translated/7_ebov_snp_05_ms/bustools_count/\n",
            "kallisto_alignment/translated/7_ebov_snp_05_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/7_ebov_snp_05_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_05_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_28_ms/\n",
            "kallisto_alignment/translated/3_ebov_snp_28_ms/bustools_count/\n",
            "kallisto_alignment/translated/3_ebov_snp_28_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/3_ebov_snp_28_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_28_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_21_ms/\n",
            "kallisto_alignment/translated/1_ebov_snp_21_ms/bustools_count/\n",
            "kallisto_alignment/translated/1_ebov_snp_21_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/1_ebov_snp_21_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_21_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_09_ms/\n",
            "kallisto_alignment/translated/8_ebov_snp_09_ms/bustools_count/\n",
            "kallisto_alignment/translated/8_ebov_snp_09_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/8_ebov_snp_09_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_09_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_19_ms/\n",
            "kallisto_alignment/translated/9_ebov_snp_19_ms/bustools_count/\n",
            "kallisto_alignment/translated/9_ebov_snp_19_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/9_ebov_snp_19_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_19_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_29_ms/\n",
            "kallisto_alignment/translated/9_ebov_snp_29_ms/bustools_count/\n",
            "kallisto_alignment/translated/9_ebov_snp_29_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/9_ebov_snp_29_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_29_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_2_ms/\n",
            "kallisto_alignment/translated/3_ebov_snp_2_ms/bustools_count/\n",
            "kallisto_alignment/translated/3_ebov_snp_2_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/3_ebov_snp_2_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_2_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_28_ms/\n",
            "kallisto_alignment/translated/4_ebov_snp_28_ms/bustools_count/\n",
            "kallisto_alignment/translated/4_ebov_snp_28_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/4_ebov_snp_28_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_28_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_11_ms/\n",
            "kallisto_alignment/translated/3_ebov_snp_11_ms/bustools_count/\n",
            "kallisto_alignment/translated/3_ebov_snp_11_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/3_ebov_snp_11_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_11_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_08_ms/\n",
            "kallisto_alignment/translated/6_ebov_snp_08_ms/bustools_count/\n",
            "kallisto_alignment/translated/6_ebov_snp_08_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/6_ebov_snp_08_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_08_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_18_ms/\n",
            "kallisto_alignment/translated/0_ebov_snp_18_ms/bustools_count/\n",
            "kallisto_alignment/translated/0_ebov_snp_18_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/0_ebov_snp_18_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_18_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_27_ms/\n",
            "kallisto_alignment/translated/8_ebov_snp_27_ms/bustools_count/\n",
            "kallisto_alignment/translated/8_ebov_snp_27_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/8_ebov_snp_27_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_27_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_04_ms/\n",
            "kallisto_alignment/translated/3_ebov_snp_04_ms/bustools_count/\n",
            "kallisto_alignment/translated/3_ebov_snp_04_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/3_ebov_snp_04_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_04_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_09_ms/\n",
            "kallisto_alignment/translated/4_ebov_snp_09_ms/bustools_count/\n",
            "kallisto_alignment/translated/4_ebov_snp_09_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/4_ebov_snp_09_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_09_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_08_ms/\n",
            "kallisto_alignment/translated/5_ebov_snp_08_ms/bustools_count/\n",
            "kallisto_alignment/translated/5_ebov_snp_08_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/5_ebov_snp_08_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_08_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_13_ms/\n",
            "kallisto_alignment/translated/0_ebov_snp_13_ms/bustools_count/\n",
            "kallisto_alignment/translated/0_ebov_snp_13_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/0_ebov_snp_13_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_13_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_1_ms/\n",
            "kallisto_alignment/translated/7_ebov_snp_1_ms/bustools_count/\n",
            "kallisto_alignment/translated/7_ebov_snp_1_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/7_ebov_snp_1_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_1_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_02_ms/\n",
            "kallisto_alignment/translated/3_ebov_snp_02_ms/bustools_count/\n",
            "kallisto_alignment/translated/3_ebov_snp_02_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/3_ebov_snp_02_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_02_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_17_ms/\n",
            "kallisto_alignment/translated/5_ebov_snp_17_ms/bustools_count/\n",
            "kallisto_alignment/translated/5_ebov_snp_17_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/5_ebov_snp_17_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_17_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_3_ms/\n",
            "kallisto_alignment/translated/6_ebov_snp_3_ms/bustools_count/\n",
            "kallisto_alignment/translated/6_ebov_snp_3_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/6_ebov_snp_3_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_3_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_2_ms/\n",
            "kallisto_alignment/translated/0_ebov_snp_2_ms/bustools_count/\n",
            "kallisto_alignment/translated/0_ebov_snp_2_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/0_ebov_snp_2_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_2_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_21_ms/\n",
            "kallisto_alignment/translated/4_ebov_snp_21_ms/bustools_count/\n",
            "kallisto_alignment/translated/4_ebov_snp_21_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/4_ebov_snp_21_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_21_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_13_ms/\n",
            "kallisto_alignment/translated/1_ebov_snp_13_ms/bustools_count/\n",
            "kallisto_alignment/translated/1_ebov_snp_13_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/1_ebov_snp_13_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_13_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_19_ms/\n",
            "kallisto_alignment/translated/1_ebov_snp_19_ms/bustools_count/\n",
            "kallisto_alignment/translated/1_ebov_snp_19_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/1_ebov_snp_19_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_19_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_12_ms/\n",
            "kallisto_alignment/translated/6_ebov_snp_12_ms/bustools_count/\n",
            "kallisto_alignment/translated/6_ebov_snp_12_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/6_ebov_snp_12_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_12_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_24_ms/\n",
            "kallisto_alignment/translated/6_ebov_snp_24_ms/bustools_count/\n",
            "kallisto_alignment/translated/6_ebov_snp_24_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/6_ebov_snp_24_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_24_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_29_ms/\n",
            "kallisto_alignment/translated/7_ebov_snp_29_ms/bustools_count/\n",
            "kallisto_alignment/translated/7_ebov_snp_29_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/7_ebov_snp_29_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_29_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_2_ms/\n",
            "kallisto_alignment/translated/1_ebov_snp_2_ms/bustools_count/\n",
            "kallisto_alignment/translated/1_ebov_snp_2_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/1_ebov_snp_2_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_2_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_22_ms/\n",
            "kallisto_alignment/translated/2_ebov_snp_22_ms/bustools_count/\n",
            "kallisto_alignment/translated/2_ebov_snp_22_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/2_ebov_snp_22_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_22_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_19_ms/\n",
            "kallisto_alignment/translated/5_ebov_snp_19_ms/bustools_count/\n",
            "kallisto_alignment/translated/5_ebov_snp_19_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/5_ebov_snp_19_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_19_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_14_ms/\n",
            "kallisto_alignment/translated/8_ebov_snp_14_ms/bustools_count/\n",
            "kallisto_alignment/translated/8_ebov_snp_14_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/8_ebov_snp_14_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_14_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_02_ms/\n",
            "kallisto_alignment/translated/6_ebov_snp_02_ms/bustools_count/\n",
            "kallisto_alignment/translated/6_ebov_snp_02_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/6_ebov_snp_02_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_02_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_09_ms/\n",
            "kallisto_alignment/translated/1_ebov_snp_09_ms/bustools_count/\n",
            "kallisto_alignment/translated/1_ebov_snp_09_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/1_ebov_snp_09_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_09_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_12_ms/\n",
            "kallisto_alignment/translated/3_ebov_snp_12_ms/bustools_count/\n",
            "kallisto_alignment/translated/3_ebov_snp_12_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/3_ebov_snp_12_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_12_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_08_ms/\n",
            "kallisto_alignment/translated/9_ebov_snp_08_ms/bustools_count/\n",
            "kallisto_alignment/translated/9_ebov_snp_08_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/9_ebov_snp_08_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_08_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_04_ms/\n",
            "kallisto_alignment/translated/0_ebov_snp_04_ms/bustools_count/\n",
            "kallisto_alignment/translated/0_ebov_snp_04_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/0_ebov_snp_04_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_04_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_05_ms/\n",
            "kallisto_alignment/translated/2_ebov_snp_05_ms/bustools_count/\n",
            "kallisto_alignment/translated/2_ebov_snp_05_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/2_ebov_snp_05_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_05_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_14_ms/\n",
            "kallisto_alignment/translated/1_ebov_snp_14_ms/bustools_count/\n",
            "kallisto_alignment/translated/1_ebov_snp_14_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/1_ebov_snp_14_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_14_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_19_ms/\n",
            "kallisto_alignment/translated/0_ebov_snp_19_ms/bustools_count/\n",
            "kallisto_alignment/translated/0_ebov_snp_19_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/0_ebov_snp_19_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_19_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_29_ms/\n",
            "kallisto_alignment/translated/1_ebov_snp_29_ms/bustools_count/\n",
            "kallisto_alignment/translated/1_ebov_snp_29_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/1_ebov_snp_29_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_29_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_14_ms/\n",
            "kallisto_alignment/translated/3_ebov_snp_14_ms/bustools_count/\n",
            "kallisto_alignment/translated/3_ebov_snp_14_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/3_ebov_snp_14_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_14_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_02_ms/\n",
            "kallisto_alignment/translated/8_ebov_snp_02_ms/bustools_count/\n",
            "kallisto_alignment/translated/8_ebov_snp_02_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/8_ebov_snp_02_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_02_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_11_ms/\n",
            "kallisto_alignment/translated/8_ebov_snp_11_ms/bustools_count/\n",
            "kallisto_alignment/translated/8_ebov_snp_11_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/8_ebov_snp_11_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_11_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_13_ms/\n",
            "kallisto_alignment/translated/8_ebov_snp_13_ms/bustools_count/\n",
            "kallisto_alignment/translated/8_ebov_snp_13_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/8_ebov_snp_13_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_13_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_28_ms/\n",
            "kallisto_alignment/translated/5_ebov_snp_28_ms/bustools_count/\n",
            "kallisto_alignment/translated/5_ebov_snp_28_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/5_ebov_snp_28_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_28_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_09_ms/\n",
            "kallisto_alignment/translated/5_ebov_snp_09_ms/bustools_count/\n",
            "kallisto_alignment/translated/5_ebov_snp_09_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/5_ebov_snp_09_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_09_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_23_ms/\n",
            "kallisto_alignment/translated/2_ebov_snp_23_ms/bustools_count/\n",
            "kallisto_alignment/translated/2_ebov_snp_23_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/2_ebov_snp_23_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_23_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_27_ms/\n",
            "kallisto_alignment/translated/1_ebov_snp_27_ms/bustools_count/\n",
            "kallisto_alignment/translated/1_ebov_snp_27_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/1_ebov_snp_27_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_27_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_04_ms/\n",
            "kallisto_alignment/translated/8_ebov_snp_04_ms/bustools_count/\n",
            "kallisto_alignment/translated/8_ebov_snp_04_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/8_ebov_snp_04_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_04_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_24_ms/\n",
            "kallisto_alignment/translated/2_ebov_snp_24_ms/bustools_count/\n",
            "kallisto_alignment/translated/2_ebov_snp_24_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/2_ebov_snp_24_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_24_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_08_ms/\n",
            "kallisto_alignment/translated/4_ebov_snp_08_ms/bustools_count/\n",
            "kallisto_alignment/translated/4_ebov_snp_08_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/4_ebov_snp_08_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_08_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_3_ms/\n",
            "kallisto_alignment/translated/9_ebov_snp_3_ms/bustools_count/\n",
            "kallisto_alignment/translated/9_ebov_snp_3_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/9_ebov_snp_3_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_3_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_22_ms/\n",
            "kallisto_alignment/translated/4_ebov_snp_22_ms/bustools_count/\n",
            "kallisto_alignment/translated/4_ebov_snp_22_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/4_ebov_snp_22_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_22_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_29_ms/\n",
            "kallisto_alignment/translated/8_ebov_snp_29_ms/bustools_count/\n",
            "kallisto_alignment/translated/8_ebov_snp_29_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/8_ebov_snp_29_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_29_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_25_ms/\n",
            "kallisto_alignment/translated/2_ebov_snp_25_ms/bustools_count/\n",
            "kallisto_alignment/translated/2_ebov_snp_25_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/2_ebov_snp_25_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_25_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_24_ms/\n",
            "kallisto_alignment/translated/5_ebov_snp_24_ms/bustools_count/\n",
            "kallisto_alignment/translated/5_ebov_snp_24_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/5_ebov_snp_24_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_24_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_3_ms/\n",
            "kallisto_alignment/translated/8_ebov_snp_3_ms/bustools_count/\n",
            "kallisto_alignment/translated/8_ebov_snp_3_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/8_ebov_snp_3_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_3_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_14_ms/\n",
            "kallisto_alignment/translated/9_ebov_snp_14_ms/bustools_count/\n",
            "kallisto_alignment/translated/9_ebov_snp_14_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/9_ebov_snp_14_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_14_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_11_ms/\n",
            "kallisto_alignment/translated/5_ebov_snp_11_ms/bustools_count/\n",
            "kallisto_alignment/translated/5_ebov_snp_11_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/5_ebov_snp_11_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_11_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_3_ms/\n",
            "kallisto_alignment/translated/3_ebov_snp_3_ms/bustools_count/\n",
            "kallisto_alignment/translated/3_ebov_snp_3_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/3_ebov_snp_3_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_3_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_0_ms/\n",
            "kallisto_alignment/translated/0_ebov_snp_0_ms/bustools_count/\n",
            "kallisto_alignment/translated/0_ebov_snp_0_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/0_ebov_snp_0_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_0_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_24_ms/\n",
            "kallisto_alignment/translated/4_ebov_snp_24_ms/bustools_count/\n",
            "kallisto_alignment/translated/4_ebov_snp_24_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/4_ebov_snp_24_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_24_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_25_ms/\n",
            "kallisto_alignment/translated/6_ebov_snp_25_ms/bustools_count/\n",
            "kallisto_alignment/translated/6_ebov_snp_25_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/6_ebov_snp_25_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_25_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_11_ms/\n",
            "kallisto_alignment/translated/4_ebov_snp_11_ms/bustools_count/\n",
            "kallisto_alignment/translated/4_ebov_snp_11_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/4_ebov_snp_11_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_11_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_22_ms/\n",
            "kallisto_alignment/translated/0_ebov_snp_22_ms/bustools_count/\n",
            "kallisto_alignment/translated/0_ebov_snp_22_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/0_ebov_snp_22_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_22_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_12_ms/\n",
            "kallisto_alignment/translated/2_ebov_snp_12_ms/bustools_count/\n",
            "kallisto_alignment/translated/2_ebov_snp_12_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/2_ebov_snp_12_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_12_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_2_ms/\n",
            "kallisto_alignment/translated/6_ebov_snp_2_ms/bustools_count/\n",
            "kallisto_alignment/translated/6_ebov_snp_2_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/6_ebov_snp_2_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_2_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_02_ms/\n",
            "kallisto_alignment/translated/4_ebov_snp_02_ms/bustools_count/\n",
            "kallisto_alignment/translated/4_ebov_snp_02_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/4_ebov_snp_02_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_02_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_02_ms/\n",
            "kallisto_alignment/translated/0_ebov_snp_02_ms/bustools_count/\n",
            "kallisto_alignment/translated/0_ebov_snp_02_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/0_ebov_snp_02_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_02_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_28_ms/\n",
            "kallisto_alignment/translated/7_ebov_snp_28_ms/bustools_count/\n",
            "kallisto_alignment/translated/7_ebov_snp_28_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/7_ebov_snp_28_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_28_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_05_ms/\n",
            "kallisto_alignment/translated/6_ebov_snp_05_ms/bustools_count/\n",
            "kallisto_alignment/translated/6_ebov_snp_05_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/6_ebov_snp_05_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_05_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_03_ms/\n",
            "kallisto_alignment/translated/6_ebov_snp_03_ms/bustools_count/\n",
            "kallisto_alignment/translated/6_ebov_snp_03_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/6_ebov_snp_03_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_03_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_18_ms/\n",
            "kallisto_alignment/translated/5_ebov_snp_18_ms/bustools_count/\n",
            "kallisto_alignment/translated/5_ebov_snp_18_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/5_ebov_snp_18_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_18_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_15_ms/\n",
            "kallisto_alignment/translated/1_ebov_snp_15_ms/bustools_count/\n",
            "kallisto_alignment/translated/1_ebov_snp_15_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/1_ebov_snp_15_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_15_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_06_ms/\n",
            "kallisto_alignment/translated/2_ebov_snp_06_ms/bustools_count/\n",
            "kallisto_alignment/translated/2_ebov_snp_06_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/2_ebov_snp_06_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_06_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_29_ms/\n",
            "kallisto_alignment/translated/3_ebov_snp_29_ms/bustools_count/\n",
            "kallisto_alignment/translated/3_ebov_snp_29_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/3_ebov_snp_29_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_29_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_15_ms/\n",
            "kallisto_alignment/translated/7_ebov_snp_15_ms/bustools_count/\n",
            "kallisto_alignment/translated/7_ebov_snp_15_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/7_ebov_snp_15_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_15_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_12_ms/\n",
            "kallisto_alignment/translated/4_ebov_snp_12_ms/bustools_count/\n",
            "kallisto_alignment/translated/4_ebov_snp_12_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/4_ebov_snp_12_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_12_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_2_ms/\n",
            "kallisto_alignment/translated/4_ebov_snp_2_ms/bustools_count/\n",
            "kallisto_alignment/translated/4_ebov_snp_2_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/4_ebov_snp_2_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_2_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_19_ms/\n",
            "kallisto_alignment/translated/7_ebov_snp_19_ms/bustools_count/\n",
            "kallisto_alignment/translated/7_ebov_snp_19_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/7_ebov_snp_19_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_19_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_14_ms/\n",
            "kallisto_alignment/translated/0_ebov_snp_14_ms/bustools_count/\n",
            "kallisto_alignment/translated/0_ebov_snp_14_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/0_ebov_snp_14_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_14_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_08_ms/\n",
            "kallisto_alignment/translated/3_ebov_snp_08_ms/bustools_count/\n",
            "kallisto_alignment/translated/3_ebov_snp_08_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/3_ebov_snp_08_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_08_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_12_ms/\n",
            "kallisto_alignment/translated/1_ebov_snp_12_ms/bustools_count/\n",
            "kallisto_alignment/translated/1_ebov_snp_12_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/1_ebov_snp_12_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_12_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_15_ms/\n",
            "kallisto_alignment/translated/2_ebov_snp_15_ms/bustools_count/\n",
            "kallisto_alignment/translated/2_ebov_snp_15_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/2_ebov_snp_15_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_15_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_17_ms/\n",
            "kallisto_alignment/translated/3_ebov_snp_17_ms/bustools_count/\n",
            "kallisto_alignment/translated/3_ebov_snp_17_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/3_ebov_snp_17_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_17_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_06_ms/\n",
            "kallisto_alignment/translated/4_ebov_snp_06_ms/bustools_count/\n",
            "kallisto_alignment/translated/4_ebov_snp_06_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/4_ebov_snp_06_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_06_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_22_ms/\n",
            "kallisto_alignment/translated/8_ebov_snp_22_ms/bustools_count/\n",
            "kallisto_alignment/translated/8_ebov_snp_22_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/8_ebov_snp_22_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_22_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_21_ms/\n",
            "kallisto_alignment/translated/3_ebov_snp_21_ms/bustools_count/\n",
            "kallisto_alignment/translated/3_ebov_snp_21_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/3_ebov_snp_21_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_21_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_03_ms/\n",
            "kallisto_alignment/translated/4_ebov_snp_03_ms/bustools_count/\n",
            "kallisto_alignment/translated/4_ebov_snp_03_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/4_ebov_snp_03_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_03_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_25_ms/\n",
            "kallisto_alignment/translated/5_ebov_snp_25_ms/bustools_count/\n",
            "kallisto_alignment/translated/5_ebov_snp_25_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/5_ebov_snp_25_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_25_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_28_ms/\n",
            "kallisto_alignment/translated/2_ebov_snp_28_ms/bustools_count/\n",
            "kallisto_alignment/translated/2_ebov_snp_28_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/2_ebov_snp_28_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_28_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_23_ms/\n",
            "kallisto_alignment/translated/4_ebov_snp_23_ms/bustools_count/\n",
            "kallisto_alignment/translated/4_ebov_snp_23_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/4_ebov_snp_23_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_23_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_06_ms/\n",
            "kallisto_alignment/translated/1_ebov_snp_06_ms/bustools_count/\n",
            "kallisto_alignment/translated/1_ebov_snp_06_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/1_ebov_snp_06_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_06_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_07_ms/\n",
            "kallisto_alignment/translated/3_ebov_snp_07_ms/bustools_count/\n",
            "kallisto_alignment/translated/3_ebov_snp_07_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/3_ebov_snp_07_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_07_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_17_ms/\n",
            "kallisto_alignment/translated/1_ebov_snp_17_ms/bustools_count/\n",
            "kallisto_alignment/translated/1_ebov_snp_17_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/1_ebov_snp_17_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_17_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_16_ms/\n",
            "kallisto_alignment/translated/1_ebov_snp_16_ms/bustools_count/\n",
            "kallisto_alignment/translated/1_ebov_snp_16_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/1_ebov_snp_16_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_16_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_07_ms/\n",
            "kallisto_alignment/translated/8_ebov_snp_07_ms/bustools_count/\n",
            "kallisto_alignment/translated/8_ebov_snp_07_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/8_ebov_snp_07_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_07_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_25_ms/\n",
            "kallisto_alignment/translated/7_ebov_snp_25_ms/bustools_count/\n",
            "kallisto_alignment/translated/7_ebov_snp_25_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/7_ebov_snp_25_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_25_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_03_ms/\n",
            "kallisto_alignment/translated/7_ebov_snp_03_ms/bustools_count/\n",
            "kallisto_alignment/translated/7_ebov_snp_03_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/7_ebov_snp_03_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_03_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_04_ms/\n",
            "kallisto_alignment/translated/5_ebov_snp_04_ms/bustools_count/\n",
            "kallisto_alignment/translated/5_ebov_snp_04_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/5_ebov_snp_04_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_04_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_08_ms/\n",
            "kallisto_alignment/translated/1_ebov_snp_08_ms/bustools_count/\n",
            "kallisto_alignment/translated/1_ebov_snp_08_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/1_ebov_snp_08_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_08_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_02_ms/\n",
            "kallisto_alignment/translated/2_ebov_snp_02_ms/bustools_count/\n",
            "kallisto_alignment/translated/2_ebov_snp_02_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/2_ebov_snp_02_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_02_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_23_ms/\n",
            "kallisto_alignment/translated/0_ebov_snp_23_ms/bustools_count/\n",
            "kallisto_alignment/translated/0_ebov_snp_23_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/0_ebov_snp_23_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_23_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_08_ms/\n",
            "kallisto_alignment/translated/2_ebov_snp_08_ms/bustools_count/\n",
            "kallisto_alignment/translated/2_ebov_snp_08_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/2_ebov_snp_08_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_08_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_18_ms/\n",
            "kallisto_alignment/translated/2_ebov_snp_18_ms/bustools_count/\n",
            "kallisto_alignment/translated/2_ebov_snp_18_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/2_ebov_snp_18_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_18_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_08_ms/\n",
            "kallisto_alignment/translated/8_ebov_snp_08_ms/bustools_count/\n",
            "kallisto_alignment/translated/8_ebov_snp_08_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/8_ebov_snp_08_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_08_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_16_ms/\n",
            "kallisto_alignment/translated/6_ebov_snp_16_ms/bustools_count/\n",
            "kallisto_alignment/translated/6_ebov_snp_16_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/6_ebov_snp_16_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_16_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_05_ms/\n",
            "kallisto_alignment/translated/1_ebov_snp_05_ms/bustools_count/\n",
            "kallisto_alignment/translated/1_ebov_snp_05_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/1_ebov_snp_05_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_05_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_24_ms/\n",
            "kallisto_alignment/translated/3_ebov_snp_24_ms/bustools_count/\n",
            "kallisto_alignment/translated/3_ebov_snp_24_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/3_ebov_snp_24_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_24_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_23_ms/\n",
            "kallisto_alignment/translated/1_ebov_snp_23_ms/bustools_count/\n",
            "kallisto_alignment/translated/1_ebov_snp_23_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/1_ebov_snp_23_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_23_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_09_ms/\n",
            "kallisto_alignment/translated/0_ebov_snp_09_ms/bustools_count/\n",
            "kallisto_alignment/translated/0_ebov_snp_09_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/0_ebov_snp_09_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_09_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_3_ms/\n",
            "kallisto_alignment/translated/7_ebov_snp_3_ms/bustools_count/\n",
            "kallisto_alignment/translated/7_ebov_snp_3_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/7_ebov_snp_3_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_3_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_1_ms/\n",
            "kallisto_alignment/translated/6_ebov_snp_1_ms/bustools_count/\n",
            "kallisto_alignment/translated/6_ebov_snp_1_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/6_ebov_snp_1_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_1_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_21_ms/\n",
            "kallisto_alignment/translated/5_ebov_snp_21_ms/bustools_count/\n",
            "kallisto_alignment/translated/5_ebov_snp_21_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/5_ebov_snp_21_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_21_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_14_ms/\n",
            "kallisto_alignment/translated/7_ebov_snp_14_ms/bustools_count/\n",
            "kallisto_alignment/translated/7_ebov_snp_14_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/7_ebov_snp_14_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_14_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_12_ms/\n",
            "kallisto_alignment/translated/8_ebov_snp_12_ms/bustools_count/\n",
            "kallisto_alignment/translated/8_ebov_snp_12_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/8_ebov_snp_12_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_12_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_2_ms/\n",
            "kallisto_alignment/translated/5_ebov_snp_2_ms/bustools_count/\n",
            "kallisto_alignment/translated/5_ebov_snp_2_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/5_ebov_snp_2_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_2_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_21_ms/\n",
            "kallisto_alignment/translated/7_ebov_snp_21_ms/bustools_count/\n",
            "kallisto_alignment/translated/7_ebov_snp_21_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/7_ebov_snp_21_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_21_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_07_ms/\n",
            "kallisto_alignment/translated/7_ebov_snp_07_ms/bustools_count/\n",
            "kallisto_alignment/translated/7_ebov_snp_07_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/7_ebov_snp_07_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_07_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_19_ms/\n",
            "kallisto_alignment/translated/2_ebov_snp_19_ms/bustools_count/\n",
            "kallisto_alignment/translated/2_ebov_snp_19_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/2_ebov_snp_19_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_19_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_23_ms/\n",
            "kallisto_alignment/translated/5_ebov_snp_23_ms/bustools_count/\n",
            "kallisto_alignment/translated/5_ebov_snp_23_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/5_ebov_snp_23_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_23_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_15_ms/\n",
            "kallisto_alignment/translated/8_ebov_snp_15_ms/bustools_count/\n",
            "kallisto_alignment/translated/8_ebov_snp_15_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/8_ebov_snp_15_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_15_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_26_ms/\n",
            "kallisto_alignment/translated/6_ebov_snp_26_ms/bustools_count/\n",
            "kallisto_alignment/translated/6_ebov_snp_26_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/6_ebov_snp_26_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_26_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_05_ms/\n",
            "kallisto_alignment/translated/9_ebov_snp_05_ms/bustools_count/\n",
            "kallisto_alignment/translated/9_ebov_snp_05_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/9_ebov_snp_05_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_05_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_07_ms/\n",
            "kallisto_alignment/translated/1_ebov_snp_07_ms/bustools_count/\n",
            "kallisto_alignment/translated/1_ebov_snp_07_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/1_ebov_snp_07_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_07_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_05_ms/\n",
            "kallisto_alignment/translated/8_ebov_snp_05_ms/bustools_count/\n",
            "kallisto_alignment/translated/8_ebov_snp_05_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/8_ebov_snp_05_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_05_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_06_ms/\n",
            "kallisto_alignment/translated/8_ebov_snp_06_ms/bustools_count/\n",
            "kallisto_alignment/translated/8_ebov_snp_06_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/8_ebov_snp_06_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_06_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_04_ms/\n",
            "kallisto_alignment/translated/6_ebov_snp_04_ms/bustools_count/\n",
            "kallisto_alignment/translated/6_ebov_snp_04_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/6_ebov_snp_04_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_04_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_13_ms/\n",
            "kallisto_alignment/translated/7_ebov_snp_13_ms/bustools_count/\n",
            "kallisto_alignment/translated/7_ebov_snp_13_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/7_ebov_snp_13_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_13_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_21_ms/\n",
            "kallisto_alignment/translated/0_ebov_snp_21_ms/bustools_count/\n",
            "kallisto_alignment/translated/0_ebov_snp_21_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/0_ebov_snp_21_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_21_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_09_ms/\n",
            "kallisto_alignment/translated/6_ebov_snp_09_ms/bustools_count/\n",
            "kallisto_alignment/translated/6_ebov_snp_09_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/6_ebov_snp_09_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_09_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_04_ms/\n",
            "kallisto_alignment/translated/1_ebov_snp_04_ms/bustools_count/\n",
            "kallisto_alignment/translated/1_ebov_snp_04_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/1_ebov_snp_04_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_04_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_26_ms/\n",
            "kallisto_alignment/translated/7_ebov_snp_26_ms/bustools_count/\n",
            "kallisto_alignment/translated/7_ebov_snp_26_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/7_ebov_snp_26_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_26_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_29_ms/\n",
            "kallisto_alignment/translated/6_ebov_snp_29_ms/bustools_count/\n",
            "kallisto_alignment/translated/6_ebov_snp_29_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/6_ebov_snp_29_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_29_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_23_ms/\n",
            "kallisto_alignment/translated/9_ebov_snp_23_ms/bustools_count/\n",
            "kallisto_alignment/translated/9_ebov_snp_23_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/9_ebov_snp_23_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_23_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_18_ms/\n",
            "kallisto_alignment/translated/9_ebov_snp_18_ms/bustools_count/\n",
            "kallisto_alignment/translated/9_ebov_snp_18_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/9_ebov_snp_18_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_18_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_24_ms/\n",
            "kallisto_alignment/translated/7_ebov_snp_24_ms/bustools_count/\n",
            "kallisto_alignment/translated/7_ebov_snp_24_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/7_ebov_snp_24_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_24_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_03_ms/\n",
            "kallisto_alignment/translated/8_ebov_snp_03_ms/bustools_count/\n",
            "kallisto_alignment/translated/8_ebov_snp_03_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/8_ebov_snp_03_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_03_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_09_ms/\n",
            "kallisto_alignment/translated/2_ebov_snp_09_ms/bustools_count/\n",
            "kallisto_alignment/translated/2_ebov_snp_09_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/2_ebov_snp_09_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_09_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_18_ms/\n",
            "kallisto_alignment/translated/7_ebov_snp_18_ms/bustools_count/\n",
            "kallisto_alignment/translated/7_ebov_snp_18_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/7_ebov_snp_18_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_18_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_15_ms/\n",
            "kallisto_alignment/translated/3_ebov_snp_15_ms/bustools_count/\n",
            "kallisto_alignment/translated/3_ebov_snp_15_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/3_ebov_snp_15_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_15_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_11_ms/\n",
            "kallisto_alignment/translated/6_ebov_snp_11_ms/bustools_count/\n",
            "kallisto_alignment/translated/6_ebov_snp_11_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/6_ebov_snp_11_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_11_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_07_ms/\n",
            "kallisto_alignment/translated/6_ebov_snp_07_ms/bustools_count/\n",
            "kallisto_alignment/translated/6_ebov_snp_07_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/6_ebov_snp_07_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_07_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_1_ms/\n",
            "kallisto_alignment/translated/5_ebov_snp_1_ms/bustools_count/\n",
            "kallisto_alignment/translated/5_ebov_snp_1_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/5_ebov_snp_1_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_1_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_28_ms/\n",
            "kallisto_alignment/translated/8_ebov_snp_28_ms/bustools_count/\n",
            "kallisto_alignment/translated/8_ebov_snp_28_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/8_ebov_snp_28_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_28_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_06_ms/\n",
            "kallisto_alignment/translated/3_ebov_snp_06_ms/bustools_count/\n",
            "kallisto_alignment/translated/3_ebov_snp_06_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/3_ebov_snp_06_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_06_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_15_ms/\n",
            "kallisto_alignment/translated/4_ebov_snp_15_ms/bustools_count/\n",
            "kallisto_alignment/translated/4_ebov_snp_15_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/4_ebov_snp_15_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_15_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_26_ms/\n",
            "kallisto_alignment/translated/9_ebov_snp_26_ms/bustools_count/\n",
            "kallisto_alignment/translated/9_ebov_snp_26_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/9_ebov_snp_26_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_26_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_19_ms/\n",
            "kallisto_alignment/translated/8_ebov_snp_19_ms/bustools_count/\n",
            "kallisto_alignment/translated/8_ebov_snp_19_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/8_ebov_snp_19_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_19_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_26_ms/\n",
            "kallisto_alignment/translated/1_ebov_snp_26_ms/bustools_count/\n",
            "kallisto_alignment/translated/1_ebov_snp_26_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/1_ebov_snp_26_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_26_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_14_ms/\n",
            "kallisto_alignment/translated/4_ebov_snp_14_ms/bustools_count/\n",
            "kallisto_alignment/translated/4_ebov_snp_14_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/4_ebov_snp_14_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_14_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_13_ms/\n",
            "kallisto_alignment/translated/2_ebov_snp_13_ms/bustools_count/\n",
            "kallisto_alignment/translated/2_ebov_snp_13_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/2_ebov_snp_13_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_13_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_3_ms/\n",
            "kallisto_alignment/translated/0_ebov_snp_3_ms/bustools_count/\n",
            "kallisto_alignment/translated/0_ebov_snp_3_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/0_ebov_snp_3_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_3_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_15_ms/\n",
            "kallisto_alignment/translated/5_ebov_snp_15_ms/bustools_count/\n",
            "kallisto_alignment/translated/5_ebov_snp_15_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/5_ebov_snp_15_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_15_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_18_ms/\n",
            "kallisto_alignment/translated/3_ebov_snp_18_ms/bustools_count/\n",
            "kallisto_alignment/translated/3_ebov_snp_18_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/3_ebov_snp_18_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_18_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_24_ms/\n",
            "kallisto_alignment/translated/1_ebov_snp_24_ms/bustools_count/\n",
            "kallisto_alignment/translated/1_ebov_snp_24_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/1_ebov_snp_24_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_24_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_27_ms/\n",
            "kallisto_alignment/translated/3_ebov_snp_27_ms/bustools_count/\n",
            "kallisto_alignment/translated/3_ebov_snp_27_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/3_ebov_snp_27_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_27_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_19_ms/\n",
            "kallisto_alignment/translated/6_ebov_snp_19_ms/bustools_count/\n",
            "kallisto_alignment/translated/6_ebov_snp_19_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/6_ebov_snp_19_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_19_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_3_ms/\n",
            "kallisto_alignment/translated/5_ebov_snp_3_ms/bustools_count/\n",
            "kallisto_alignment/translated/5_ebov_snp_3_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/5_ebov_snp_3_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_3_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_13_ms/\n",
            "kallisto_alignment/translated/6_ebov_snp_13_ms/bustools_count/\n",
            "kallisto_alignment/translated/6_ebov_snp_13_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/6_ebov_snp_13_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_13_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_22_ms/\n",
            "kallisto_alignment/translated/3_ebov_snp_22_ms/bustools_count/\n",
            "kallisto_alignment/translated/3_ebov_snp_22_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/3_ebov_snp_22_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_22_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_1_ms/\n",
            "kallisto_alignment/translated/3_ebov_snp_1_ms/bustools_count/\n",
            "kallisto_alignment/translated/3_ebov_snp_1_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/3_ebov_snp_1_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_1_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_02_ms/\n",
            "kallisto_alignment/translated/7_ebov_snp_02_ms/bustools_count/\n",
            "kallisto_alignment/translated/7_ebov_snp_02_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/7_ebov_snp_02_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_02_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_27_ms/\n",
            "kallisto_alignment/translated/0_ebov_snp_27_ms/bustools_count/\n",
            "kallisto_alignment/translated/0_ebov_snp_27_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/0_ebov_snp_27_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_27_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_21_ms/\n",
            "kallisto_alignment/translated/2_ebov_snp_21_ms/bustools_count/\n",
            "kallisto_alignment/translated/2_ebov_snp_21_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/2_ebov_snp_21_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_21_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_12_ms/\n",
            "kallisto_alignment/translated/9_ebov_snp_12_ms/bustools_count/\n",
            "kallisto_alignment/translated/9_ebov_snp_12_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/9_ebov_snp_12_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_12_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_18_ms/\n",
            "kallisto_alignment/translated/4_ebov_snp_18_ms/bustools_count/\n",
            "kallisto_alignment/translated/4_ebov_snp_18_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/4_ebov_snp_18_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_18_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_16_ms/\n",
            "kallisto_alignment/translated/9_ebov_snp_16_ms/bustools_count/\n",
            "kallisto_alignment/translated/9_ebov_snp_16_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/9_ebov_snp_16_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_16_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_06_ms/\n",
            "kallisto_alignment/translated/7_ebov_snp_06_ms/bustools_count/\n",
            "kallisto_alignment/translated/7_ebov_snp_06_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/7_ebov_snp_06_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_06_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_22_ms/\n",
            "kallisto_alignment/translated/7_ebov_snp_22_ms/bustools_count/\n",
            "kallisto_alignment/translated/7_ebov_snp_22_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/7_ebov_snp_22_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_22_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_27_ms/\n",
            "kallisto_alignment/translated/2_ebov_snp_27_ms/bustools_count/\n",
            "kallisto_alignment/translated/2_ebov_snp_27_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/2_ebov_snp_27_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_27_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_14_ms/\n",
            "kallisto_alignment/translated/5_ebov_snp_14_ms/bustools_count/\n",
            "kallisto_alignment/translated/5_ebov_snp_14_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/5_ebov_snp_14_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_14_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_2_ms/\n",
            "kallisto_alignment/translated/9_ebov_snp_2_ms/bustools_count/\n",
            "kallisto_alignment/translated/9_ebov_snp_2_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/9_ebov_snp_2_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_2_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_07_ms/\n",
            "kallisto_alignment/translated/4_ebov_snp_07_ms/bustools_count/\n",
            "kallisto_alignment/translated/4_ebov_snp_07_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/4_ebov_snp_07_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_07_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_14_ms/\n",
            "kallisto_alignment/translated/2_ebov_snp_14_ms/bustools_count/\n",
            "kallisto_alignment/translated/2_ebov_snp_14_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/2_ebov_snp_14_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_14_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_1_ms/\n",
            "kallisto_alignment/translated/9_ebov_snp_1_ms/bustools_count/\n",
            "kallisto_alignment/translated/9_ebov_snp_1_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/9_ebov_snp_1_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_1_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_27_ms/\n",
            "kallisto_alignment/translated/7_ebov_snp_27_ms/bustools_count/\n",
            "kallisto_alignment/translated/7_ebov_snp_27_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/7_ebov_snp_27_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_27_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_16_ms/\n",
            "kallisto_alignment/translated/7_ebov_snp_16_ms/bustools_count/\n",
            "kallisto_alignment/translated/7_ebov_snp_16_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/7_ebov_snp_16_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_16_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_3_ms/\n",
            "kallisto_alignment/translated/2_ebov_snp_3_ms/bustools_count/\n",
            "kallisto_alignment/translated/2_ebov_snp_3_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/2_ebov_snp_3_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_3_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_13_ms/\n",
            "kallisto_alignment/translated/5_ebov_snp_13_ms/bustools_count/\n",
            "kallisto_alignment/translated/5_ebov_snp_13_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/5_ebov_snp_13_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_13_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_04_ms/\n",
            "kallisto_alignment/translated/4_ebov_snp_04_ms/bustools_count/\n",
            "kallisto_alignment/translated/4_ebov_snp_04_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/4_ebov_snp_04_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_04_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_13_ms/\n",
            "kallisto_alignment/translated/3_ebov_snp_13_ms/bustools_count/\n",
            "kallisto_alignment/translated/3_ebov_snp_13_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/3_ebov_snp_13_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_13_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_04_ms/\n",
            "kallisto_alignment/translated/2_ebov_snp_04_ms/bustools_count/\n",
            "kallisto_alignment/translated/2_ebov_snp_04_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/2_ebov_snp_04_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_04_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_12_ms/\n",
            "kallisto_alignment/translated/5_ebov_snp_12_ms/bustools_count/\n",
            "kallisto_alignment/translated/5_ebov_snp_12_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/5_ebov_snp_12_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_12_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_08_ms/\n",
            "kallisto_alignment/translated/0_ebov_snp_08_ms/bustools_count/\n",
            "kallisto_alignment/translated/0_ebov_snp_08_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/0_ebov_snp_08_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_08_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_22_ms/\n",
            "kallisto_alignment/translated/9_ebov_snp_22_ms/bustools_count/\n",
            "kallisto_alignment/translated/9_ebov_snp_22_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/9_ebov_snp_22_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_22_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_16_ms/\n",
            "kallisto_alignment/translated/3_ebov_snp_16_ms/bustools_count/\n",
            "kallisto_alignment/translated/3_ebov_snp_16_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/3_ebov_snp_16_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_16_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_24_ms/\n",
            "kallisto_alignment/translated/9_ebov_snp_24_ms/bustools_count/\n",
            "kallisto_alignment/translated/9_ebov_snp_24_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/9_ebov_snp_24_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_24_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_07_ms/\n",
            "kallisto_alignment/translated/9_ebov_snp_07_ms/bustools_count/\n",
            "kallisto_alignment/translated/9_ebov_snp_07_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/9_ebov_snp_07_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_07_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_05_ms/\n",
            "kallisto_alignment/translated/0_ebov_snp_05_ms/bustools_count/\n",
            "kallisto_alignment/translated/0_ebov_snp_05_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/0_ebov_snp_05_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_05_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_17_ms/\n",
            "kallisto_alignment/translated/2_ebov_snp_17_ms/bustools_count/\n",
            "kallisto_alignment/translated/2_ebov_snp_17_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/2_ebov_snp_17_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_17_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_19_ms/\n",
            "kallisto_alignment/translated/3_ebov_snp_19_ms/bustools_count/\n",
            "kallisto_alignment/translated/3_ebov_snp_19_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/3_ebov_snp_19_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_19_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_27_ms/\n",
            "kallisto_alignment/translated/6_ebov_snp_27_ms/bustools_count/\n",
            "kallisto_alignment/translated/6_ebov_snp_27_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/6_ebov_snp_27_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_27_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_27_ms/\n",
            "kallisto_alignment/translated/5_ebov_snp_27_ms/bustools_count/\n",
            "kallisto_alignment/translated/5_ebov_snp_27_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/5_ebov_snp_27_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_27_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_18_ms/\n",
            "kallisto_alignment/translated/6_ebov_snp_18_ms/bustools_count/\n",
            "kallisto_alignment/translated/6_ebov_snp_18_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/6_ebov_snp_18_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_18_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_16_ms/\n",
            "kallisto_alignment/translated/0_ebov_snp_16_ms/bustools_count/\n",
            "kallisto_alignment/translated/0_ebov_snp_16_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/0_ebov_snp_16_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_16_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_17_ms/\n",
            "kallisto_alignment/translated/4_ebov_snp_17_ms/bustools_count/\n",
            "kallisto_alignment/translated/4_ebov_snp_17_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/4_ebov_snp_17_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_17_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_06_ms/\n",
            "kallisto_alignment/translated/6_ebov_snp_06_ms/bustools_count/\n",
            "kallisto_alignment/translated/6_ebov_snp_06_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/6_ebov_snp_06_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_06_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_22_ms/\n",
            "kallisto_alignment/translated/6_ebov_snp_22_ms/bustools_count/\n",
            "kallisto_alignment/translated/6_ebov_snp_22_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/6_ebov_snp_22_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_22_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_24_ms/\n",
            "kallisto_alignment/translated/0_ebov_snp_24_ms/bustools_count/\n",
            "kallisto_alignment/translated/0_ebov_snp_24_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/0_ebov_snp_24_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_24_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_29_ms/\n",
            "kallisto_alignment/translated/5_ebov_snp_29_ms/bustools_count/\n",
            "kallisto_alignment/translated/5_ebov_snp_29_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/5_ebov_snp_29_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_29_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_25_ms/\n",
            "kallisto_alignment/translated/1_ebov_snp_25_ms/bustools_count/\n",
            "kallisto_alignment/translated/1_ebov_snp_25_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/1_ebov_snp_25_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_25_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_02_ms/\n",
            "kallisto_alignment/translated/1_ebov_snp_02_ms/bustools_count/\n",
            "kallisto_alignment/translated/1_ebov_snp_02_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/1_ebov_snp_02_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_02_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_26_ms/\n",
            "kallisto_alignment/translated/4_ebov_snp_26_ms/bustools_count/\n",
            "kallisto_alignment/translated/4_ebov_snp_26_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/4_ebov_snp_26_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_26_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_02_ms/\n",
            "kallisto_alignment/translated/9_ebov_snp_02_ms/bustools_count/\n",
            "kallisto_alignment/translated/9_ebov_snp_02_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/9_ebov_snp_02_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_02_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_16_ms/\n",
            "kallisto_alignment/translated/8_ebov_snp_16_ms/bustools_count/\n",
            "kallisto_alignment/translated/8_ebov_snp_16_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/8_ebov_snp_16_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_16_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_17_ms/\n",
            "kallisto_alignment/translated/7_ebov_snp_17_ms/bustools_count/\n",
            "kallisto_alignment/translated/7_ebov_snp_17_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/7_ebov_snp_17_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_17_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_3_ms/\n",
            "kallisto_alignment/translated/4_ebov_snp_3_ms/bustools_count/\n",
            "kallisto_alignment/translated/4_ebov_snp_3_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/4_ebov_snp_3_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_3_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_16_ms/\n",
            "kallisto_alignment/translated/5_ebov_snp_16_ms/bustools_count/\n",
            "kallisto_alignment/translated/5_ebov_snp_16_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/5_ebov_snp_16_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_16_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_27_ms/\n",
            "kallisto_alignment/translated/4_ebov_snp_27_ms/bustools_count/\n",
            "kallisto_alignment/translated/4_ebov_snp_27_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/4_ebov_snp_27_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_27_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_04_ms/\n",
            "kallisto_alignment/translated/9_ebov_snp_04_ms/bustools_count/\n",
            "kallisto_alignment/translated/9_ebov_snp_04_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/9_ebov_snp_04_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_04_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_12_ms/\n",
            "kallisto_alignment/translated/0_ebov_snp_12_ms/bustools_count/\n",
            "kallisto_alignment/translated/0_ebov_snp_12_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/0_ebov_snp_12_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_12_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_23_ms/\n",
            "kallisto_alignment/translated/8_ebov_snp_23_ms/bustools_count/\n",
            "kallisto_alignment/translated/8_ebov_snp_23_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/8_ebov_snp_23_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_23_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_1_ms/\n",
            "kallisto_alignment/translated/0_ebov_snp_1_ms/bustools_count/\n",
            "kallisto_alignment/translated/0_ebov_snp_1_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/0_ebov_snp_1_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_1_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_16_ms/\n",
            "kallisto_alignment/translated/4_ebov_snp_16_ms/bustools_count/\n",
            "kallisto_alignment/translated/4_ebov_snp_16_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/4_ebov_snp_16_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_16_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_13_ms/\n",
            "kallisto_alignment/translated/4_ebov_snp_13_ms/bustools_count/\n",
            "kallisto_alignment/translated/4_ebov_snp_13_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/4_ebov_snp_13_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_13_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_07_ms/\n",
            "kallisto_alignment/translated/0_ebov_snp_07_ms/bustools_count/\n",
            "kallisto_alignment/translated/0_ebov_snp_07_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/0_ebov_snp_07_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_07_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_26_ms/\n",
            "kallisto_alignment/translated/0_ebov_snp_26_ms/bustools_count/\n",
            "kallisto_alignment/translated/0_ebov_snp_26_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/0_ebov_snp_26_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_26_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_21_ms/\n",
            "kallisto_alignment/translated/8_ebov_snp_21_ms/bustools_count/\n",
            "kallisto_alignment/translated/8_ebov_snp_21_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/8_ebov_snp_21_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_21_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_11_ms/\n",
            "kallisto_alignment/translated/0_ebov_snp_11_ms/bustools_count/\n",
            "kallisto_alignment/translated/0_ebov_snp_11_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/0_ebov_snp_11_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_11_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_11_ms/\n",
            "kallisto_alignment/translated/7_ebov_snp_11_ms/bustools_count/\n",
            "kallisto_alignment/translated/7_ebov_snp_11_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/7_ebov_snp_11_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_11_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_07_ms/\n",
            "kallisto_alignment/translated/2_ebov_snp_07_ms/bustools_count/\n",
            "kallisto_alignment/translated/2_ebov_snp_07_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/2_ebov_snp_07_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_07_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_21_ms/\n",
            "kallisto_alignment/translated/6_ebov_snp_21_ms/bustools_count/\n",
            "kallisto_alignment/translated/6_ebov_snp_21_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/6_ebov_snp_21_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_21_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_17_ms/\n",
            "kallisto_alignment/translated/6_ebov_snp_17_ms/bustools_count/\n",
            "kallisto_alignment/translated/6_ebov_snp_17_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/6_ebov_snp_17_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_17_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_29_ms/\n",
            "kallisto_alignment/translated/2_ebov_snp_29_ms/bustools_count/\n",
            "kallisto_alignment/translated/2_ebov_snp_29_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/2_ebov_snp_29_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_29_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_06_ms/\n",
            "kallisto_alignment/translated/9_ebov_snp_06_ms/bustools_count/\n",
            "kallisto_alignment/translated/9_ebov_snp_06_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/9_ebov_snp_06_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_06_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_25_ms/\n",
            "kallisto_alignment/translated/3_ebov_snp_25_ms/bustools_count/\n",
            "kallisto_alignment/translated/3_ebov_snp_25_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/3_ebov_snp_25_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_25_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_25_ms/\n",
            "kallisto_alignment/translated/0_ebov_snp_25_ms/bustools_count/\n",
            "kallisto_alignment/translated/0_ebov_snp_25_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/0_ebov_snp_25_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_25_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_27_ms/\n",
            "kallisto_alignment/translated/9_ebov_snp_27_ms/bustools_count/\n",
            "kallisto_alignment/translated/9_ebov_snp_27_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/9_ebov_snp_27_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_27_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_17_ms/\n",
            "kallisto_alignment/translated/8_ebov_snp_17_ms/bustools_count/\n",
            "kallisto_alignment/translated/8_ebov_snp_17_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/8_ebov_snp_17_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_17_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_15_ms/\n",
            "kallisto_alignment/translated/0_ebov_snp_15_ms/bustools_count/\n",
            "kallisto_alignment/translated/0_ebov_snp_15_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/0_ebov_snp_15_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_15_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_03_ms/\n",
            "kallisto_alignment/translated/1_ebov_snp_03_ms/bustools_count/\n",
            "kallisto_alignment/translated/1_ebov_snp_03_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/1_ebov_snp_03_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_03_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_28_ms/\n",
            "kallisto_alignment/translated/6_ebov_snp_28_ms/bustools_count/\n",
            "kallisto_alignment/translated/6_ebov_snp_28_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/6_ebov_snp_28_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_28_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_1_ms/\n",
            "kallisto_alignment/translated/2_ebov_snp_1_ms/bustools_count/\n",
            "kallisto_alignment/translated/2_ebov_snp_1_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/2_ebov_snp_1_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_1_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_17_ms/\n",
            "kallisto_alignment/translated/9_ebov_snp_17_ms/bustools_count/\n",
            "kallisto_alignment/translated/9_ebov_snp_17_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/9_ebov_snp_17_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_17_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_26_ms/\n",
            "kallisto_alignment/translated/5_ebov_snp_26_ms/bustools_count/\n",
            "kallisto_alignment/translated/5_ebov_snp_26_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/5_ebov_snp_26_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_26_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_23_ms/\n",
            "kallisto_alignment/translated/7_ebov_snp_23_ms/bustools_count/\n",
            "kallisto_alignment/translated/7_ebov_snp_23_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/7_ebov_snp_23_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_23_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_26_ms/\n",
            "kallisto_alignment/translated/3_ebov_snp_26_ms/bustools_count/\n",
            "kallisto_alignment/translated/3_ebov_snp_26_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/3_ebov_snp_26_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_26_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_05_ms/\n",
            "kallisto_alignment/translated/3_ebov_snp_05_ms/bustools_count/\n",
            "kallisto_alignment/translated/3_ebov_snp_05_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/3_ebov_snp_05_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_05_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_22_ms/\n",
            "kallisto_alignment/translated/5_ebov_snp_22_ms/bustools_count/\n",
            "kallisto_alignment/translated/5_ebov_snp_22_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/5_ebov_snp_22_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_22_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_15_ms/\n",
            "kallisto_alignment/translated/9_ebov_snp_15_ms/bustools_count/\n",
            "kallisto_alignment/translated/9_ebov_snp_15_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/9_ebov_snp_15_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_15_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_18_ms/\n",
            "kallisto_alignment/translated/8_ebov_snp_18_ms/bustools_count/\n",
            "kallisto_alignment/translated/8_ebov_snp_18_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/8_ebov_snp_18_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_18_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_23_ms/\n",
            "kallisto_alignment/translated/3_ebov_snp_23_ms/bustools_count/\n",
            "kallisto_alignment/translated/3_ebov_snp_23_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/3_ebov_snp_23_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_23_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_05_ms/\n",
            "kallisto_alignment/translated/4_ebov_snp_05_ms/bustools_count/\n",
            "kallisto_alignment/translated/4_ebov_snp_05_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/4_ebov_snp_05_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_05_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_11_ms/\n",
            "kallisto_alignment/translated/9_ebov_snp_11_ms/bustools_count/\n",
            "kallisto_alignment/translated/9_ebov_snp_11_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/9_ebov_snp_11_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_11_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_09_ms/\n",
            "kallisto_alignment/translated/7_ebov_snp_09_ms/bustools_count/\n",
            "kallisto_alignment/translated/7_ebov_snp_09_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/7_ebov_snp_09_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_09_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_15_ms/\n",
            "kallisto_alignment/translated/6_ebov_snp_15_ms/bustools_count/\n",
            "kallisto_alignment/translated/6_ebov_snp_15_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/6_ebov_snp_15_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_15_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_03_ms/\n",
            "kallisto_alignment/translated/3_ebov_snp_03_ms/bustools_count/\n",
            "kallisto_alignment/translated/3_ebov_snp_03_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/3_ebov_snp_03_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/3_ebov_snp_03_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_12_ms/\n",
            "kallisto_alignment/translated/7_ebov_snp_12_ms/bustools_count/\n",
            "kallisto_alignment/translated/7_ebov_snp_12_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/7_ebov_snp_12_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_12_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_25_ms/\n",
            "kallisto_alignment/translated/8_ebov_snp_25_ms/bustools_count/\n",
            "kallisto_alignment/translated/8_ebov_snp_25_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/8_ebov_snp_25_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_25_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_03_ms/\n",
            "kallisto_alignment/translated/9_ebov_snp_03_ms/bustools_count/\n",
            "kallisto_alignment/translated/9_ebov_snp_03_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/9_ebov_snp_03_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_03_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_2_ms/\n",
            "kallisto_alignment/translated/8_ebov_snp_2_ms/bustools_count/\n",
            "kallisto_alignment/translated/8_ebov_snp_2_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/8_ebov_snp_2_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_2_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_03_ms/\n",
            "kallisto_alignment/translated/0_ebov_snp_03_ms/bustools_count/\n",
            "kallisto_alignment/translated/0_ebov_snp_03_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/0_ebov_snp_03_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_03_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_28_ms/\n",
            "kallisto_alignment/translated/1_ebov_snp_28_ms/bustools_count/\n",
            "kallisto_alignment/translated/1_ebov_snp_28_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/1_ebov_snp_28_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_28_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_26_ms/\n",
            "kallisto_alignment/translated/8_ebov_snp_26_ms/bustools_count/\n",
            "kallisto_alignment/translated/8_ebov_snp_26_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/8_ebov_snp_26_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_26_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_14_ms/\n",
            "kallisto_alignment/translated/6_ebov_snp_14_ms/bustools_count/\n",
            "kallisto_alignment/translated/6_ebov_snp_14_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/6_ebov_snp_14_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_14_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_08_ms/\n",
            "kallisto_alignment/translated/7_ebov_snp_08_ms/bustools_count/\n",
            "kallisto_alignment/translated/7_ebov_snp_08_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/7_ebov_snp_08_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_08_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_03_ms/\n",
            "kallisto_alignment/translated/5_ebov_snp_03_ms/bustools_count/\n",
            "kallisto_alignment/translated/5_ebov_snp_03_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/5_ebov_snp_03_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_03_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_29_ms/\n",
            "kallisto_alignment/translated/4_ebov_snp_29_ms/bustools_count/\n",
            "kallisto_alignment/translated/4_ebov_snp_29_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/4_ebov_snp_29_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_29_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_13_ms/\n",
            "kallisto_alignment/translated/9_ebov_snp_13_ms/bustools_count/\n",
            "kallisto_alignment/translated/9_ebov_snp_13_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/9_ebov_snp_13_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_13_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_3_ms/\n",
            "kallisto_alignment/translated/1_ebov_snp_3_ms/bustools_count/\n",
            "kallisto_alignment/translated/1_ebov_snp_3_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/1_ebov_snp_3_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_3_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_2_ms/\n",
            "kallisto_alignment/translated/2_ebov_snp_2_ms/bustools_count/\n",
            "kallisto_alignment/translated/2_ebov_snp_2_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/2_ebov_snp_2_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_2_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_25_ms/\n",
            "kallisto_alignment/translated/9_ebov_snp_25_ms/bustools_count/\n",
            "kallisto_alignment/translated/9_ebov_snp_25_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/9_ebov_snp_25_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_25_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_28_ms/\n",
            "kallisto_alignment/translated/0_ebov_snp_28_ms/bustools_count/\n",
            "kallisto_alignment/translated/0_ebov_snp_28_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/0_ebov_snp_28_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_28_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_17_ms/\n",
            "kallisto_alignment/translated/0_ebov_snp_17_ms/bustools_count/\n",
            "kallisto_alignment/translated/0_ebov_snp_17_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/0_ebov_snp_17_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_17_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_23_ms/\n",
            "kallisto_alignment/translated/6_ebov_snp_23_ms/bustools_count/\n",
            "kallisto_alignment/translated/6_ebov_snp_23_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/6_ebov_snp_23_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/6_ebov_snp_23_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_07_ms/\n",
            "kallisto_alignment/translated/5_ebov_snp_07_ms/bustools_count/\n",
            "kallisto_alignment/translated/5_ebov_snp_07_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/5_ebov_snp_07_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_07_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_09_ms/\n",
            "kallisto_alignment/translated/9_ebov_snp_09_ms/bustools_count/\n",
            "kallisto_alignment/translated/9_ebov_snp_09_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/9_ebov_snp_09_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_09_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_04_ms/\n",
            "kallisto_alignment/translated/7_ebov_snp_04_ms/bustools_count/\n",
            "kallisto_alignment/translated/7_ebov_snp_04_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/7_ebov_snp_04_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/7_ebov_snp_04_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_11_ms/\n",
            "kallisto_alignment/translated/1_ebov_snp_11_ms/bustools_count/\n",
            "kallisto_alignment/translated/1_ebov_snp_11_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/1_ebov_snp_11_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_11_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_29_ms/\n",
            "kallisto_alignment/translated/0_ebov_snp_29_ms/bustools_count/\n",
            "kallisto_alignment/translated/0_ebov_snp_29_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/0_ebov_snp_29_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/0_ebov_snp_29_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_1_ms/\n",
            "kallisto_alignment/translated/4_ebov_snp_1_ms/bustools_count/\n",
            "kallisto_alignment/translated/4_ebov_snp_1_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/4_ebov_snp_1_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_1_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_21_ms/\n",
            "kallisto_alignment/translated/9_ebov_snp_21_ms/bustools_count/\n",
            "kallisto_alignment/translated/9_ebov_snp_21_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/9_ebov_snp_21_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/9_ebov_snp_21_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_26_ms/\n",
            "kallisto_alignment/translated/2_ebov_snp_26_ms/bustools_count/\n",
            "kallisto_alignment/translated/2_ebov_snp_26_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/2_ebov_snp_26_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/2_ebov_snp_26_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_24_ms/\n",
            "kallisto_alignment/translated/8_ebov_snp_24_ms/bustools_count/\n",
            "kallisto_alignment/translated/8_ebov_snp_24_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/8_ebov_snp_24_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/8_ebov_snp_24_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_19_ms/\n",
            "kallisto_alignment/translated/4_ebov_snp_19_ms/bustools_count/\n",
            "kallisto_alignment/translated/4_ebov_snp_19_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/4_ebov_snp_19_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/4_ebov_snp_19_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_1_ms/\n",
            "kallisto_alignment/translated/1_ebov_snp_1_ms/bustools_count/\n",
            "kallisto_alignment/translated/1_ebov_snp_1_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/1_ebov_snp_1_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/1_ebov_snp_1_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_05_ms/\n",
            "kallisto_alignment/translated/5_ebov_snp_05_ms/bustools_count/\n",
            "kallisto_alignment/translated/5_ebov_snp_05_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/translated/5_ebov_snp_05_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/translated/5_ebov_snp_05_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/\n",
            "kallisto_alignment/standard/2_ebov_snp_03_ms/\n",
            "kallisto_alignment/standard/2_ebov_snp_03_ms/bustools_count/\n",
            "kallisto_alignment/standard/2_ebov_snp_03_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/2_ebov_snp_03_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_03_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_22_ms/\n",
            "kallisto_alignment/standard/1_ebov_snp_22_ms/bustools_count/\n",
            "kallisto_alignment/standard/1_ebov_snp_22_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/1_ebov_snp_22_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_22_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_09_ms/\n",
            "kallisto_alignment/standard/3_ebov_snp_09_ms/bustools_count/\n",
            "kallisto_alignment/standard/3_ebov_snp_09_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/3_ebov_snp_09_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_09_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_1_ms/\n",
            "kallisto_alignment/standard/8_ebov_snp_1_ms/bustools_count/\n",
            "kallisto_alignment/standard/8_ebov_snp_1_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/8_ebov_snp_1_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_1_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_11_ms/\n",
            "kallisto_alignment/standard/2_ebov_snp_11_ms/bustools_count/\n",
            "kallisto_alignment/standard/2_ebov_snp_11_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/2_ebov_snp_11_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_11_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_06_ms/\n",
            "kallisto_alignment/standard/0_ebov_snp_06_ms/bustools_count/\n",
            "kallisto_alignment/standard/0_ebov_snp_06_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/0_ebov_snp_06_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_06_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_16_ms/\n",
            "kallisto_alignment/standard/2_ebov_snp_16_ms/bustools_count/\n",
            "kallisto_alignment/standard/2_ebov_snp_16_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/2_ebov_snp_16_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_16_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_02_ms/\n",
            "kallisto_alignment/standard/5_ebov_snp_02_ms/bustools_count/\n",
            "kallisto_alignment/standard/5_ebov_snp_02_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/5_ebov_snp_02_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_02_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_25_ms/\n",
            "kallisto_alignment/standard/4_ebov_snp_25_ms/bustools_count/\n",
            "kallisto_alignment/standard/4_ebov_snp_25_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/4_ebov_snp_25_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_25_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_2_ms/\n",
            "kallisto_alignment/standard/7_ebov_snp_2_ms/bustools_count/\n",
            "kallisto_alignment/standard/7_ebov_snp_2_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/7_ebov_snp_2_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_2_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_28_ms/\n",
            "kallisto_alignment/standard/9_ebov_snp_28_ms/bustools_count/\n",
            "kallisto_alignment/standard/9_ebov_snp_28_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/9_ebov_snp_28_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_28_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_06_ms/\n",
            "kallisto_alignment/standard/5_ebov_snp_06_ms/bustools_count/\n",
            "kallisto_alignment/standard/5_ebov_snp_06_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/5_ebov_snp_06_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_06_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_18_ms/\n",
            "kallisto_alignment/standard/1_ebov_snp_18_ms/bustools_count/\n",
            "kallisto_alignment/standard/1_ebov_snp_18_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/1_ebov_snp_18_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_18_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_05_ms/\n",
            "kallisto_alignment/standard/7_ebov_snp_05_ms/bustools_count/\n",
            "kallisto_alignment/standard/7_ebov_snp_05_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/7_ebov_snp_05_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_05_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_28_ms/\n",
            "kallisto_alignment/standard/3_ebov_snp_28_ms/bustools_count/\n",
            "kallisto_alignment/standard/3_ebov_snp_28_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/3_ebov_snp_28_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_28_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_21_ms/\n",
            "kallisto_alignment/standard/1_ebov_snp_21_ms/bustools_count/\n",
            "kallisto_alignment/standard/1_ebov_snp_21_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/1_ebov_snp_21_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_21_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_09_ms/\n",
            "kallisto_alignment/standard/8_ebov_snp_09_ms/bustools_count/\n",
            "kallisto_alignment/standard/8_ebov_snp_09_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/8_ebov_snp_09_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_09_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_19_ms/\n",
            "kallisto_alignment/standard/9_ebov_snp_19_ms/bustools_count/\n",
            "kallisto_alignment/standard/9_ebov_snp_19_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/9_ebov_snp_19_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_19_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_29_ms/\n",
            "kallisto_alignment/standard/9_ebov_snp_29_ms/bustools_count/\n",
            "kallisto_alignment/standard/9_ebov_snp_29_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/9_ebov_snp_29_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_29_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_2_ms/\n",
            "kallisto_alignment/standard/3_ebov_snp_2_ms/bustools_count/\n",
            "kallisto_alignment/standard/3_ebov_snp_2_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/3_ebov_snp_2_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_2_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_28_ms/\n",
            "kallisto_alignment/standard/4_ebov_snp_28_ms/bustools_count/\n",
            "kallisto_alignment/standard/4_ebov_snp_28_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/4_ebov_snp_28_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_28_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_11_ms/\n",
            "kallisto_alignment/standard/3_ebov_snp_11_ms/bustools_count/\n",
            "kallisto_alignment/standard/3_ebov_snp_11_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/3_ebov_snp_11_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_11_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_08_ms/\n",
            "kallisto_alignment/standard/6_ebov_snp_08_ms/bustools_count/\n",
            "kallisto_alignment/standard/6_ebov_snp_08_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/6_ebov_snp_08_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_08_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_18_ms/\n",
            "kallisto_alignment/standard/0_ebov_snp_18_ms/bustools_count/\n",
            "kallisto_alignment/standard/0_ebov_snp_18_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/0_ebov_snp_18_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_18_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_27_ms/\n",
            "kallisto_alignment/standard/8_ebov_snp_27_ms/bustools_count/\n",
            "kallisto_alignment/standard/8_ebov_snp_27_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/8_ebov_snp_27_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_27_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_04_ms/\n",
            "kallisto_alignment/standard/3_ebov_snp_04_ms/bustools_count/\n",
            "kallisto_alignment/standard/3_ebov_snp_04_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/3_ebov_snp_04_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_04_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_09_ms/\n",
            "kallisto_alignment/standard/4_ebov_snp_09_ms/bustools_count/\n",
            "kallisto_alignment/standard/4_ebov_snp_09_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/4_ebov_snp_09_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_09_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_08_ms/\n",
            "kallisto_alignment/standard/5_ebov_snp_08_ms/bustools_count/\n",
            "kallisto_alignment/standard/5_ebov_snp_08_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/5_ebov_snp_08_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_08_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_13_ms/\n",
            "kallisto_alignment/standard/0_ebov_snp_13_ms/bustools_count/\n",
            "kallisto_alignment/standard/0_ebov_snp_13_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/0_ebov_snp_13_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_13_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_1_ms/\n",
            "kallisto_alignment/standard/7_ebov_snp_1_ms/bustools_count/\n",
            "kallisto_alignment/standard/7_ebov_snp_1_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/7_ebov_snp_1_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_1_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_02_ms/\n",
            "kallisto_alignment/standard/3_ebov_snp_02_ms/bustools_count/\n",
            "kallisto_alignment/standard/3_ebov_snp_02_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/3_ebov_snp_02_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_02_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_17_ms/\n",
            "kallisto_alignment/standard/5_ebov_snp_17_ms/bustools_count/\n",
            "kallisto_alignment/standard/5_ebov_snp_17_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/5_ebov_snp_17_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_17_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_3_ms/\n",
            "kallisto_alignment/standard/6_ebov_snp_3_ms/bustools_count/\n",
            "kallisto_alignment/standard/6_ebov_snp_3_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/6_ebov_snp_3_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_3_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_2_ms/\n",
            "kallisto_alignment/standard/0_ebov_snp_2_ms/bustools_count/\n",
            "kallisto_alignment/standard/0_ebov_snp_2_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/0_ebov_snp_2_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_2_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_21_ms/\n",
            "kallisto_alignment/standard/4_ebov_snp_21_ms/bustools_count/\n",
            "kallisto_alignment/standard/4_ebov_snp_21_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/4_ebov_snp_21_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_21_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_13_ms/\n",
            "kallisto_alignment/standard/1_ebov_snp_13_ms/bustools_count/\n",
            "kallisto_alignment/standard/1_ebov_snp_13_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/1_ebov_snp_13_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_13_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_19_ms/\n",
            "kallisto_alignment/standard/1_ebov_snp_19_ms/bustools_count/\n",
            "kallisto_alignment/standard/1_ebov_snp_19_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/1_ebov_snp_19_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_19_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_12_ms/\n",
            "kallisto_alignment/standard/6_ebov_snp_12_ms/bustools_count/\n",
            "kallisto_alignment/standard/6_ebov_snp_12_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/6_ebov_snp_12_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_12_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_24_ms/\n",
            "kallisto_alignment/standard/6_ebov_snp_24_ms/bustools_count/\n",
            "kallisto_alignment/standard/6_ebov_snp_24_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/6_ebov_snp_24_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_24_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_29_ms/\n",
            "kallisto_alignment/standard/7_ebov_snp_29_ms/bustools_count/\n",
            "kallisto_alignment/standard/7_ebov_snp_29_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/7_ebov_snp_29_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_29_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_2_ms/\n",
            "kallisto_alignment/standard/1_ebov_snp_2_ms/bustools_count/\n",
            "kallisto_alignment/standard/1_ebov_snp_2_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/1_ebov_snp_2_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_2_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_22_ms/\n",
            "kallisto_alignment/standard/2_ebov_snp_22_ms/bustools_count/\n",
            "kallisto_alignment/standard/2_ebov_snp_22_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/2_ebov_snp_22_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_22_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_19_ms/\n",
            "kallisto_alignment/standard/5_ebov_snp_19_ms/bustools_count/\n",
            "kallisto_alignment/standard/5_ebov_snp_19_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/5_ebov_snp_19_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_19_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_14_ms/\n",
            "kallisto_alignment/standard/8_ebov_snp_14_ms/bustools_count/\n",
            "kallisto_alignment/standard/8_ebov_snp_14_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/8_ebov_snp_14_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_14_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_02_ms/\n",
            "kallisto_alignment/standard/6_ebov_snp_02_ms/bustools_count/\n",
            "kallisto_alignment/standard/6_ebov_snp_02_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/6_ebov_snp_02_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_02_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_09_ms/\n",
            "kallisto_alignment/standard/1_ebov_snp_09_ms/bustools_count/\n",
            "kallisto_alignment/standard/1_ebov_snp_09_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/1_ebov_snp_09_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_09_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_12_ms/\n",
            "kallisto_alignment/standard/3_ebov_snp_12_ms/bustools_count/\n",
            "kallisto_alignment/standard/3_ebov_snp_12_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/3_ebov_snp_12_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_12_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_08_ms/\n",
            "kallisto_alignment/standard/9_ebov_snp_08_ms/bustools_count/\n",
            "kallisto_alignment/standard/9_ebov_snp_08_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/9_ebov_snp_08_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_08_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_04_ms/\n",
            "kallisto_alignment/standard/0_ebov_snp_04_ms/bustools_count/\n",
            "kallisto_alignment/standard/0_ebov_snp_04_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/0_ebov_snp_04_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_04_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_05_ms/\n",
            "kallisto_alignment/standard/2_ebov_snp_05_ms/bustools_count/\n",
            "kallisto_alignment/standard/2_ebov_snp_05_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/2_ebov_snp_05_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_05_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_14_ms/\n",
            "kallisto_alignment/standard/1_ebov_snp_14_ms/bustools_count/\n",
            "kallisto_alignment/standard/1_ebov_snp_14_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/1_ebov_snp_14_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_14_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_19_ms/\n",
            "kallisto_alignment/standard/0_ebov_snp_19_ms/bustools_count/\n",
            "kallisto_alignment/standard/0_ebov_snp_19_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/0_ebov_snp_19_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_19_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_29_ms/\n",
            "kallisto_alignment/standard/1_ebov_snp_29_ms/bustools_count/\n",
            "kallisto_alignment/standard/1_ebov_snp_29_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/1_ebov_snp_29_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_29_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_14_ms/\n",
            "kallisto_alignment/standard/3_ebov_snp_14_ms/bustools_count/\n",
            "kallisto_alignment/standard/3_ebov_snp_14_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/3_ebov_snp_14_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_14_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_02_ms/\n",
            "kallisto_alignment/standard/8_ebov_snp_02_ms/bustools_count/\n",
            "kallisto_alignment/standard/8_ebov_snp_02_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/8_ebov_snp_02_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_02_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_11_ms/\n",
            "kallisto_alignment/standard/8_ebov_snp_11_ms/bustools_count/\n",
            "kallisto_alignment/standard/8_ebov_snp_11_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/8_ebov_snp_11_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_11_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_13_ms/\n",
            "kallisto_alignment/standard/8_ebov_snp_13_ms/bustools_count/\n",
            "kallisto_alignment/standard/8_ebov_snp_13_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/8_ebov_snp_13_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_13_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_28_ms/\n",
            "kallisto_alignment/standard/5_ebov_snp_28_ms/bustools_count/\n",
            "kallisto_alignment/standard/5_ebov_snp_28_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/5_ebov_snp_28_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_28_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_09_ms/\n",
            "kallisto_alignment/standard/5_ebov_snp_09_ms/bustools_count/\n",
            "kallisto_alignment/standard/5_ebov_snp_09_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/5_ebov_snp_09_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_09_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_23_ms/\n",
            "kallisto_alignment/standard/2_ebov_snp_23_ms/bustools_count/\n",
            "kallisto_alignment/standard/2_ebov_snp_23_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/2_ebov_snp_23_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_23_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_27_ms/\n",
            "kallisto_alignment/standard/1_ebov_snp_27_ms/bustools_count/\n",
            "kallisto_alignment/standard/1_ebov_snp_27_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/1_ebov_snp_27_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_27_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_04_ms/\n",
            "kallisto_alignment/standard/8_ebov_snp_04_ms/bustools_count/\n",
            "kallisto_alignment/standard/8_ebov_snp_04_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/8_ebov_snp_04_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_04_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_24_ms/\n",
            "kallisto_alignment/standard/2_ebov_snp_24_ms/bustools_count/\n",
            "kallisto_alignment/standard/2_ebov_snp_24_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/2_ebov_snp_24_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_24_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_08_ms/\n",
            "kallisto_alignment/standard/4_ebov_snp_08_ms/bustools_count/\n",
            "kallisto_alignment/standard/4_ebov_snp_08_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/4_ebov_snp_08_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_08_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_3_ms/\n",
            "kallisto_alignment/standard/9_ebov_snp_3_ms/bustools_count/\n",
            "kallisto_alignment/standard/9_ebov_snp_3_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/9_ebov_snp_3_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_3_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_22_ms/\n",
            "kallisto_alignment/standard/4_ebov_snp_22_ms/bustools_count/\n",
            "kallisto_alignment/standard/4_ebov_snp_22_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/4_ebov_snp_22_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_22_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_29_ms/\n",
            "kallisto_alignment/standard/8_ebov_snp_29_ms/bustools_count/\n",
            "kallisto_alignment/standard/8_ebov_snp_29_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/8_ebov_snp_29_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_29_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_25_ms/\n",
            "kallisto_alignment/standard/2_ebov_snp_25_ms/bustools_count/\n",
            "kallisto_alignment/standard/2_ebov_snp_25_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/2_ebov_snp_25_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_25_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_24_ms/\n",
            "kallisto_alignment/standard/5_ebov_snp_24_ms/bustools_count/\n",
            "kallisto_alignment/standard/5_ebov_snp_24_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/5_ebov_snp_24_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_24_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_3_ms/\n",
            "kallisto_alignment/standard/8_ebov_snp_3_ms/bustools_count/\n",
            "kallisto_alignment/standard/8_ebov_snp_3_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/8_ebov_snp_3_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_3_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_14_ms/\n",
            "kallisto_alignment/standard/9_ebov_snp_14_ms/bustools_count/\n",
            "kallisto_alignment/standard/9_ebov_snp_14_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/9_ebov_snp_14_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_14_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_11_ms/\n",
            "kallisto_alignment/standard/5_ebov_snp_11_ms/bustools_count/\n",
            "kallisto_alignment/standard/5_ebov_snp_11_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/5_ebov_snp_11_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_11_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_3_ms/\n",
            "kallisto_alignment/standard/3_ebov_snp_3_ms/bustools_count/\n",
            "kallisto_alignment/standard/3_ebov_snp_3_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/3_ebov_snp_3_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_3_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_0_ms/\n",
            "kallisto_alignment/standard/0_ebov_snp_0_ms/bustools_count/\n",
            "kallisto_alignment/standard/0_ebov_snp_0_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/0_ebov_snp_0_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_0_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_24_ms/\n",
            "kallisto_alignment/standard/4_ebov_snp_24_ms/bustools_count/\n",
            "kallisto_alignment/standard/4_ebov_snp_24_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/4_ebov_snp_24_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_24_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_25_ms/\n",
            "kallisto_alignment/standard/6_ebov_snp_25_ms/bustools_count/\n",
            "kallisto_alignment/standard/6_ebov_snp_25_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/6_ebov_snp_25_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_25_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_11_ms/\n",
            "kallisto_alignment/standard/4_ebov_snp_11_ms/bustools_count/\n",
            "kallisto_alignment/standard/4_ebov_snp_11_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/4_ebov_snp_11_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_11_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_22_ms/\n",
            "kallisto_alignment/standard/0_ebov_snp_22_ms/bustools_count/\n",
            "kallisto_alignment/standard/0_ebov_snp_22_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/0_ebov_snp_22_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_22_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_12_ms/\n",
            "kallisto_alignment/standard/2_ebov_snp_12_ms/bustools_count/\n",
            "kallisto_alignment/standard/2_ebov_snp_12_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/2_ebov_snp_12_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_12_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_2_ms/\n",
            "kallisto_alignment/standard/6_ebov_snp_2_ms/bustools_count/\n",
            "kallisto_alignment/standard/6_ebov_snp_2_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/6_ebov_snp_2_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_2_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_02_ms/\n",
            "kallisto_alignment/standard/4_ebov_snp_02_ms/bustools_count/\n",
            "kallisto_alignment/standard/4_ebov_snp_02_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/4_ebov_snp_02_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_02_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_02_ms/\n",
            "kallisto_alignment/standard/0_ebov_snp_02_ms/bustools_count/\n",
            "kallisto_alignment/standard/0_ebov_snp_02_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/0_ebov_snp_02_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_02_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_28_ms/\n",
            "kallisto_alignment/standard/7_ebov_snp_28_ms/bustools_count/\n",
            "kallisto_alignment/standard/7_ebov_snp_28_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/7_ebov_snp_28_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_28_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_05_ms/\n",
            "kallisto_alignment/standard/6_ebov_snp_05_ms/bustools_count/\n",
            "kallisto_alignment/standard/6_ebov_snp_05_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/6_ebov_snp_05_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_05_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_03_ms/\n",
            "kallisto_alignment/standard/6_ebov_snp_03_ms/bustools_count/\n",
            "kallisto_alignment/standard/6_ebov_snp_03_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/6_ebov_snp_03_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_03_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_18_ms/\n",
            "kallisto_alignment/standard/5_ebov_snp_18_ms/bustools_count/\n",
            "kallisto_alignment/standard/5_ebov_snp_18_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/5_ebov_snp_18_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_18_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_15_ms/\n",
            "kallisto_alignment/standard/1_ebov_snp_15_ms/bustools_count/\n",
            "kallisto_alignment/standard/1_ebov_snp_15_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/1_ebov_snp_15_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_15_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_06_ms/\n",
            "kallisto_alignment/standard/2_ebov_snp_06_ms/bustools_count/\n",
            "kallisto_alignment/standard/2_ebov_snp_06_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/2_ebov_snp_06_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_06_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_29_ms/\n",
            "kallisto_alignment/standard/3_ebov_snp_29_ms/bustools_count/\n",
            "kallisto_alignment/standard/3_ebov_snp_29_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/3_ebov_snp_29_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_29_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_15_ms/\n",
            "kallisto_alignment/standard/7_ebov_snp_15_ms/bustools_count/\n",
            "kallisto_alignment/standard/7_ebov_snp_15_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/7_ebov_snp_15_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_15_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_12_ms/\n",
            "kallisto_alignment/standard/4_ebov_snp_12_ms/bustools_count/\n",
            "kallisto_alignment/standard/4_ebov_snp_12_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/4_ebov_snp_12_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_12_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_2_ms/\n",
            "kallisto_alignment/standard/4_ebov_snp_2_ms/bustools_count/\n",
            "kallisto_alignment/standard/4_ebov_snp_2_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/4_ebov_snp_2_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_2_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_19_ms/\n",
            "kallisto_alignment/standard/7_ebov_snp_19_ms/bustools_count/\n",
            "kallisto_alignment/standard/7_ebov_snp_19_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/7_ebov_snp_19_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_19_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_14_ms/\n",
            "kallisto_alignment/standard/0_ebov_snp_14_ms/bustools_count/\n",
            "kallisto_alignment/standard/0_ebov_snp_14_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/0_ebov_snp_14_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_14_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_08_ms/\n",
            "kallisto_alignment/standard/3_ebov_snp_08_ms/bustools_count/\n",
            "kallisto_alignment/standard/3_ebov_snp_08_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/3_ebov_snp_08_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_08_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_12_ms/\n",
            "kallisto_alignment/standard/1_ebov_snp_12_ms/bustools_count/\n",
            "kallisto_alignment/standard/1_ebov_snp_12_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/1_ebov_snp_12_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_12_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_15_ms/\n",
            "kallisto_alignment/standard/2_ebov_snp_15_ms/bustools_count/\n",
            "kallisto_alignment/standard/2_ebov_snp_15_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/2_ebov_snp_15_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_15_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_17_ms/\n",
            "kallisto_alignment/standard/3_ebov_snp_17_ms/bustools_count/\n",
            "kallisto_alignment/standard/3_ebov_snp_17_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/3_ebov_snp_17_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_17_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_06_ms/\n",
            "kallisto_alignment/standard/4_ebov_snp_06_ms/bustools_count/\n",
            "kallisto_alignment/standard/4_ebov_snp_06_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/4_ebov_snp_06_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_06_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_22_ms/\n",
            "kallisto_alignment/standard/8_ebov_snp_22_ms/bustools_count/\n",
            "kallisto_alignment/standard/8_ebov_snp_22_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/8_ebov_snp_22_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_22_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_21_ms/\n",
            "kallisto_alignment/standard/3_ebov_snp_21_ms/bustools_count/\n",
            "kallisto_alignment/standard/3_ebov_snp_21_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/3_ebov_snp_21_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_21_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_03_ms/\n",
            "kallisto_alignment/standard/4_ebov_snp_03_ms/bustools_count/\n",
            "kallisto_alignment/standard/4_ebov_snp_03_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/4_ebov_snp_03_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_03_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_25_ms/\n",
            "kallisto_alignment/standard/5_ebov_snp_25_ms/bustools_count/\n",
            "kallisto_alignment/standard/5_ebov_snp_25_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/5_ebov_snp_25_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_25_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_28_ms/\n",
            "kallisto_alignment/standard/2_ebov_snp_28_ms/bustools_count/\n",
            "kallisto_alignment/standard/2_ebov_snp_28_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/2_ebov_snp_28_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_28_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_23_ms/\n",
            "kallisto_alignment/standard/4_ebov_snp_23_ms/bustools_count/\n",
            "kallisto_alignment/standard/4_ebov_snp_23_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/4_ebov_snp_23_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_23_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_06_ms/\n",
            "kallisto_alignment/standard/1_ebov_snp_06_ms/bustools_count/\n",
            "kallisto_alignment/standard/1_ebov_snp_06_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/1_ebov_snp_06_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_06_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_07_ms/\n",
            "kallisto_alignment/standard/3_ebov_snp_07_ms/bustools_count/\n",
            "kallisto_alignment/standard/3_ebov_snp_07_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/3_ebov_snp_07_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_07_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_17_ms/\n",
            "kallisto_alignment/standard/1_ebov_snp_17_ms/bustools_count/\n",
            "kallisto_alignment/standard/1_ebov_snp_17_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/1_ebov_snp_17_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_17_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_16_ms/\n",
            "kallisto_alignment/standard/1_ebov_snp_16_ms/bustools_count/\n",
            "kallisto_alignment/standard/1_ebov_snp_16_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/1_ebov_snp_16_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_16_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_07_ms/\n",
            "kallisto_alignment/standard/8_ebov_snp_07_ms/bustools_count/\n",
            "kallisto_alignment/standard/8_ebov_snp_07_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/8_ebov_snp_07_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_07_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_25_ms/\n",
            "kallisto_alignment/standard/7_ebov_snp_25_ms/bustools_count/\n",
            "kallisto_alignment/standard/7_ebov_snp_25_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/7_ebov_snp_25_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_25_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_03_ms/\n",
            "kallisto_alignment/standard/7_ebov_snp_03_ms/bustools_count/\n",
            "kallisto_alignment/standard/7_ebov_snp_03_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/7_ebov_snp_03_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_03_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_04_ms/\n",
            "kallisto_alignment/standard/5_ebov_snp_04_ms/bustools_count/\n",
            "kallisto_alignment/standard/5_ebov_snp_04_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/5_ebov_snp_04_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_04_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_08_ms/\n",
            "kallisto_alignment/standard/1_ebov_snp_08_ms/bustools_count/\n",
            "kallisto_alignment/standard/1_ebov_snp_08_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/1_ebov_snp_08_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_08_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_02_ms/\n",
            "kallisto_alignment/standard/2_ebov_snp_02_ms/bustools_count/\n",
            "kallisto_alignment/standard/2_ebov_snp_02_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/2_ebov_snp_02_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_02_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_23_ms/\n",
            "kallisto_alignment/standard/0_ebov_snp_23_ms/bustools_count/\n",
            "kallisto_alignment/standard/0_ebov_snp_23_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/0_ebov_snp_23_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_23_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_08_ms/\n",
            "kallisto_alignment/standard/2_ebov_snp_08_ms/bustools_count/\n",
            "kallisto_alignment/standard/2_ebov_snp_08_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/2_ebov_snp_08_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_08_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_18_ms/\n",
            "kallisto_alignment/standard/2_ebov_snp_18_ms/bustools_count/\n",
            "kallisto_alignment/standard/2_ebov_snp_18_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/2_ebov_snp_18_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_18_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_08_ms/\n",
            "kallisto_alignment/standard/8_ebov_snp_08_ms/bustools_count/\n",
            "kallisto_alignment/standard/8_ebov_snp_08_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/8_ebov_snp_08_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_08_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_16_ms/\n",
            "kallisto_alignment/standard/6_ebov_snp_16_ms/bustools_count/\n",
            "kallisto_alignment/standard/6_ebov_snp_16_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/6_ebov_snp_16_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_16_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_05_ms/\n",
            "kallisto_alignment/standard/1_ebov_snp_05_ms/bustools_count/\n",
            "kallisto_alignment/standard/1_ebov_snp_05_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/1_ebov_snp_05_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_05_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_24_ms/\n",
            "kallisto_alignment/standard/3_ebov_snp_24_ms/bustools_count/\n",
            "kallisto_alignment/standard/3_ebov_snp_24_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/3_ebov_snp_24_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_24_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_23_ms/\n",
            "kallisto_alignment/standard/1_ebov_snp_23_ms/bustools_count/\n",
            "kallisto_alignment/standard/1_ebov_snp_23_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/1_ebov_snp_23_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_23_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_09_ms/\n",
            "kallisto_alignment/standard/0_ebov_snp_09_ms/bustools_count/\n",
            "kallisto_alignment/standard/0_ebov_snp_09_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/0_ebov_snp_09_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_09_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_3_ms/\n",
            "kallisto_alignment/standard/7_ebov_snp_3_ms/bustools_count/\n",
            "kallisto_alignment/standard/7_ebov_snp_3_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/7_ebov_snp_3_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_3_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_1_ms/\n",
            "kallisto_alignment/standard/6_ebov_snp_1_ms/bustools_count/\n",
            "kallisto_alignment/standard/6_ebov_snp_1_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/6_ebov_snp_1_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_1_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_21_ms/\n",
            "kallisto_alignment/standard/5_ebov_snp_21_ms/bustools_count/\n",
            "kallisto_alignment/standard/5_ebov_snp_21_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/5_ebov_snp_21_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_21_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_14_ms/\n",
            "kallisto_alignment/standard/7_ebov_snp_14_ms/bustools_count/\n",
            "kallisto_alignment/standard/7_ebov_snp_14_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/7_ebov_snp_14_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_14_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_12_ms/\n",
            "kallisto_alignment/standard/8_ebov_snp_12_ms/bustools_count/\n",
            "kallisto_alignment/standard/8_ebov_snp_12_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/8_ebov_snp_12_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_12_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_2_ms/\n",
            "kallisto_alignment/standard/5_ebov_snp_2_ms/bustools_count/\n",
            "kallisto_alignment/standard/5_ebov_snp_2_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/5_ebov_snp_2_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_2_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_21_ms/\n",
            "kallisto_alignment/standard/7_ebov_snp_21_ms/bustools_count/\n",
            "kallisto_alignment/standard/7_ebov_snp_21_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/7_ebov_snp_21_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_21_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_07_ms/\n",
            "kallisto_alignment/standard/7_ebov_snp_07_ms/bustools_count/\n",
            "kallisto_alignment/standard/7_ebov_snp_07_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/7_ebov_snp_07_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_07_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_19_ms/\n",
            "kallisto_alignment/standard/2_ebov_snp_19_ms/bustools_count/\n",
            "kallisto_alignment/standard/2_ebov_snp_19_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/2_ebov_snp_19_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_19_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_23_ms/\n",
            "kallisto_alignment/standard/5_ebov_snp_23_ms/bustools_count/\n",
            "kallisto_alignment/standard/5_ebov_snp_23_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/5_ebov_snp_23_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_23_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_15_ms/\n",
            "kallisto_alignment/standard/8_ebov_snp_15_ms/bustools_count/\n",
            "kallisto_alignment/standard/8_ebov_snp_15_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/8_ebov_snp_15_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_15_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_26_ms/\n",
            "kallisto_alignment/standard/6_ebov_snp_26_ms/bustools_count/\n",
            "kallisto_alignment/standard/6_ebov_snp_26_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/6_ebov_snp_26_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_26_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_05_ms/\n",
            "kallisto_alignment/standard/9_ebov_snp_05_ms/bustools_count/\n",
            "kallisto_alignment/standard/9_ebov_snp_05_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/9_ebov_snp_05_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_05_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_07_ms/\n",
            "kallisto_alignment/standard/1_ebov_snp_07_ms/bustools_count/\n",
            "kallisto_alignment/standard/1_ebov_snp_07_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/1_ebov_snp_07_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_07_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_05_ms/\n",
            "kallisto_alignment/standard/8_ebov_snp_05_ms/bustools_count/\n",
            "kallisto_alignment/standard/8_ebov_snp_05_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/8_ebov_snp_05_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_05_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_06_ms/\n",
            "kallisto_alignment/standard/8_ebov_snp_06_ms/bustools_count/\n",
            "kallisto_alignment/standard/8_ebov_snp_06_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/8_ebov_snp_06_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_06_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_04_ms/\n",
            "kallisto_alignment/standard/6_ebov_snp_04_ms/bustools_count/\n",
            "kallisto_alignment/standard/6_ebov_snp_04_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/6_ebov_snp_04_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_04_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_13_ms/\n",
            "kallisto_alignment/standard/7_ebov_snp_13_ms/bustools_count/\n",
            "kallisto_alignment/standard/7_ebov_snp_13_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/7_ebov_snp_13_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_13_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_21_ms/\n",
            "kallisto_alignment/standard/0_ebov_snp_21_ms/bustools_count/\n",
            "kallisto_alignment/standard/0_ebov_snp_21_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/0_ebov_snp_21_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_21_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_09_ms/\n",
            "kallisto_alignment/standard/6_ebov_snp_09_ms/bustools_count/\n",
            "kallisto_alignment/standard/6_ebov_snp_09_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/6_ebov_snp_09_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_09_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_04_ms/\n",
            "kallisto_alignment/standard/1_ebov_snp_04_ms/bustools_count/\n",
            "kallisto_alignment/standard/1_ebov_snp_04_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/1_ebov_snp_04_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_04_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_26_ms/\n",
            "kallisto_alignment/standard/7_ebov_snp_26_ms/bustools_count/\n",
            "kallisto_alignment/standard/7_ebov_snp_26_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/7_ebov_snp_26_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_26_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_29_ms/\n",
            "kallisto_alignment/standard/6_ebov_snp_29_ms/bustools_count/\n",
            "kallisto_alignment/standard/6_ebov_snp_29_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/6_ebov_snp_29_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_29_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_23_ms/\n",
            "kallisto_alignment/standard/9_ebov_snp_23_ms/bustools_count/\n",
            "kallisto_alignment/standard/9_ebov_snp_23_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/9_ebov_snp_23_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_23_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_18_ms/\n",
            "kallisto_alignment/standard/9_ebov_snp_18_ms/bustools_count/\n",
            "kallisto_alignment/standard/9_ebov_snp_18_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/9_ebov_snp_18_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_18_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_24_ms/\n",
            "kallisto_alignment/standard/7_ebov_snp_24_ms/bustools_count/\n",
            "kallisto_alignment/standard/7_ebov_snp_24_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/7_ebov_snp_24_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_24_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_03_ms/\n",
            "kallisto_alignment/standard/8_ebov_snp_03_ms/bustools_count/\n",
            "kallisto_alignment/standard/8_ebov_snp_03_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/8_ebov_snp_03_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_03_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_09_ms/\n",
            "kallisto_alignment/standard/2_ebov_snp_09_ms/bustools_count/\n",
            "kallisto_alignment/standard/2_ebov_snp_09_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/2_ebov_snp_09_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_09_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_18_ms/\n",
            "kallisto_alignment/standard/7_ebov_snp_18_ms/bustools_count/\n",
            "kallisto_alignment/standard/7_ebov_snp_18_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/7_ebov_snp_18_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_18_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_15_ms/\n",
            "kallisto_alignment/standard/3_ebov_snp_15_ms/bustools_count/\n",
            "kallisto_alignment/standard/3_ebov_snp_15_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/3_ebov_snp_15_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_15_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_11_ms/\n",
            "kallisto_alignment/standard/6_ebov_snp_11_ms/bustools_count/\n",
            "kallisto_alignment/standard/6_ebov_snp_11_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/6_ebov_snp_11_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_11_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_07_ms/\n",
            "kallisto_alignment/standard/6_ebov_snp_07_ms/bustools_count/\n",
            "kallisto_alignment/standard/6_ebov_snp_07_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/6_ebov_snp_07_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_07_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_1_ms/\n",
            "kallisto_alignment/standard/5_ebov_snp_1_ms/bustools_count/\n",
            "kallisto_alignment/standard/5_ebov_snp_1_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/5_ebov_snp_1_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_1_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_28_ms/\n",
            "kallisto_alignment/standard/8_ebov_snp_28_ms/bustools_count/\n",
            "kallisto_alignment/standard/8_ebov_snp_28_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/8_ebov_snp_28_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_28_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_06_ms/\n",
            "kallisto_alignment/standard/3_ebov_snp_06_ms/bustools_count/\n",
            "kallisto_alignment/standard/3_ebov_snp_06_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/3_ebov_snp_06_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_06_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_15_ms/\n",
            "kallisto_alignment/standard/4_ebov_snp_15_ms/bustools_count/\n",
            "kallisto_alignment/standard/4_ebov_snp_15_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/4_ebov_snp_15_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_15_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_26_ms/\n",
            "kallisto_alignment/standard/9_ebov_snp_26_ms/bustools_count/\n",
            "kallisto_alignment/standard/9_ebov_snp_26_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/9_ebov_snp_26_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_26_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_19_ms/\n",
            "kallisto_alignment/standard/8_ebov_snp_19_ms/bustools_count/\n",
            "kallisto_alignment/standard/8_ebov_snp_19_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/8_ebov_snp_19_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_19_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_26_ms/\n",
            "kallisto_alignment/standard/1_ebov_snp_26_ms/bustools_count/\n",
            "kallisto_alignment/standard/1_ebov_snp_26_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/1_ebov_snp_26_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_26_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_14_ms/\n",
            "kallisto_alignment/standard/4_ebov_snp_14_ms/bustools_count/\n",
            "kallisto_alignment/standard/4_ebov_snp_14_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/4_ebov_snp_14_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_14_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_13_ms/\n",
            "kallisto_alignment/standard/2_ebov_snp_13_ms/bustools_count/\n",
            "kallisto_alignment/standard/2_ebov_snp_13_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/2_ebov_snp_13_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_13_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_3_ms/\n",
            "kallisto_alignment/standard/0_ebov_snp_3_ms/bustools_count/\n",
            "kallisto_alignment/standard/0_ebov_snp_3_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/0_ebov_snp_3_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_3_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_15_ms/\n",
            "kallisto_alignment/standard/5_ebov_snp_15_ms/bustools_count/\n",
            "kallisto_alignment/standard/5_ebov_snp_15_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/5_ebov_snp_15_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_15_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_18_ms/\n",
            "kallisto_alignment/standard/3_ebov_snp_18_ms/bustools_count/\n",
            "kallisto_alignment/standard/3_ebov_snp_18_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/3_ebov_snp_18_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_18_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_24_ms/\n",
            "kallisto_alignment/standard/1_ebov_snp_24_ms/bustools_count/\n",
            "kallisto_alignment/standard/1_ebov_snp_24_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/1_ebov_snp_24_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_24_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_27_ms/\n",
            "kallisto_alignment/standard/3_ebov_snp_27_ms/bustools_count/\n",
            "kallisto_alignment/standard/3_ebov_snp_27_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/3_ebov_snp_27_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_27_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_19_ms/\n",
            "kallisto_alignment/standard/6_ebov_snp_19_ms/bustools_count/\n",
            "kallisto_alignment/standard/6_ebov_snp_19_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/6_ebov_snp_19_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_19_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_3_ms/\n",
            "kallisto_alignment/standard/5_ebov_snp_3_ms/bustools_count/\n",
            "kallisto_alignment/standard/5_ebov_snp_3_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/5_ebov_snp_3_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_3_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_13_ms/\n",
            "kallisto_alignment/standard/6_ebov_snp_13_ms/bustools_count/\n",
            "kallisto_alignment/standard/6_ebov_snp_13_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/6_ebov_snp_13_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_13_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_22_ms/\n",
            "kallisto_alignment/standard/3_ebov_snp_22_ms/bustools_count/\n",
            "kallisto_alignment/standard/3_ebov_snp_22_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/3_ebov_snp_22_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_22_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_1_ms/\n",
            "kallisto_alignment/standard/3_ebov_snp_1_ms/bustools_count/\n",
            "kallisto_alignment/standard/3_ebov_snp_1_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/3_ebov_snp_1_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_1_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_02_ms/\n",
            "kallisto_alignment/standard/7_ebov_snp_02_ms/bustools_count/\n",
            "kallisto_alignment/standard/7_ebov_snp_02_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/7_ebov_snp_02_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_02_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_27_ms/\n",
            "kallisto_alignment/standard/0_ebov_snp_27_ms/bustools_count/\n",
            "kallisto_alignment/standard/0_ebov_snp_27_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/0_ebov_snp_27_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_27_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_21_ms/\n",
            "kallisto_alignment/standard/2_ebov_snp_21_ms/bustools_count/\n",
            "kallisto_alignment/standard/2_ebov_snp_21_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/2_ebov_snp_21_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_21_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_12_ms/\n",
            "kallisto_alignment/standard/9_ebov_snp_12_ms/bustools_count/\n",
            "kallisto_alignment/standard/9_ebov_snp_12_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/9_ebov_snp_12_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_12_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_18_ms/\n",
            "kallisto_alignment/standard/4_ebov_snp_18_ms/bustools_count/\n",
            "kallisto_alignment/standard/4_ebov_snp_18_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/4_ebov_snp_18_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_18_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_16_ms/\n",
            "kallisto_alignment/standard/9_ebov_snp_16_ms/bustools_count/\n",
            "kallisto_alignment/standard/9_ebov_snp_16_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/9_ebov_snp_16_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_16_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_06_ms/\n",
            "kallisto_alignment/standard/7_ebov_snp_06_ms/bustools_count/\n",
            "kallisto_alignment/standard/7_ebov_snp_06_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/7_ebov_snp_06_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_06_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_22_ms/\n",
            "kallisto_alignment/standard/7_ebov_snp_22_ms/bustools_count/\n",
            "kallisto_alignment/standard/7_ebov_snp_22_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/7_ebov_snp_22_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_22_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_27_ms/\n",
            "kallisto_alignment/standard/2_ebov_snp_27_ms/bustools_count/\n",
            "kallisto_alignment/standard/2_ebov_snp_27_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/2_ebov_snp_27_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_27_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_14_ms/\n",
            "kallisto_alignment/standard/5_ebov_snp_14_ms/bustools_count/\n",
            "kallisto_alignment/standard/5_ebov_snp_14_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/5_ebov_snp_14_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_14_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_2_ms/\n",
            "kallisto_alignment/standard/9_ebov_snp_2_ms/bustools_count/\n",
            "kallisto_alignment/standard/9_ebov_snp_2_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/9_ebov_snp_2_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_2_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_07_ms/\n",
            "kallisto_alignment/standard/4_ebov_snp_07_ms/bustools_count/\n",
            "kallisto_alignment/standard/4_ebov_snp_07_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/4_ebov_snp_07_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_07_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_14_ms/\n",
            "kallisto_alignment/standard/2_ebov_snp_14_ms/bustools_count/\n",
            "kallisto_alignment/standard/2_ebov_snp_14_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/2_ebov_snp_14_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_14_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_1_ms/\n",
            "kallisto_alignment/standard/9_ebov_snp_1_ms/bustools_count/\n",
            "kallisto_alignment/standard/9_ebov_snp_1_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/9_ebov_snp_1_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_1_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_27_ms/\n",
            "kallisto_alignment/standard/7_ebov_snp_27_ms/bustools_count/\n",
            "kallisto_alignment/standard/7_ebov_snp_27_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/7_ebov_snp_27_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_27_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_16_ms/\n",
            "kallisto_alignment/standard/7_ebov_snp_16_ms/bustools_count/\n",
            "kallisto_alignment/standard/7_ebov_snp_16_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/7_ebov_snp_16_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_16_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_3_ms/\n",
            "kallisto_alignment/standard/2_ebov_snp_3_ms/bustools_count/\n",
            "kallisto_alignment/standard/2_ebov_snp_3_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/2_ebov_snp_3_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_3_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_13_ms/\n",
            "kallisto_alignment/standard/5_ebov_snp_13_ms/bustools_count/\n",
            "kallisto_alignment/standard/5_ebov_snp_13_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/5_ebov_snp_13_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_13_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_04_ms/\n",
            "kallisto_alignment/standard/4_ebov_snp_04_ms/bustools_count/\n",
            "kallisto_alignment/standard/4_ebov_snp_04_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/4_ebov_snp_04_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_04_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_13_ms/\n",
            "kallisto_alignment/standard/3_ebov_snp_13_ms/bustools_count/\n",
            "kallisto_alignment/standard/3_ebov_snp_13_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/3_ebov_snp_13_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_13_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_04_ms/\n",
            "kallisto_alignment/standard/2_ebov_snp_04_ms/bustools_count/\n",
            "kallisto_alignment/standard/2_ebov_snp_04_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/2_ebov_snp_04_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_04_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_12_ms/\n",
            "kallisto_alignment/standard/5_ebov_snp_12_ms/bustools_count/\n",
            "kallisto_alignment/standard/5_ebov_snp_12_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/5_ebov_snp_12_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_12_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_08_ms/\n",
            "kallisto_alignment/standard/0_ebov_snp_08_ms/bustools_count/\n",
            "kallisto_alignment/standard/0_ebov_snp_08_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/0_ebov_snp_08_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_08_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_22_ms/\n",
            "kallisto_alignment/standard/9_ebov_snp_22_ms/bustools_count/\n",
            "kallisto_alignment/standard/9_ebov_snp_22_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/9_ebov_snp_22_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_22_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_16_ms/\n",
            "kallisto_alignment/standard/3_ebov_snp_16_ms/bustools_count/\n",
            "kallisto_alignment/standard/3_ebov_snp_16_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/3_ebov_snp_16_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_16_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_24_ms/\n",
            "kallisto_alignment/standard/9_ebov_snp_24_ms/bustools_count/\n",
            "kallisto_alignment/standard/9_ebov_snp_24_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/9_ebov_snp_24_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_24_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_07_ms/\n",
            "kallisto_alignment/standard/9_ebov_snp_07_ms/bustools_count/\n",
            "kallisto_alignment/standard/9_ebov_snp_07_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/9_ebov_snp_07_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_07_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_05_ms/\n",
            "kallisto_alignment/standard/0_ebov_snp_05_ms/bustools_count/\n",
            "kallisto_alignment/standard/0_ebov_snp_05_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/0_ebov_snp_05_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_05_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_17_ms/\n",
            "kallisto_alignment/standard/2_ebov_snp_17_ms/bustools_count/\n",
            "kallisto_alignment/standard/2_ebov_snp_17_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/2_ebov_snp_17_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_17_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_19_ms/\n",
            "kallisto_alignment/standard/3_ebov_snp_19_ms/bustools_count/\n",
            "kallisto_alignment/standard/3_ebov_snp_19_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/3_ebov_snp_19_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_19_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_27_ms/\n",
            "kallisto_alignment/standard/6_ebov_snp_27_ms/bustools_count/\n",
            "kallisto_alignment/standard/6_ebov_snp_27_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/6_ebov_snp_27_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_27_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_27_ms/\n",
            "kallisto_alignment/standard/5_ebov_snp_27_ms/bustools_count/\n",
            "kallisto_alignment/standard/5_ebov_snp_27_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/5_ebov_snp_27_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_27_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_18_ms/\n",
            "kallisto_alignment/standard/6_ebov_snp_18_ms/bustools_count/\n",
            "kallisto_alignment/standard/6_ebov_snp_18_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/6_ebov_snp_18_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_18_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_16_ms/\n",
            "kallisto_alignment/standard/0_ebov_snp_16_ms/bustools_count/\n",
            "kallisto_alignment/standard/0_ebov_snp_16_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/0_ebov_snp_16_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_16_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_17_ms/\n",
            "kallisto_alignment/standard/4_ebov_snp_17_ms/bustools_count/\n",
            "kallisto_alignment/standard/4_ebov_snp_17_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/4_ebov_snp_17_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_17_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_06_ms/\n",
            "kallisto_alignment/standard/6_ebov_snp_06_ms/bustools_count/\n",
            "kallisto_alignment/standard/6_ebov_snp_06_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/6_ebov_snp_06_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_06_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_22_ms/\n",
            "kallisto_alignment/standard/6_ebov_snp_22_ms/bustools_count/\n",
            "kallisto_alignment/standard/6_ebov_snp_22_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/6_ebov_snp_22_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_22_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_24_ms/\n",
            "kallisto_alignment/standard/0_ebov_snp_24_ms/bustools_count/\n",
            "kallisto_alignment/standard/0_ebov_snp_24_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/0_ebov_snp_24_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_24_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_29_ms/\n",
            "kallisto_alignment/standard/5_ebov_snp_29_ms/bustools_count/\n",
            "kallisto_alignment/standard/5_ebov_snp_29_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/5_ebov_snp_29_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_29_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_25_ms/\n",
            "kallisto_alignment/standard/1_ebov_snp_25_ms/bustools_count/\n",
            "kallisto_alignment/standard/1_ebov_snp_25_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/1_ebov_snp_25_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_25_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_02_ms/\n",
            "kallisto_alignment/standard/1_ebov_snp_02_ms/bustools_count/\n",
            "kallisto_alignment/standard/1_ebov_snp_02_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/1_ebov_snp_02_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_02_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_26_ms/\n",
            "kallisto_alignment/standard/4_ebov_snp_26_ms/bustools_count/\n",
            "kallisto_alignment/standard/4_ebov_snp_26_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/4_ebov_snp_26_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_26_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_02_ms/\n",
            "kallisto_alignment/standard/9_ebov_snp_02_ms/bustools_count/\n",
            "kallisto_alignment/standard/9_ebov_snp_02_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/9_ebov_snp_02_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_02_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_16_ms/\n",
            "kallisto_alignment/standard/8_ebov_snp_16_ms/bustools_count/\n",
            "kallisto_alignment/standard/8_ebov_snp_16_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/8_ebov_snp_16_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_16_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_17_ms/\n",
            "kallisto_alignment/standard/7_ebov_snp_17_ms/bustools_count/\n",
            "kallisto_alignment/standard/7_ebov_snp_17_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/7_ebov_snp_17_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_17_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_3_ms/\n",
            "kallisto_alignment/standard/4_ebov_snp_3_ms/bustools_count/\n",
            "kallisto_alignment/standard/4_ebov_snp_3_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/4_ebov_snp_3_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_3_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_16_ms/\n",
            "kallisto_alignment/standard/5_ebov_snp_16_ms/bustools_count/\n",
            "kallisto_alignment/standard/5_ebov_snp_16_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/5_ebov_snp_16_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_16_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_27_ms/\n",
            "kallisto_alignment/standard/4_ebov_snp_27_ms/bustools_count/\n",
            "kallisto_alignment/standard/4_ebov_snp_27_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/4_ebov_snp_27_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_27_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_04_ms/\n",
            "kallisto_alignment/standard/9_ebov_snp_04_ms/bustools_count/\n",
            "kallisto_alignment/standard/9_ebov_snp_04_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/9_ebov_snp_04_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_04_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_12_ms/\n",
            "kallisto_alignment/standard/0_ebov_snp_12_ms/bustools_count/\n",
            "kallisto_alignment/standard/0_ebov_snp_12_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/0_ebov_snp_12_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_12_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_23_ms/\n",
            "kallisto_alignment/standard/8_ebov_snp_23_ms/bustools_count/\n",
            "kallisto_alignment/standard/8_ebov_snp_23_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/8_ebov_snp_23_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_23_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_1_ms/\n",
            "kallisto_alignment/standard/0_ebov_snp_1_ms/bustools_count/\n",
            "kallisto_alignment/standard/0_ebov_snp_1_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/0_ebov_snp_1_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_1_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_16_ms/\n",
            "kallisto_alignment/standard/4_ebov_snp_16_ms/bustools_count/\n",
            "kallisto_alignment/standard/4_ebov_snp_16_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/4_ebov_snp_16_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_16_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_13_ms/\n",
            "kallisto_alignment/standard/4_ebov_snp_13_ms/bustools_count/\n",
            "kallisto_alignment/standard/4_ebov_snp_13_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/4_ebov_snp_13_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_13_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_07_ms/\n",
            "kallisto_alignment/standard/0_ebov_snp_07_ms/bustools_count/\n",
            "kallisto_alignment/standard/0_ebov_snp_07_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/0_ebov_snp_07_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_07_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_26_ms/\n",
            "kallisto_alignment/standard/0_ebov_snp_26_ms/bustools_count/\n",
            "kallisto_alignment/standard/0_ebov_snp_26_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/0_ebov_snp_26_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_26_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_21_ms/\n",
            "kallisto_alignment/standard/8_ebov_snp_21_ms/bustools_count/\n",
            "kallisto_alignment/standard/8_ebov_snp_21_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/8_ebov_snp_21_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_21_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_11_ms/\n",
            "kallisto_alignment/standard/0_ebov_snp_11_ms/bustools_count/\n",
            "kallisto_alignment/standard/0_ebov_snp_11_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/0_ebov_snp_11_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_11_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_11_ms/\n",
            "kallisto_alignment/standard/7_ebov_snp_11_ms/bustools_count/\n",
            "kallisto_alignment/standard/7_ebov_snp_11_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/7_ebov_snp_11_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_11_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_07_ms/\n",
            "kallisto_alignment/standard/2_ebov_snp_07_ms/bustools_count/\n",
            "kallisto_alignment/standard/2_ebov_snp_07_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/2_ebov_snp_07_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_07_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_21_ms/\n",
            "kallisto_alignment/standard/6_ebov_snp_21_ms/bustools_count/\n",
            "kallisto_alignment/standard/6_ebov_snp_21_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/6_ebov_snp_21_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_21_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_17_ms/\n",
            "kallisto_alignment/standard/6_ebov_snp_17_ms/bustools_count/\n",
            "kallisto_alignment/standard/6_ebov_snp_17_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/6_ebov_snp_17_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_17_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_29_ms/\n",
            "kallisto_alignment/standard/2_ebov_snp_29_ms/bustools_count/\n",
            "kallisto_alignment/standard/2_ebov_snp_29_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/2_ebov_snp_29_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_29_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_06_ms/\n",
            "kallisto_alignment/standard/9_ebov_snp_06_ms/bustools_count/\n",
            "kallisto_alignment/standard/9_ebov_snp_06_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/9_ebov_snp_06_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_06_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_25_ms/\n",
            "kallisto_alignment/standard/3_ebov_snp_25_ms/bustools_count/\n",
            "kallisto_alignment/standard/3_ebov_snp_25_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/3_ebov_snp_25_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_25_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_25_ms/\n",
            "kallisto_alignment/standard/0_ebov_snp_25_ms/bustools_count/\n",
            "kallisto_alignment/standard/0_ebov_snp_25_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/0_ebov_snp_25_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_25_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_27_ms/\n",
            "kallisto_alignment/standard/9_ebov_snp_27_ms/bustools_count/\n",
            "kallisto_alignment/standard/9_ebov_snp_27_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/9_ebov_snp_27_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_27_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_17_ms/\n",
            "kallisto_alignment/standard/8_ebov_snp_17_ms/bustools_count/\n",
            "kallisto_alignment/standard/8_ebov_snp_17_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/8_ebov_snp_17_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_17_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_15_ms/\n",
            "kallisto_alignment/standard/0_ebov_snp_15_ms/bustools_count/\n",
            "kallisto_alignment/standard/0_ebov_snp_15_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/0_ebov_snp_15_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_15_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_03_ms/\n",
            "kallisto_alignment/standard/1_ebov_snp_03_ms/bustools_count/\n",
            "kallisto_alignment/standard/1_ebov_snp_03_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/1_ebov_snp_03_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_03_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_28_ms/\n",
            "kallisto_alignment/standard/6_ebov_snp_28_ms/bustools_count/\n",
            "kallisto_alignment/standard/6_ebov_snp_28_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/6_ebov_snp_28_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_28_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_1_ms/\n",
            "kallisto_alignment/standard/2_ebov_snp_1_ms/bustools_count/\n",
            "kallisto_alignment/standard/2_ebov_snp_1_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/2_ebov_snp_1_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_1_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_17_ms/\n",
            "kallisto_alignment/standard/9_ebov_snp_17_ms/bustools_count/\n",
            "kallisto_alignment/standard/9_ebov_snp_17_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/9_ebov_snp_17_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_17_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_26_ms/\n",
            "kallisto_alignment/standard/5_ebov_snp_26_ms/bustools_count/\n",
            "kallisto_alignment/standard/5_ebov_snp_26_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/5_ebov_snp_26_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_26_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_23_ms/\n",
            "kallisto_alignment/standard/7_ebov_snp_23_ms/bustools_count/\n",
            "kallisto_alignment/standard/7_ebov_snp_23_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/7_ebov_snp_23_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_23_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_26_ms/\n",
            "kallisto_alignment/standard/3_ebov_snp_26_ms/bustools_count/\n",
            "kallisto_alignment/standard/3_ebov_snp_26_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/3_ebov_snp_26_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_26_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_05_ms/\n",
            "kallisto_alignment/standard/3_ebov_snp_05_ms/bustools_count/\n",
            "kallisto_alignment/standard/3_ebov_snp_05_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/3_ebov_snp_05_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_05_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_22_ms/\n",
            "kallisto_alignment/standard/5_ebov_snp_22_ms/bustools_count/\n",
            "kallisto_alignment/standard/5_ebov_snp_22_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/5_ebov_snp_22_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_22_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_15_ms/\n",
            "kallisto_alignment/standard/9_ebov_snp_15_ms/bustools_count/\n",
            "kallisto_alignment/standard/9_ebov_snp_15_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/9_ebov_snp_15_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_15_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_18_ms/\n",
            "kallisto_alignment/standard/8_ebov_snp_18_ms/bustools_count/\n",
            "kallisto_alignment/standard/8_ebov_snp_18_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/8_ebov_snp_18_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_18_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_23_ms/\n",
            "kallisto_alignment/standard/3_ebov_snp_23_ms/bustools_count/\n",
            "kallisto_alignment/standard/3_ebov_snp_23_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/3_ebov_snp_23_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_23_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_05_ms/\n",
            "kallisto_alignment/standard/4_ebov_snp_05_ms/bustools_count/\n",
            "kallisto_alignment/standard/4_ebov_snp_05_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/4_ebov_snp_05_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_05_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_11_ms/\n",
            "kallisto_alignment/standard/9_ebov_snp_11_ms/bustools_count/\n",
            "kallisto_alignment/standard/9_ebov_snp_11_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/9_ebov_snp_11_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_11_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_09_ms/\n",
            "kallisto_alignment/standard/7_ebov_snp_09_ms/bustools_count/\n",
            "kallisto_alignment/standard/7_ebov_snp_09_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/7_ebov_snp_09_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_09_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_15_ms/\n",
            "kallisto_alignment/standard/6_ebov_snp_15_ms/bustools_count/\n",
            "kallisto_alignment/standard/6_ebov_snp_15_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/6_ebov_snp_15_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_15_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_03_ms/\n",
            "kallisto_alignment/standard/3_ebov_snp_03_ms/bustools_count/\n",
            "kallisto_alignment/standard/3_ebov_snp_03_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/3_ebov_snp_03_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/3_ebov_snp_03_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_12_ms/\n",
            "kallisto_alignment/standard/7_ebov_snp_12_ms/bustools_count/\n",
            "kallisto_alignment/standard/7_ebov_snp_12_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/7_ebov_snp_12_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_12_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_25_ms/\n",
            "kallisto_alignment/standard/8_ebov_snp_25_ms/bustools_count/\n",
            "kallisto_alignment/standard/8_ebov_snp_25_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/8_ebov_snp_25_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_25_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_03_ms/\n",
            "kallisto_alignment/standard/9_ebov_snp_03_ms/bustools_count/\n",
            "kallisto_alignment/standard/9_ebov_snp_03_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/9_ebov_snp_03_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_03_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_2_ms/\n",
            "kallisto_alignment/standard/8_ebov_snp_2_ms/bustools_count/\n",
            "kallisto_alignment/standard/8_ebov_snp_2_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/8_ebov_snp_2_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_2_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_03_ms/\n",
            "kallisto_alignment/standard/0_ebov_snp_03_ms/bustools_count/\n",
            "kallisto_alignment/standard/0_ebov_snp_03_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/0_ebov_snp_03_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_03_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_28_ms/\n",
            "kallisto_alignment/standard/1_ebov_snp_28_ms/bustools_count/\n",
            "kallisto_alignment/standard/1_ebov_snp_28_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/1_ebov_snp_28_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_28_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_26_ms/\n",
            "kallisto_alignment/standard/8_ebov_snp_26_ms/bustools_count/\n",
            "kallisto_alignment/standard/8_ebov_snp_26_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/8_ebov_snp_26_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_26_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_14_ms/\n",
            "kallisto_alignment/standard/6_ebov_snp_14_ms/bustools_count/\n",
            "kallisto_alignment/standard/6_ebov_snp_14_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/6_ebov_snp_14_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_14_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_08_ms/\n",
            "kallisto_alignment/standard/7_ebov_snp_08_ms/bustools_count/\n",
            "kallisto_alignment/standard/7_ebov_snp_08_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/7_ebov_snp_08_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_08_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_03_ms/\n",
            "kallisto_alignment/standard/5_ebov_snp_03_ms/bustools_count/\n",
            "kallisto_alignment/standard/5_ebov_snp_03_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/5_ebov_snp_03_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_03_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_29_ms/\n",
            "kallisto_alignment/standard/4_ebov_snp_29_ms/bustools_count/\n",
            "kallisto_alignment/standard/4_ebov_snp_29_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/4_ebov_snp_29_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_29_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_13_ms/\n",
            "kallisto_alignment/standard/9_ebov_snp_13_ms/bustools_count/\n",
            "kallisto_alignment/standard/9_ebov_snp_13_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/9_ebov_snp_13_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_13_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_3_ms/\n",
            "kallisto_alignment/standard/1_ebov_snp_3_ms/bustools_count/\n",
            "kallisto_alignment/standard/1_ebov_snp_3_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/1_ebov_snp_3_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_3_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_2_ms/\n",
            "kallisto_alignment/standard/2_ebov_snp_2_ms/bustools_count/\n",
            "kallisto_alignment/standard/2_ebov_snp_2_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/2_ebov_snp_2_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_2_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_25_ms/\n",
            "kallisto_alignment/standard/9_ebov_snp_25_ms/bustools_count/\n",
            "kallisto_alignment/standard/9_ebov_snp_25_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/9_ebov_snp_25_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_25_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_28_ms/\n",
            "kallisto_alignment/standard/0_ebov_snp_28_ms/bustools_count/\n",
            "kallisto_alignment/standard/0_ebov_snp_28_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/0_ebov_snp_28_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_28_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_17_ms/\n",
            "kallisto_alignment/standard/0_ebov_snp_17_ms/bustools_count/\n",
            "kallisto_alignment/standard/0_ebov_snp_17_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/0_ebov_snp_17_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_17_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_23_ms/\n",
            "kallisto_alignment/standard/6_ebov_snp_23_ms/bustools_count/\n",
            "kallisto_alignment/standard/6_ebov_snp_23_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/6_ebov_snp_23_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/6_ebov_snp_23_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_07_ms/\n",
            "kallisto_alignment/standard/5_ebov_snp_07_ms/bustools_count/\n",
            "kallisto_alignment/standard/5_ebov_snp_07_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/5_ebov_snp_07_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_07_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_09_ms/\n",
            "kallisto_alignment/standard/9_ebov_snp_09_ms/bustools_count/\n",
            "kallisto_alignment/standard/9_ebov_snp_09_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/9_ebov_snp_09_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_09_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_04_ms/\n",
            "kallisto_alignment/standard/7_ebov_snp_04_ms/bustools_count/\n",
            "kallisto_alignment/standard/7_ebov_snp_04_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/7_ebov_snp_04_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/7_ebov_snp_04_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_11_ms/\n",
            "kallisto_alignment/standard/1_ebov_snp_11_ms/bustools_count/\n",
            "kallisto_alignment/standard/1_ebov_snp_11_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/1_ebov_snp_11_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_11_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_29_ms/\n",
            "kallisto_alignment/standard/0_ebov_snp_29_ms/bustools_count/\n",
            "kallisto_alignment/standard/0_ebov_snp_29_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/0_ebov_snp_29_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/0_ebov_snp_29_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_1_ms/\n",
            "kallisto_alignment/standard/4_ebov_snp_1_ms/bustools_count/\n",
            "kallisto_alignment/standard/4_ebov_snp_1_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/4_ebov_snp_1_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_1_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_21_ms/\n",
            "kallisto_alignment/standard/9_ebov_snp_21_ms/bustools_count/\n",
            "kallisto_alignment/standard/9_ebov_snp_21_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/9_ebov_snp_21_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/9_ebov_snp_21_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_26_ms/\n",
            "kallisto_alignment/standard/2_ebov_snp_26_ms/bustools_count/\n",
            "kallisto_alignment/standard/2_ebov_snp_26_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/2_ebov_snp_26_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/2_ebov_snp_26_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_24_ms/\n",
            "kallisto_alignment/standard/8_ebov_snp_24_ms/bustools_count/\n",
            "kallisto_alignment/standard/8_ebov_snp_24_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/8_ebov_snp_24_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/8_ebov_snp_24_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_19_ms/\n",
            "kallisto_alignment/standard/4_ebov_snp_19_ms/bustools_count/\n",
            "kallisto_alignment/standard/4_ebov_snp_19_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/4_ebov_snp_19_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/4_ebov_snp_19_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_1_ms/\n",
            "kallisto_alignment/standard/1_ebov_snp_1_ms/bustools_count/\n",
            "kallisto_alignment/standard/1_ebov_snp_1_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/1_ebov_snp_1_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/1_ebov_snp_1_ms/bustools_count/output.barcodes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_05_ms/\n",
            "kallisto_alignment/standard/5_ebov_snp_05_ms/bustools_count/\n",
            "kallisto_alignment/standard/5_ebov_snp_05_ms/bustools_count/output.mtx\n",
            "kallisto_alignment/standard/5_ebov_snp_05_ms/bustools_count/output.genes.txt\n",
            "kallisto_alignment/standard/5_ebov_snp_05_ms/bustools_count/output.barcodes.txt\n"
          ]
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<IPython.core.display.Javascript object>"
            ],
            "application/javascript": [
              "\n",
              "    async function download(id, filename, size) {\n",
              "      if (!google.colab.kernel.accessAllowed) {\n",
              "        return;\n",
              "      }\n",
              "      const div = document.createElement('div');\n",
              "      const label = document.createElement('label');\n",
              "      label.textContent = `Downloading \"${filename}\": `;\n",
              "      div.appendChild(label);\n",
              "      const progress = document.createElement('progress');\n",
              "      progress.max = size;\n",
              "      div.appendChild(progress);\n",
              "      document.body.appendChild(div);\n",
              "\n",
              "      const buffers = [];\n",
              "      let downloaded = 0;\n",
              "\n",
              "      const channel = await google.colab.kernel.comms.open(id);\n",
              "      // Send a message to notify the kernel that we're ready.\n",
              "      channel.send({})\n",
              "\n",
              "      for await (const message of channel.messages) {\n",
              "        // Send a message to notify the kernel that we're ready.\n",
              "        channel.send({})\n",
              "        if (message.buffers) {\n",
              "          for (const buffer of message.buffers) {\n",
              "            buffers.push(buffer);\n",
              "            downloaded += buffer.byteLength;\n",
              "            progress.value = downloaded;\n",
              "          }\n",
              "        }\n",
              "      }\n",
              "      const blob = new Blob(buffers, {type: 'application/binary'});\n",
              "      const a = document.createElement('a');\n",
              "      a.href = window.URL.createObjectURL(blob);\n",
              "      a.download = filename;\n",
              "      div.appendChild(a);\n",
              "      a.click();\n",
              "      div.remove();\n",
              "    }\n",
              "  "
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<IPython.core.display.Javascript object>"
            ],
            "application/javascript": [
              "download(\"download_71bfde0b-bb02-41cf-92b4-f24e4182ab49\", \"kallisto_alignment.tar.gz\", 66725121)"
            ]
          },
          "metadata": {}
        }
      ],
      "source": [
        "from google.colab import files\n",
        "folders_to_download = [\"kraken2_alignment\", \"kallisto_alignment\", \"diamond_alignment\"]\n",
        "\n",
        "for folder in folders_to_download:\n",
        "  zipped = f\"{folder}.tar.gz\"\n",
        "  !tar -zcvf $zipped $folder\n",
        "  files.download(zipped)"
      ]
    },
    {
      "cell_type": "code",
      "source": [],
      "metadata": {
        "id": "PbxYY4w58fLm"
      },
      "id": "PbxYY4w58fLm",
      "execution_count": null,
      "outputs": []
    }
  ],
  "metadata": {
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "codemirror_mode": {
        "name": "ipython",
        "version": 3
      },
      "file_extension": ".py",
      "mimetype": "text/x-python",
      "name": "python",
      "nbconvert_exporter": "python",
      "pygments_lexer": "ipython3",
      "version": "3.9.12"
    },
    "colab": {
      "provenance": [],
      "gpuType": "V28",
      "include_colab_link": true
    },
    "accelerator": "TPU"
  },
  "nbformat": 4,
  "nbformat_minor": 5
}