If you are using kallisto, bustools, or kb-python in a publication:   

- **Please cite:**  
   protocols paper

- **When performing alignment of single-cell RNA seq data, please also cite:**  
   Melsted P*, Booeshaghi AS*, Liu L, Gao F, Lu Lambda, Min KH, da Veiga Beltrame E, Hjörleifsson KE, Gehring J, Pachter L. `Modular, efficient and constant-memory single-cell RNA-seq preprocessing. <https://doi.org/10.1038/s41587-021-00870-2>`_ Nature Biotechnology 39, 813–818 (2021). https://doi.org/10.1038/s41587-021-00870-2

- **When performing alignment of bulk RNA seq data or referencing pseudo-alignment, please also cite:**  
   Bray N, Pimentel H, Melsted P, Pachter L. `Near-optimal probabilistic RNA-seq quantification. <https://doi.org/10.1038/nbt.3519>`_ Nature Biotechnology 34, 525–527 (2016). https://doi.org/10.1038/nbt.3519

- **When analyzing long-read RNA seq data, please also cite:**  
   Bekka's paper

- **When performing translated alignment, please also cite:**  
   Veira paper

**Additional relevant paper(s):**  

Melsted P*, Ntranos V*, Pachter L*. `The barcode, UMI, set format and BUStools. <https://doi.org/10.1093/bioinformatics/btz279>`_ Bioinformatics 35(11), 4472–4473 (2019). https://doi.org/10.1093/bioinformatics/btz279
