.. _Overview:References:

References
===============

If you are using kallisto, bustools, or kb-python in a publication:   

**Please cite:**  
   Delaney K. Sullivan, Kyung Hoi (Joseph) Min, Kristján Eldjárn Hjörleifsson, Laura Luebbert, Guillaume Holley, Lambda Moses, Johan Gustafsson, Nicolas L. Bray, Harold Pimentel, A. Sina Booeshaghi, Páll Melsted, Lior Pachter. `kallisto, bustools and kb-python for quantifying bulk, single-cell and single-nucleus RNA-seq. <https://doi.org/10.1038/s41596-024-01057-0>`_ Nature Protocols 20, 587-607 (2025). https://doi.org/10.1038/s41596-024-01057-0

- **When performing alignment of single-cell RNA seq data, please also cite:**  
   Melsted P*, Booeshaghi AS*, Liu L, Gao F, Lu Lambda, Min KH, da Veiga Beltrame E, Hjörleifsson KE, Gehring J, Pachter L. `Modular, efficient and constant-memory single-cell RNA-seq preprocessing. <https://doi.org/10.1038/s41587-021-00870-2>`_ Nature Biotechnology 39, 813–818 (2021). https://doi.org/10.1038/s41587-021-00870-2

- **When performing alignment of bulk RNA seq data or referencing pseudo-alignment, please also cite:**  
   Bray N, Pimentel H, Melsted P, Pachter L. `Near-optimal probabilistic RNA-seq quantification. <https://doi.org/10.1038/nbt.3519>`_ Nature Biotechnology 34, 525–527 (2016). https://doi.org/10.1038/nbt.3519

- **When analyzing long-read RNA seq data, please also cite:**  
   Rebekah Loving, Delaney K. Sullivan, Fairlie Reese, Elisabeth Rebboah, Jasmine Sakr, Narges Rezaie, Heidi Y. Liang, Ghassan Filimban, Shimako Kawauchi, Conrad Oakes, Diane Trout, Brian A. Williams, Grant MacGregor, Barbara J. Wold, Ali Mortazavi, Lior Pachter. `Long-read sequencing transcriptome quantification with lr-kallisto. <https://doi.org/10.1101/2024.07.19.604364>`_ bioRxiv 2024.07.19.604364; doi: https://doi.org/10.1101/2024.07.19.604364

- **When performing translated alignment, please also cite:**  
   Luebbert L, Sullivan DK, Carilli M, Eldjárn Hjörleifsson K, Viloria Winnett A, Chari T, Pachter L. `Detection of viral sequences at single-cell resolution identifies novel viruses associated with host gene expression changes. <https://doi.org/10.1038/s41587-025-02614-y>`_ Nature Biotechnology (2025). https://doi.org/10.1038/s41587-025-02614-y

**Additional relevant paper(s):**  

   Melsted P*, Ntranos V*, Pachter L*. `The barcode, UMI, set format and BUStools. <https://doi.org/10.1093/bioinformatics/btz279>`_ Bioinformatics 35(11), 4472–4473 (2019). https://doi.org/10.1093/bioinformatics/btz279

   Einarsson PH, Melsted P. `BUSZ: compressed BUS files. <https://doi.org/10.1093/bioinformatics/btad295>`_ Bioinformatics 39(5), btad295 (2023). https://doi.org/10.1093/bioinformatics/btad295
