Dependents
===============

The following applications build on **kallisto**, **bustools**, and/or **kb_python**:

- `ARCHS4: Massive Mining of Publicly Available RNA-seq Data from Human and Mouse <http://amp.pharm.mssm.edu/archs4/>`_ using **kallisto**

- `arkas: rapid and reproducible RNA-Seq analysis for end users <https://github.com/Al3n70rn/arkas>`_

- `DEE2: Digital Expression Explorer 2 <http://dee2.io/>`_

- `sleuth: Differential analysis of RNA-Seq <http://pachterlab.github.io/sleuth>`_

- `elysium: RNA-seq alignment in the cloud <https://github.com/maayanlab/elysium>`_

- `EndoDB: a database of endothelial cell transcriptomics data <https://endotheliomics.shinyapps.io/endodb/>`_

- `Google Cloud Pilot RNA Sequencing for CCLE and TCGA <https://osf.io/gqrz9/>`_

- `kallisto-sra: Nextflow implementation of kallisto RNA-Seq Tools fetching samples directly from SRA <https://github.com/lifebit-ai/kallisto-sra>`_

- `metaR: utilizing the Language Workbench Technology to facilitate data analysis in R <http://campagnelab.org/metar-1-8-0-released/>`_

- `MiPanda: Michigan Portal for the Analysis of NGS Data <http://www.mipanda.org/>`_

- `kallisto | bustools: Single-cell RNA-seq pre-processing <https://pachterlab.github.io/kallistobustools/>`_

- `SkeletalVis: exploration of skeletal biology related expression datasets <http://phenome.manchester.ac.uk/>`_

- `The Lair: interactive (re)analysis of published RNA-Seq data with **kallisto** and **sleuth** <http://pachterlab.github.io/lair>`_


